---
author: "Darlan Conterno Minussi"
date: "Last compiled on `r format(Sys.time(), '%d %B, %Y')`"
output: bookdown::gitbook
editor_options: 
  chunk_output_type: console
---

# MDA-231 DNA & RNA

Transcript abundances for expanded clones triplicates were quantified by Salmon (v.0.14) with GENCODE transcript v30 and options -l A -1 read1 -2 read2 -p 40 --validateMappings --seqBias --gcBias. Quantified transcripts were imported into R with ‘tximport’ (v 1.14). Expanded clones e7, e39 and e71 had one technical replicate excluded due to poor RNA quality

```{r setup_functions}
source("R/run_umap.R")
```


```{r data}
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Sat Mar  6 17:34:47 2021
# RNA
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Sat Mar  6 17:34:51 2021

# reading RNA data
dds <- readRDS(here("extdata/rna/mdamb231_rna_dds.rds"))

# Quality filtering
keep <- rowSums(counts(dds) >= 5) >= 3
dds <- dds[keep,]

# computation of size factors which normalize for 
# differences in sequencing depth among samples
dds <- estimateSizeFactors(dds)
boxplot(log10(counts(dds,normalized=TRUE)+1), las = 2)

# variance-stabilizing transformation
vsd <- vst(dds)

# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Sat Mar  6 17:34:56 2021
# DNA
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Sat Mar  6 17:35:01 2021

mda_ploidy <- 2.41

# data from DNA bulk expanded clones 
# and from mda-mb-231 single-cell were merged to the same dataframe

mdamb231_sc_bulk_cna_ml <- readRDS(here("extdata/merge_levels/mdamb231_sc_bulk_cna_ml.rds"))
```

## Co-clustering

```{r}
mdamb231_sc_bulk_umap <- run_umap(mdamb231_sc_bulk_cna_ml,
                                  umap_n_neighbors = 25,
                                  seed = 5)

mdamb231_sc_bulk_clustering <- run_clustering(mdamb231_sc_bulk_umap,
                                 k_snn_major = 43,
                                 k_snn_minor = 14) 

mdamb231_sc_bulk_clustering_umap <- left_join(mdamb231_sc_bulk_umap,
                                              mdamb231_sc_bulk_clustering, 
                                              by = c("cell" = "cells"))

mdamb231_sc_bulk_clustering_umap <- mdamb231_sc_bulk_clustering_umap %>%
  mutate(
    type = case_when(str_detect(cell, "gDNA") ~ "bulk",
                     TRUE ~ subclones),
    shape = case_when(str_detect(cell, "gDNA") ~ 8,
                      TRUE ~ 20),
    size = case_when(str_detect(cell, "gDNA") ~ 1.2,
                     TRUE ~ 1),
  )

p1 <- mdamb231_sc_bulk_clustering_umap %>% arrange(desc(type)) %>% 
  ggplot() + 
  geom_point(aes(V1,V2, color = superclones), size = 6) +
  geom_point(aes(V1,V2, color = type, shape = shape, size = size), stroke = 1.5) +
  scale_color_manual(values = c(colors_vector$superclones, colors_vector$subclones, "bulk" = "black")) +
  scale_shape_identity() + 
  scale_size_identity() +
  theme_classic() +
  theme(axis.title.x=element_text(colour = "gray50", size = 20),
        axis.text.x= element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y = element_text(colour = "gray50", size = 20),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.line = element_blank(),
        legend.position = "none",
        panel.border = element_rect(color = "black", 
                                    fill = NA,
                                    size = 2),
        # legend.position = "null"
  )  +
  labs(color = "") + 
  xlab("umap1") +
  ylab("umap2") 

p1
```

```{r more_setup}
# this setup will be used to plot the heatmap later

mda_ordered <- order_dataset(popseg_long = mdamb231_sc_bulk_cna_ml,
                             clustering = mdamb231_sc_bulk_clustering)

mda_cocluster_consensus <- calculate_consensus(df = mda_ordered$dataset_ordered,
                                     clusters = mda_ordered$clustering_ordered$subclones)

mda_cocluster_me_tree <- run_me_tree(consensus_df = mda_cocluster_consensus,
                                     clusters = mdamb231_sc_bulk_clustering,
                                     ploidy_VAL = mda_ploidy)
```

## Single-cell + Bulk Heatmap

```{r sc_bulk_heat,  fig.height=8, warning=FALSE}
plot_heatmap(df = mda_ordered$dataset_ordered,
             ploidy_VAL = mda_ploidy,
             ploidy_trunc = 2*(round(mda_ploidy))+2,
             clusters = mda_ordered$clustering_ordered,
             genomic_classes = NULL,
             keep_gene = NULL,
             tree_order = mda_cocluster_me_tree$cs_tree_order,
             show_legend = TRUE,
             mda_cocluster = TRUE)
```

## Bulk Heatmap
```{r bulk_heatmap, warning=FALSE}
mdamb231_bulk_cna_long <- mda_ordered$dataset_ordered[str_detect(rownames(mda_ordered$dataset_ordered), "gDNA"),]

# clustering info
mdamb231_bulk_cl_info <- mda_ordered$clustering_ordered %>% 
  filter(str_detect(cells, "gDNA")) %>% 
  arrange(superclones, subclones)

# matching order
mdamb231_bulk_ordered <- mdamb231_bulk_cna_long[mdamb231_bulk_cl_info$cells,]

plot_heatmap(df = mdamb231_bulk_ordered,
             ploidy_VAL = mda_ploidy,
             ploidy_trunc = 2*(round(mda_ploidy))+2,
             clusters = mdamb231_bulk_cl_info,
             genomic_classes = NULL,
             keep_gene = NULL,
             tree_order = NULL,
             show_legend = TRUE,
             mda_cocluster = TRUE)
```

## DNA & RNA

```{r merging, message = FALSE}
# Biomart
grch37 <-
  useMart(
    biomart = "ENSEMBL_MART_ENSEMBL",
    host = "grch37.ensembl.org",
    path = "/biomart/martservice",
    dataset = "hsapiens_gene_ensembl"
  )

# hg19 chromosome positions from http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes
hg19_chrom_sizes <- readRDS(here("extdata/lib/hg19.chrom.sizes.rds"))

# transforming into a named vector and sorting by chrom
hg19_chrom_sizes <- hg19_chrom_sizes[1:23,]
hg19_chrom_sizes <- deframe(hg19_chrom_sizes)
hg19_chrom_sizes <- hg19_chrom_sizes[gtools::mixedsort(names(hg19_chrom_sizes))]

# obtaining the position for every ensembl gene id 
bms <- list()

for(i in 1:length(hg19_chrom_sizes)) {
  chrom <- str_remove(names(hg19_chrom_sizes)[i], "chr")
  
  if (chrom != "X")
    chrom <- as.numeric(chrom)
  
  length_chrom <- hg19_chrom_sizes[i]
  
  bms[[i]] <-
    getBM(
      c(
        "ensembl_gene_id",
        "hgnc_symbol",
        "start_position",
        "end_position"
      ),
      filters = c("chromosome_name", "start", "end"),
      values = list(chrom,
                    0, length_chrom),
      mart = grch37
    )
  
  bms[[i]] <- bms[[i]] %>% arrange(start_position)
  
  names(bms)[[i]] <- names(hg19_chrom_sizes)[i]
}

bms_df <- bind_rows(bms, .id = "chr") %>%
  dplyr::rename(gene = "hgnc_symbol",
                gene_id = "ensembl_gene_id")

# obtaining the count matrix of the MDAMB231 bulk RNAseq
cnt <- assay(vsd)

# gene id clean up
rownames(cnt) <- str_extract(rownames(cnt), "[A-Z]+[0-9]+")

# averaging the expression of the RNA triplicates and calculating a zscore
cnt_long <- cnt %>% 
  as.data.frame() %>% 
  tibble::rownames_to_column("gene_id") %>% 
  tidyr::gather(key = "sample",
                value = "count",
                -gene_id) %>% 
  mutate(sample = str_remove(sample, "[A-C]$"))

cnt_long_avg <- cnt_long %>%
  group_by(gene_id, sample) %>% 
  summarise(count_avg = mean(count)) %>% 
  ungroup()  %>% 
  group_by(gene_id) %>%
  mutate(z_score = (count_avg-mean(count_avg))/sd(count_avg)) %>%
  ungroup() 

# wide df will be used later for moving average
cnt_avg <- tidyr::pivot_wider(cnt_long_avg,
                              names_from = sample,
                              values_from = z_score,
                              id_cols = gene_id)


cnt_avg <- as.data.frame(cnt_avg)
rownames(cnt_avg) <- cnt_avg$gene_id
cnt_avg <- cnt_avg %>%
  dplyr::select(-gene_id)

```

## Moving averages
```{r moving_averages}
# Methods:
# DNA copy number profiles from the expanded clusters are shown 
# by taking the mode of the ith segment from their profiles 
# according to the co-clustering identities.
```

## RNA PCA
```{r rna_pca}
# pcaData <- plotPCA(vst(dds), "cluster", returnData = T) %>% 
#   mutate(triplicate = str_extract(name, "C[0-9]+"),
#          triplicate = tolower(triplicate),
#          triplicate = paste0("e", triplicate)) %>% 
#   mutate(sample = str_extract(name, "MDAMB231C[0-9]+"))
# percentVar <- round(100 * attr(pcaData, "percentVar"))
# 
# cluster_anno <- sc_bulk_231_leiden_umap$mc_classification %>% 
#   filter(str_detect(cells, "gDNA")) %>% 
#   mutate(sample = str_remove(cells, "gDNA")) %>% 
#   mutate(sample = str_extract(sample, "MDAMB231C[0-9]+"))
# 
# pcaData <- pcaData %>% 
#   left_join(cluster_anno, by = c("group" = "hdb", "sample" = "sample"))
# 
# p_pca <- ggplot(pcaData, aes(x = PC1,
#                              y = PC2)) +
#   scale_color_manual(values = c(sc_bulk_231_umap_colors$subclones,sc_bulk_231_umap_colors$superclones)) +
#   geom_point(aes(color = fct_relevel(g_cl, gtools::mixedsort(unique(as.character(pcaData$g_cl))))), size = 8) +
#   geom_point(aes(color = fct_relevel(group, gtools::mixedsort(unique(as.character(pcaData$group))))), size = 3) + 
#   # xlab(paste0("PC1: ", percentVar[1], "% variance")) +
#   # ylab(paste0("PC2: ", percentVar[2], "% variance")) +
#   xlab("PC1") +
#   ylab("PC2") +
#   # coord_fixed() +
#   theme_cowplot() + 
#   theme(axis.title.x=element_text(size = 16),
#         axis.text.x= element_blank(),
#         axis.ticks.x=element_blank(),
#         axis.title.y = element_text(size = 16),
#         axis.text.y = element_blank(),
#         axis.ticks.y = element_blank(),
#         axis.line = element_blank(),
#         plot.title = element_text(hjust = 0.5),
#         legend.position = "none",
#         panel.border = element_rect(color = "black", 
#                                     fill = NA,
#                                     size = 2)) +
#   labs(color = "subclone")
```


