<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Exome and WGD | ACT Paper</title>
  <meta name="description" content="6 Exome and WGD | ACT Paper" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Exome and WGD | ACT Paper" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Exome and WGD | ACT Paper" />
  
  
  

<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="act-10x-cna.html"/>
<link rel="next" href="cell-lines.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="instructions.html"><a href="instructions.html"><i class="fa fa-check"></i><b>1</b> Instructions</a></li>
<li class="chapter" data-level="2" data-path="technical-properties.html"><a href="technical-properties.html"><i class="fa fa-check"></i><b>2</b> Technical properties</a><ul>
<li class="chapter" data-level="2.1" data-path="technical-properties.html"><a href="technical-properties.html#breadth-of-coverage"><i class="fa fa-check"></i><b>2.1</b> Breadth of coverage</a></li>
<li class="chapter" data-level="2.2" data-path="technical-properties.html"><a href="technical-properties.html#overdispersion"><i class="fa fa-check"></i><b>2.2</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="tumors.html"><a href="tumors.html"><i class="fa fa-check"></i><b>3</b> Tumors</a><ul>
<li class="chapter" data-level="3.1" data-path="tumors.html"><a href="tumors.html#tn1"><i class="fa fa-check"></i><b>3.1</b> TN1</a></li>
<li class="chapter" data-level="3.2" data-path="tumors.html"><a href="tumors.html#tn2"><i class="fa fa-check"></i><b>3.2</b> TN2</a></li>
<li class="chapter" data-level="3.3" data-path="tumors.html"><a href="tumors.html#tn3"><i class="fa fa-check"></i><b>3.3</b> TN3</a></li>
<li class="chapter" data-level="3.4" data-path="tumors.html"><a href="tumors.html#tn4"><i class="fa fa-check"></i><b>3.4</b> TN4</a></li>
<li class="chapter" data-level="3.5" data-path="tumors.html"><a href="tumors.html#tn5"><i class="fa fa-check"></i><b>3.5</b> TN5</a></li>
<li class="chapter" data-level="3.6" data-path="tumors.html"><a href="tumors.html#tn6"><i class="fa fa-check"></i><b>3.6</b> TN6</a></li>
<li class="chapter" data-level="3.7" data-path="tumors.html"><a href="tumors.html#tn7"><i class="fa fa-check"></i><b>3.7</b> TN7</a></li>
<li class="chapter" data-level="3.8" data-path="tumors.html"><a href="tumors.html#tn8"><i class="fa fa-check"></i><b>3.8</b> TN8</a></li>
<li class="chapter" data-level="3.9" data-path="tumors.html"><a href="tumors.html#clones-barplot"><i class="fa fa-check"></i><b>3.9</b> Clones barplot</a></li>
<li class="chapter" data-level="3.10" data-path="tumors.html"><a href="tumors.html#shannon-diversity"><i class="fa fa-check"></i><b>3.10</b> Shannon diversity</a></li>
<li class="chapter" data-level="3.11" data-path="tumors.html"><a href="tumors.html#gainslosses-plot"><i class="fa fa-check"></i><b>3.11</b> Gains/Losses plot</a></li>
<li class="chapter" data-level="3.12" data-path="tumors.html"><a href="tumors.html#genomic-classes-plots"><i class="fa fa-check"></i><b>3.12</b> Genomic classes plots</a><ul>
<li class="chapter" data-level="3.12.1" data-path="tumors.html"><a href="tumors.html#counts"><i class="fa fa-check"></i><b>3.12.1</b> Counts</a></li>
<li class="chapter" data-level="3.12.2" data-path="tumors.html"><a href="tumors.html#segment-length"><i class="fa fa-check"></i><b>3.12.2</b> Segment length</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="single-cell-trees.html"><a href="single-cell-trees.html"><i class="fa fa-check"></i><b>4</b> Single-Cell Trees</a></li>
<li class="chapter" data-level="5" data-path="act-10x-cna.html"><a href="act-10x-cna.html"><i class="fa fa-check"></i><b>5</b> ACT &amp; 10X CNA</a><ul>
<li class="chapter" data-level="5.1" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn1-co-clustering-umaps"><i class="fa fa-check"></i><b>5.1</b> TN1 Co-clustering umaps</a></li>
<li class="chapter" data-level="5.2" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn1-gene-frequency-barplot"><i class="fa fa-check"></i><b>5.2</b> TN1 Gene Frequency Barplot</a></li>
<li class="chapter" data-level="5.3" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn3-co-clustering-umaps"><i class="fa fa-check"></i><b>5.3</b> TN3 Co-clustering umaps</a></li>
<li class="chapter" data-level="5.4" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn3-gene-frequency-barplot"><i class="fa fa-check"></i><b>5.4</b> TN3 Gene Frequency Barplot</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html"><i class="fa fa-check"></i><b>6</b> Exome and WGD</a><ul>
<li class="chapter" data-level="6.1" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#data"><i class="fa fa-check"></i><b>6.1</b> Data</a></li>
<li class="chapter" data-level="6.2" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#filtering"><i class="fa fa-check"></i><b>6.2</b> Filtering</a></li>
<li class="chapter" data-level="6.3" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#mut-timer-information"><i class="fa fa-check"></i><b>6.3</b> Mut timer information</a></li>
<li class="chapter" data-level="6.4" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#mutation-burden"><i class="fa fa-check"></i><b>6.4</b> Mutation burden</a></li>
<li class="chapter" data-level="6.5" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#oncomap"><i class="fa fa-check"></i><b>6.5</b> Oncomap</a></li>
<li class="chapter" data-level="6.6" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#wgd-estimation-in-mutational-timing"><i class="fa fa-check"></i><b>6.6</b> WGD estimation in mutational timing</a></li>
<li class="chapter" data-level="6.7" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#wgd-estimations-in-copy-number-time"><i class="fa fa-check"></i><b>6.7</b> WGD estimations in copy number time</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cell-lines.html"><a href="cell-lines.html"><i class="fa fa-check"></i><b>7</b> Cell lines</a><ul>
<li class="chapter" data-level="7.1" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231"><i class="fa fa-check"></i><b>7.1</b> MDA-MB-231</a></li>
<li class="chapter" data-level="7.2" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231-ex1"><i class="fa fa-check"></i><b>7.2</b> MDA-MB-231-EX1</a></li>
<li class="chapter" data-level="7.3" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231-ex2"><i class="fa fa-check"></i><b>7.3</b> MDA-MB-231-EX2</a></li>
<li class="chapter" data-level="7.4" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-453"><i class="fa fa-check"></i><b>7.4</b> MDA-MB-453</a></li>
<li class="chapter" data-level="7.5" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-157"><i class="fa fa-check"></i><b>7.5</b> MDA-MB-157</a></li>
<li class="chapter" data-level="7.6" data-path="cell-lines.html"><a href="cell-lines.html#bt20"><i class="fa fa-check"></i><b>7.6</b> BT20</a></li>
<li class="chapter" data-level="7.7" data-path="cell-lines.html"><a href="cell-lines.html#clones-barplot-1"><i class="fa fa-check"></i><b>7.7</b> Clones barplot</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html"><i class="fa fa-check"></i><b>8</b> MDA-231 DNA &amp; RNA</a><ul>
<li class="chapter" data-level="8.1" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#co-clustering"><i class="fa fa-check"></i><b>8.1</b> Co-clustering</a></li>
<li class="chapter" data-level="8.2" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#single-cell-bulk-heatmap"><i class="fa fa-check"></i><b>8.2</b> Single-cell + Bulk Heatmap</a></li>
<li class="chapter" data-level="8.3" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#bulk-heatmap"><i class="fa fa-check"></i><b>8.3</b> Bulk Heatmap</a></li>
<li class="chapter" data-level="8.4" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#dna-rna"><i class="fa fa-check"></i><b>8.4</b> DNA &amp; RNA</a></li>
<li class="chapter" data-level="8.5" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#moving-averages"><i class="fa fa-check"></i><b>8.5</b> Moving averages</a></li>
<li class="chapter" data-level="8.6" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#de"><i class="fa fa-check"></i><b>8.6</b> DE</a></li>
<li class="chapter" data-level="8.7" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#rna-pca"><i class="fa fa-check"></i><b>8.7</b> RNA PCA</a></li>
<li class="chapter" data-level="8.8" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#fgsea"><i class="fa fa-check"></i><b>8.8</b> FGSEA</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>9</b> Session Info</a><ul>
<li class="chapter" data-level="9.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>9.1</b> Dockerfile</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="instructions-1.html"><a href="instructions-1.html"><i class="fa fa-check"></i><b>10</b> Instructions</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ACT Paper</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exome-and-wgd" class="section level1">
<h1><span class="header-section-number">6</span> Exome and WGD</h1>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb194-1" data-line-number="1"><span class="kw">source</span>(<span class="st">&quot;R/setup.R&quot;</span>)</a>
<a class="sourceLine" id="cb194-2" data-line-number="2"><span class="kw">source</span>(<span class="st">&quot;R/vcf_filtering.R&quot;</span>)</a>
<a class="sourceLine" id="cb194-3" data-line-number="3"><span class="kw">source</span>(<span class="st">&quot;R/parse_mtimer_vcf.R&quot;</span>)</a></code></pre></div>
<div id="data" class="section level2">
<h2><span class="header-section-number">6.1</span> Data</h2>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb195-1" data-line-number="1">TN1_mutect &lt;-<span class="st"> </span><span class="kw">read.vcfR</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mutect/TN1.vcf&quot;</span>))</a></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 165
##   header_line: 166
##   variant count: 141
##   column count: 15
## 
Meta line 165 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 141
##   Character matrix gt cols: 15
##   skip: 0
##   nrows: 141
##   row_num: 0
## 
Processed variant: 141
## All variants processed</code></pre>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb197-1" data-line-number="1">TN1_annovar &lt;-<span class="st"> </span><span class="kw">read_delim</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/annovar/TN1.annovar.csv&quot;</span>),</a>
<a class="sourceLine" id="cb197-2" data-line-number="2">                           <span class="dt">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb197-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">CHROM =</span> <span class="st">&quot;Chr&quot;</span>,</a>
<a class="sourceLine" id="cb197-4" data-line-number="4">         <span class="dt">POS =</span> <span class="st">&quot;Start&quot;</span>)</a></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_character(),
##   Start = col_double(),
##   End = col_double()
## )</code></pre>
<pre><code>## See spec(...) for full column specifications.</code></pre>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb200-1" data-line-number="1">TN2_mutect &lt;-<span class="st"> </span><span class="kw">read.vcfR</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mutect/TN2.vcf&quot;</span>))</a></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 165
##   header_line: 166
##   variant count: 96
##   column count: 15
## 
Meta line 165 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 96
##   Character matrix gt cols: 15
##   skip: 0
##   nrows: 96
##   row_num: 0
## 
Processed variant: 96
## All variants processed</code></pre>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb202-1" data-line-number="1">TN2_annovar &lt;-<span class="st"> </span><span class="kw">read_delim</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/annovar/TN2.annovar.csv&quot;</span>),</a>
<a class="sourceLine" id="cb202-2" data-line-number="2">                           <span class="dt">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb202-3" data-line-number="3"><span class="st">  </span><span class="kw">rename</span>(<span class="dt">CHROM =</span> <span class="st">&quot;Chr&quot;</span>,</a>
<a class="sourceLine" id="cb202-4" data-line-number="4">         <span class="dt">POS =</span> <span class="st">&quot;Start&quot;</span>)</a></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_character(),
##   Start = col_double(),
##   End = col_double()
## )
## See spec(...) for full column specifications.</code></pre>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb204-1" data-line-number="1">TN3_mutect &lt;-<span class="st"> </span><span class="kw">read.vcfR</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mutect/TN3.vcf&quot;</span>))</a></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 165
##   header_line: 166
##   variant count: 1057
##   column count: 15
## 
Meta line 165 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 1057
##   Character matrix gt cols: 15
##   skip: 0
##   nrows: 1057
##   row_num: 0
## 
Processed variant 1000
Processed variant: 1057
## All variants processed</code></pre>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb206-1" data-line-number="1">TN3_annovar &lt;-<span class="st"> </span><span class="kw">read_delim</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/annovar/TN3.annovar.csv&quot;</span>),</a>
<a class="sourceLine" id="cb206-2" data-line-number="2">                           <span class="dt">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb206-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">CHROM =</span> <span class="st">&quot;Chr&quot;</span>,</a>
<a class="sourceLine" id="cb206-4" data-line-number="4">         <span class="dt">POS =</span> <span class="st">&quot;Start&quot;</span>)</a></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_character(),
##   Start = col_double(),
##   End = col_double()
## )
## See spec(...) for full column specifications.</code></pre>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb208-1" data-line-number="1">TN4_mutect &lt;-<span class="st"> </span><span class="kw">read.vcfR</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mutect/TN4.vcf&quot;</span>))</a></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 165
##   header_line: 166
##   variant count: 97
##   column count: 15
## 
Meta line 165 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 97
##   Character matrix gt cols: 15
##   skip: 0
##   nrows: 97
##   row_num: 0
## 
Processed variant: 97
## All variants processed</code></pre>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb210-1" data-line-number="1">TN4_annovar &lt;-<span class="st"> </span><span class="kw">read_delim</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/annovar/TN4.annovar.csv&quot;</span>),</a>
<a class="sourceLine" id="cb210-2" data-line-number="2">                           <span class="dt">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb210-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">CHROM =</span> <span class="st">&quot;Chr&quot;</span>,</a>
<a class="sourceLine" id="cb210-4" data-line-number="4">         <span class="dt">POS =</span> <span class="st">&quot;Start&quot;</span>)</a></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_character(),
##   Start = col_double(),
##   End = col_double()
## )
## See spec(...) for full column specifications.</code></pre>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb212-1" data-line-number="1">TN5_mutect &lt;-<span class="st"> </span><span class="kw">read.vcfR</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mutect/TN5.vcf&quot;</span>))</a></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 166
##   header_line: 167
##   variant count: 300
##   column count: 16
## 
Meta line 166 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 300
##   Character matrix gt cols: 16
##   skip: 0
##   nrows: 300
##   row_num: 0
## 
Processed variant: 300
## All variants processed</code></pre>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb214-1" data-line-number="1">TN5_annovar &lt;-<span class="st"> </span><span class="kw">read_delim</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/annovar/TN5.annovar.csv&quot;</span>),</a>
<a class="sourceLine" id="cb214-2" data-line-number="2">                           <span class="dt">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb214-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">CHROM =</span> <span class="st">&quot;Chr&quot;</span>,</a>
<a class="sourceLine" id="cb214-4" data-line-number="4">         <span class="dt">POS =</span> <span class="st">&quot;Start&quot;</span>)</a></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_character(),
##   Start = col_double(),
##   End = col_double()
## )
## See spec(...) for full column specifications.</code></pre>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb216-1" data-line-number="1">TN6_mutect &lt;-<span class="st"> </span><span class="kw">read.vcfR</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mutect/TN6.vcf&quot;</span>))</a></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 166
##   header_line: 167
##   variant count: 291
##   column count: 16
## 
Meta line 166 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 291
##   Character matrix gt cols: 16
##   skip: 0
##   nrows: 291
##   row_num: 0
## 
Processed variant: 291
## All variants processed</code></pre>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb218-1" data-line-number="1">TN6_annovar &lt;-<span class="st"> </span><span class="kw">read_delim</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/annovar/TN6.annovar.csv&quot;</span>),</a>
<a class="sourceLine" id="cb218-2" data-line-number="2">                           <span class="dt">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb218-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">CHROM =</span> <span class="st">&quot;Chr&quot;</span>,</a>
<a class="sourceLine" id="cb218-4" data-line-number="4">         <span class="dt">POS =</span> <span class="st">&quot;Start&quot;</span>)</a></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_character(),
##   Start = col_double(),
##   End = col_double()
## )
## See spec(...) for full column specifications.</code></pre>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb220-1" data-line-number="1">TN7_mutect &lt;-<span class="st"> </span><span class="kw">read.vcfR</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mutect/TN7.vcf&quot;</span>))</a></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 164
##   header_line: 165
##   variant count: 406
##   column count: 14
## 
Meta line 164 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 406
##   Character matrix gt cols: 14
##   skip: 0
##   nrows: 406
##   row_num: 0
## 
Processed variant: 406
## All variants processed</code></pre>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb222-1" data-line-number="1">TN7_annovar &lt;-<span class="st"> </span><span class="kw">read_delim</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/annovar/TN7.annovar.csv&quot;</span>),</a>
<a class="sourceLine" id="cb222-2" data-line-number="2">                           <span class="dt">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb222-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">CHROM =</span> <span class="st">&quot;Chr&quot;</span>,</a>
<a class="sourceLine" id="cb222-4" data-line-number="4">         <span class="dt">POS =</span> <span class="st">&quot;Start&quot;</span>)</a></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_character(),
##   Start = col_double(),
##   End = col_double()
## )
## See spec(...) for full column specifications.</code></pre>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb224-1" data-line-number="1">TN8_mutect &lt;-<span class="st"> </span><span class="kw">read.vcfR</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mutect/TN8.vcf&quot;</span>))</a></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 165
##   header_line: 166
##   variant count: 6592
##   column count: 15
## 
Meta line 165 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 6592
##   Character matrix gt cols: 15
##   skip: 0
##   nrows: 6592
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant: 6592
## All variants processed</code></pre>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb226-1" data-line-number="1">TN8_annovar &lt;-<span class="st"> </span><span class="kw">read_delim</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/annovar/TN8.annovar.csv&quot;</span>),</a>
<a class="sourceLine" id="cb226-2" data-line-number="2">                           <span class="dt">delim =</span> <span class="st">&quot;,&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb226-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">CHROM =</span> <span class="st">&quot;Chr&quot;</span>,</a>
<a class="sourceLine" id="cb226-4" data-line-number="4">         <span class="dt">POS =</span> <span class="st">&quot;Start&quot;</span>)</a></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_character(),
##   Start = col_double(),
##   End = col_double()
## )
## See spec(...) for full column specifications.</code></pre>
</div>
<div id="filtering" class="section level2">
<h2><span class="header-section-number">6.2</span> Filtering</h2>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb228-1" data-line-number="1">TN1_mutations &lt;-<span class="st"> </span><span class="kw">vcf_filtering</span>(</a>
<a class="sourceLine" id="cb228-2" data-line-number="2">  <span class="dt">mutect =</span> TN1_mutect,</a>
<a class="sourceLine" id="cb228-3" data-line-number="3">  <span class="dt">annovar =</span> TN1_annovar,</a>
<a class="sourceLine" id="cb228-4" data-line-number="4">  <span class="dt">min_AD =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb228-5" data-line-number="5">  <span class="dt">bulk_normal =</span> <span class="st">&quot;TN28N&quot;</span>,</a>
<a class="sourceLine" id="cb228-6" data-line-number="6">  <span class="dt">bulk_tumor =</span> <span class="st">&quot;TN28S1A&quot;</span></a>
<a class="sourceLine" id="cb228-7" data-line-number="7">) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb228-8" data-line-number="8"><span class="st">  </span><span class="kw">left_join</span>(TN1_annovar)</a>
<a class="sourceLine" id="cb228-9" data-line-number="9"></a>
<a class="sourceLine" id="cb228-10" data-line-number="10">TN2_mutations &lt;-<span class="st"> </span><span class="kw">vcf_filtering</span>(</a>
<a class="sourceLine" id="cb228-11" data-line-number="11">  <span class="dt">mutect =</span> TN2_mutect,</a>
<a class="sourceLine" id="cb228-12" data-line-number="12">  <span class="dt">annovar =</span> TN2_annovar,</a>
<a class="sourceLine" id="cb228-13" data-line-number="13">  <span class="dt">min_AD =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb228-14" data-line-number="14">  <span class="dt">bulk_normal =</span> <span class="st">&quot;TN20N&quot;</span>,</a>
<a class="sourceLine" id="cb228-15" data-line-number="15">  <span class="dt">bulk_tumor =</span> <span class="st">&quot;TN20S1A&quot;</span></a>
<a class="sourceLine" id="cb228-16" data-line-number="16">) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb228-17" data-line-number="17"><span class="st">  </span><span class="kw">left_join</span>(TN2_annovar)</a>
<a class="sourceLine" id="cb228-18" data-line-number="18"></a>
<a class="sourceLine" id="cb228-19" data-line-number="19">TN3_mutations &lt;-<span class="st"> </span><span class="kw">vcf_filtering</span>(</a>
<a class="sourceLine" id="cb228-20" data-line-number="20">  <span class="dt">mutect =</span> TN3_mutect,</a>
<a class="sourceLine" id="cb228-21" data-line-number="21">  <span class="dt">annovar =</span> TN3_annovar,</a>
<a class="sourceLine" id="cb228-22" data-line-number="22">  <span class="dt">min_AD =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb228-23" data-line-number="23">  <span class="dt">bulk_normal =</span> <span class="st">&quot;TN17N&quot;</span>,</a>
<a class="sourceLine" id="cb228-24" data-line-number="24">  <span class="dt">bulk_tumor =</span> <span class="st">&quot;TN17A&quot;</span></a>
<a class="sourceLine" id="cb228-25" data-line-number="25">) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb228-26" data-line-number="26"><span class="st">  </span><span class="kw">left_join</span>(TN3_annovar)</a>
<a class="sourceLine" id="cb228-27" data-line-number="27"></a>
<a class="sourceLine" id="cb228-28" data-line-number="28">TN4_mutations &lt;-<span class="st"> </span><span class="kw">vcf_filtering</span>(</a>
<a class="sourceLine" id="cb228-29" data-line-number="29">  <span class="dt">mutect =</span> TN4_mutect,</a>
<a class="sourceLine" id="cb228-30" data-line-number="30">  <span class="dt">annovar =</span> TN4_annovar,</a>
<a class="sourceLine" id="cb228-31" data-line-number="31">  <span class="dt">min_AD =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb228-32" data-line-number="32">  <span class="dt">bulk_normal =</span> <span class="st">&quot;TN26N&quot;</span>,</a>
<a class="sourceLine" id="cb228-33" data-line-number="33">  <span class="dt">bulk_tumor =</span> <span class="st">&quot;TN26S2A&quot;</span></a>
<a class="sourceLine" id="cb228-34" data-line-number="34">) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb228-35" data-line-number="35"><span class="st">  </span><span class="kw">left_join</span>(TN4_annovar)</a>
<a class="sourceLine" id="cb228-36" data-line-number="36"></a>
<a class="sourceLine" id="cb228-37" data-line-number="37">TN5_mutations &lt;-<span class="st"> </span><span class="kw">vcf_filtering</span>(</a>
<a class="sourceLine" id="cb228-38" data-line-number="38">  <span class="dt">mutect =</span> TN5_mutect,</a>
<a class="sourceLine" id="cb228-39" data-line-number="39">  <span class="dt">annovar =</span> TN5_annovar,</a>
<a class="sourceLine" id="cb228-40" data-line-number="40">  <span class="dt">min_AD =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb228-41" data-line-number="41">  <span class="dt">bulk_normal =</span> <span class="st">&quot;TN21N&quot;</span>,</a>
<a class="sourceLine" id="cb228-42" data-line-number="42">  <span class="dt">bulk_tumor =</span> <span class="st">&quot;TN21S1A&quot;</span></a>
<a class="sourceLine" id="cb228-43" data-line-number="43">) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb228-44" data-line-number="44"><span class="st">  </span><span class="kw">left_join</span>(TN5_annovar)</a>
<a class="sourceLine" id="cb228-45" data-line-number="45"></a>
<a class="sourceLine" id="cb228-46" data-line-number="46">TN6_mutations &lt;-<span class="st"> </span><span class="kw">vcf_filtering</span>(</a>
<a class="sourceLine" id="cb228-47" data-line-number="47">  <span class="dt">mutect =</span> TN6_mutect,</a>
<a class="sourceLine" id="cb228-48" data-line-number="48">  <span class="dt">annovar =</span> TN6_annovar,</a>
<a class="sourceLine" id="cb228-49" data-line-number="49">  <span class="dt">min_AD =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb228-50" data-line-number="50">  <span class="dt">bulk_normal =</span> <span class="st">&quot;BRCADN&quot;</span>,</a>
<a class="sourceLine" id="cb228-51" data-line-number="51">  <span class="dt">bulk_tumor =</span> <span class="st">&quot;BRCADT&quot;</span></a>
<a class="sourceLine" id="cb228-52" data-line-number="52">) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb228-53" data-line-number="53"><span class="st">  </span><span class="kw">left_join</span>(TN6_annovar)</a>
<a class="sourceLine" id="cb228-54" data-line-number="54"></a>
<a class="sourceLine" id="cb228-55" data-line-number="55">TN7_mutations &lt;-<span class="st"> </span><span class="kw">vcf_filtering</span>(</a>
<a class="sourceLine" id="cb228-56" data-line-number="56">  <span class="dt">mutect =</span> TN7_mutect,</a>
<a class="sourceLine" id="cb228-57" data-line-number="57">  <span class="dt">annovar =</span> TN7_annovar,</a>
<a class="sourceLine" id="cb228-58" data-line-number="58">  <span class="dt">min_AD =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb228-59" data-line-number="59">  <span class="dt">bulk_normal =</span> <span class="st">&quot;BRCACN&quot;</span>,</a>
<a class="sourceLine" id="cb228-60" data-line-number="60">  <span class="dt">bulk_tumor =</span> <span class="st">&quot;BRCACT&quot;</span></a>
<a class="sourceLine" id="cb228-61" data-line-number="61">) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb228-62" data-line-number="62"><span class="st">  </span><span class="kw">left_join</span>(TN7_annovar)</a>
<a class="sourceLine" id="cb228-63" data-line-number="63"></a>
<a class="sourceLine" id="cb228-64" data-line-number="64">TN8_mutations &lt;-<span class="st"> </span><span class="kw">vcf_filtering</span>(</a>
<a class="sourceLine" id="cb228-65" data-line-number="65">  <span class="dt">mutect =</span> TN8_mutect,</a>
<a class="sourceLine" id="cb228-66" data-line-number="66">  <span class="dt">annovar =</span> TN8_annovar,</a>
<a class="sourceLine" id="cb228-67" data-line-number="67">  <span class="dt">min_AD =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb228-68" data-line-number="68">  <span class="dt">bulk_normal =</span> <span class="st">&quot;TN27N&quot;</span>,</a>
<a class="sourceLine" id="cb228-69" data-line-number="69">  <span class="dt">bulk_tumor =</span> <span class="st">&quot;TN27A&quot;</span></a>
<a class="sourceLine" id="cb228-70" data-line-number="70">) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb228-71" data-line-number="71"><span class="st">  </span><span class="kw">left_join</span>(TN8_annovar)</a>
<a class="sourceLine" id="cb228-72" data-line-number="72"></a>
<a class="sourceLine" id="cb228-73" data-line-number="73"><span class="co"># Unique data frame for mutations, excluding matched normal from the vcf</span></a>
<a class="sourceLine" id="cb228-74" data-line-number="74">TN_muts &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(</a>
<a class="sourceLine" id="cb228-75" data-line-number="75">  TN1_mutations <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(sample <span class="op">!=</span><span class="st"> &quot;TN28N&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN1&quot;</span>),</a>
<a class="sourceLine" id="cb228-76" data-line-number="76">  TN2_mutations <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(sample <span class="op">!=</span><span class="st"> &quot;TN20N&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN2&quot;</span>),</a>
<a class="sourceLine" id="cb228-77" data-line-number="77">  TN3_mutations <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(sample <span class="op">!=</span><span class="st"> &quot;TN17N&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN3&quot;</span>),</a>
<a class="sourceLine" id="cb228-78" data-line-number="78">  TN4_mutations <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(sample <span class="op">!=</span><span class="st"> &quot;TN26N&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN4&quot;</span>),</a>
<a class="sourceLine" id="cb228-79" data-line-number="79">  TN5_mutations <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(sample <span class="op">!=</span><span class="st"> &quot;TN21N&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN5&quot;</span>),</a>
<a class="sourceLine" id="cb228-80" data-line-number="80">  TN6_mutations <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(sample <span class="op">!=</span><span class="st"> &quot;BRCADN&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN6&quot;</span>),</a>
<a class="sourceLine" id="cb228-81" data-line-number="81">  TN7_mutations <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(sample <span class="op">!=</span><span class="st"> &quot;BRCACN&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN7&quot;</span>),</a>
<a class="sourceLine" id="cb228-82" data-line-number="82">  TN8_mutations <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(sample <span class="op">!=</span><span class="st"> &quot;TN27N&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN8&quot;</span>) </a>
<a class="sourceLine" id="cb228-83" data-line-number="83">)</a></code></pre></div>
</div>
<div id="mut-timer-information" class="section level2">
<h2><span class="header-section-number">6.3</span> Mut timer information</h2>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb229-1" data-line-number="1">TN1_mut_time &lt;-<span class="st"> </span><span class="kw">parse_mtimer_vcf</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mut_timer/TN1_m_timer.vcf&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN1&quot;</span>)</a>
<a class="sourceLine" id="cb229-2" data-line-number="2"></a>
<a class="sourceLine" id="cb229-3" data-line-number="3">TN2_mut_time &lt;-<span class="st"> </span><span class="kw">parse_mtimer_vcf</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mut_timer/TN2_m_timer.vcf&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN2&quot;</span>)</a>
<a class="sourceLine" id="cb229-4" data-line-number="4"></a>
<a class="sourceLine" id="cb229-5" data-line-number="5">TN3_mut_time &lt;-<span class="st"> </span><span class="kw">parse_mtimer_vcf</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mut_timer/TN3_m_timer.vcf&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN3&quot;</span>)</a>
<a class="sourceLine" id="cb229-6" data-line-number="6"></a>
<a class="sourceLine" id="cb229-7" data-line-number="7">TN4_mut_time &lt;-<span class="st"> </span><span class="kw">parse_mtimer_vcf</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mut_timer/TN4_m_timer.vcf&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN4&quot;</span>)</a>
<a class="sourceLine" id="cb229-8" data-line-number="8"></a>
<a class="sourceLine" id="cb229-9" data-line-number="9">TN5_mut_time &lt;-<span class="st"> </span><span class="kw">parse_mtimer_vcf</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mut_timer/TN5_m_timer.vcf&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN5&quot;</span>)</a>
<a class="sourceLine" id="cb229-10" data-line-number="10"></a>
<a class="sourceLine" id="cb229-11" data-line-number="11">TN6_mut_time &lt;-<span class="st"> </span><span class="kw">parse_mtimer_vcf</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mut_timer/TN6_m_timer.vcf&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN6&quot;</span>)</a>
<a class="sourceLine" id="cb229-12" data-line-number="12"></a>
<a class="sourceLine" id="cb229-13" data-line-number="13">TN7_mut_time &lt;-<span class="st"> </span><span class="kw">parse_mtimer_vcf</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mut_timer/TN7_m_timer.vcf&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN7&quot;</span>)</a>
<a class="sourceLine" id="cb229-14" data-line-number="14"></a>
<a class="sourceLine" id="cb229-15" data-line-number="15">TN8_mut_time &lt;-<span class="st"> </span><span class="kw">parse_mtimer_vcf</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/mut_timer/TN8_m_timer.vcf&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;TN8&quot;</span>)</a>
<a class="sourceLine" id="cb229-16" data-line-number="16"></a>
<a class="sourceLine" id="cb229-17" data-line-number="17">TN_mut_time &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(</a>
<a class="sourceLine" id="cb229-18" data-line-number="18">  TN1_mut_time,</a>
<a class="sourceLine" id="cb229-19" data-line-number="19">  TN2_mut_time,</a>
<a class="sourceLine" id="cb229-20" data-line-number="20">  TN3_mut_time,</a>
<a class="sourceLine" id="cb229-21" data-line-number="21">  TN4_mut_time,</a>
<a class="sourceLine" id="cb229-22" data-line-number="22">  TN5_mut_time,</a>
<a class="sourceLine" id="cb229-23" data-line-number="23">  TN6_mut_time,</a>
<a class="sourceLine" id="cb229-24" data-line-number="24">  TN7_mut_time,</a>
<a class="sourceLine" id="cb229-25" data-line-number="25">  TN8_mut_time</a>
<a class="sourceLine" id="cb229-26" data-line-number="26">)</a></code></pre></div>
</div>
<div id="mutation-burden" class="section level2">
<h2><span class="header-section-number">6.4</span> Mutation burden</h2>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb230-1" data-line-number="1"><span class="co"># merging with mutation timer</span></a>
<a class="sourceLine" id="cb230-2" data-line-number="2">TN_muts_j_time &lt;-<span class="st"> </span><span class="kw">left_join</span>(TN_muts, TN_mut_time) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb230-3" data-line-number="3"><span class="st">    </span><span class="kw">filter</span>(sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;TN28S1A&quot;</span>, <span class="st">&quot;TN20S1A&quot;</span>, <span class="st">&quot;TN17A&quot;</span>, <span class="st">&quot;TN26S2A&quot;</span>, <span class="st">&quot;TN21S1A&quot;</span>, <span class="st">&quot;BRCADT&quot;</span>, <span class="st">&quot;BRCACT&quot;</span>, <span class="st">&quot;TN27A&quot;</span>))  <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb230-4" data-line-number="4"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">indel =</span> <span class="kw">case_when</span>(<span class="kw">str_length</span>(Alt) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span> <span class="op">~</span><span class="st"> &quot;indel&quot;</span>, <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> &#39;SNV&#39;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb230-5" data-line-number="5"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">mutation_time =</span> <span class="kw">case_when</span>(<span class="kw">is.na</span>(mutation_time) <span class="op">&amp;</span><span class="st"> </span><span class="kw">str_length</span>(Alt) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span> <span class="op">~</span><span class="st"> &#39;indel&#39;</span>,</a>
<a class="sourceLine" id="cb230-6" data-line-number="6">                                   <span class="kw">is.na</span>(mutation_time) <span class="op">&amp;</span><span class="st"> </span><span class="kw">str_length</span>(Alt) <span class="op">==</span><span class="st"> </span><span class="dv">1</span> <span class="op">~</span><span class="st"> &#39;unclassified&#39;</span>,</a>
<a class="sourceLine" id="cb230-7" data-line-number="7">                                   <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> </span>mutation_time))</a></code></pre></div>
<pre><code>## Joining, by = c(&quot;CHROM&quot;, &quot;POS&quot;, &quot;patient&quot;)</code></pre>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb232-1" data-line-number="1">p_mut_burden &lt;-<span class="st"> </span>TN_muts_j_time <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb232-2" data-line-number="2"><span class="st">  </span><span class="kw">group_by</span>(sample) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb232-3" data-line-number="3"><span class="st">  </span><span class="kw">filter</span>(sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;TN28S1A&quot;</span>, <span class="st">&quot;TN20S1A&quot;</span>, <span class="st">&quot;TN17A&quot;</span>, <span class="st">&quot;TN26S2A&quot;</span>, <span class="st">&quot;TN21S1A&quot;</span>, <span class="st">&quot;BRCADT&quot;</span>, <span class="st">&quot;BRCACT&quot;</span>, <span class="st">&quot;TN27A&quot;</span>),</a>
<a class="sourceLine" id="cb232-4" data-line-number="4">         mutation_time <span class="op">!=</span><span class="st"> &quot;indel&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb232-5" data-line-number="5"><span class="st">  </span><span class="kw">add_count</span>(patient, <span class="dt">name =</span> <span class="st">&quot;mutation_total&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb232-6" data-line-number="6"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="kw">fct_reorder</span>(patient, mutation_total), <span class="dt">fill =</span> mutation_time)) <span class="op">+</span></a>
<a class="sourceLine" id="cb232-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">color =</span> <span class="st">&#39;black&#39;</span>, <span class="dt">width =</span> <span class="fl">.7</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb232-8" data-line-number="8"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;clonal&quot;</span> =<span class="st"> &quot;#625a94&quot;</span>,<span class="st">&quot;subclonal&quot;</span> =<span class="st"> &quot;#f57946&quot;</span>,<span class="st">&quot;indel&quot;</span> =<span class="st"> &quot;#ecc0a1&quot;</span>,</a>
<a class="sourceLine" id="cb232-9" data-line-number="9">                               <span class="st">&quot;unclassified&quot;</span> =<span class="st"> &quot;#4a3a3b&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb232-10" data-line-number="10"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> scales<span class="op">::</span><span class="kw">pretty_breaks</span>(<span class="dt">n =</span> <span class="dv">10</span>), </a>
<a class="sourceLine" id="cb232-11" data-line-number="11">                     <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb232-12" data-line-number="12"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>),</a>
<a class="sourceLine" id="cb232-13" data-line-number="13">        <span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb232-14" data-line-number="14"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb232-15" data-line-number="15"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;exome mutations&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb232-16" data-line-number="16"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;&quot;</span>)</a>
<a class="sourceLine" id="cb232-17" data-line-number="17"></a>
<a class="sourceLine" id="cb232-18" data-line-number="18">p_mut_burden_<span class="dv">1</span>&lt;-<span class="st"> </span>p_mut_burden <span class="op">+</span><span class="st"> </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">200</span>))</a>
<a class="sourceLine" id="cb232-19" data-line-number="19">p_mut_burden_<span class="dv">2</span> &lt;-<span class="st"> </span>p_mut_burden <span class="op">+</span><span class="st"> </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">1650</span>, <span class="dv">1750</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb232-20" data-line-number="20"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> scales<span class="op">::</span><span class="kw">pretty_breaks</span>(<span class="dt">n =</span> <span class="dv">5</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb232-21" data-line-number="21"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,</a>
<a class="sourceLine" id="cb232-22" data-line-number="22">        <span class="dt">axis.line.x =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb232-23" data-line-number="23">        <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb232-24" data-line-number="24">        <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>()) </a></code></pre></div>
<pre><code>## Scale for &#39;y&#39; is already present. Adding another scale for &#39;y&#39;, which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb234-1" data-line-number="1">p_mut_burden_break &lt;-<span class="st"> </span>p_mut_burden_<span class="dv">2</span><span class="op">/</span>p_mut_burden_<span class="dv">1</span> <span class="op">+</span><span class="st"> </span><span class="kw">plot_layout</span>(<span class="dt">heights =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</a>
<a class="sourceLine" id="cb234-2" data-line-number="2">p_mut_burden_break</a></code></pre></div>
<p><img src="_main_files/figure-html/burden_plot-1.png" width="672" /></p>
</div>
<div id="oncomap" class="section level2">
<h2><span class="header-section-number">6.5</span> Oncomap</h2>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb235-1" data-line-number="1">genes_upset &lt;-<span class="st"> </span>TN_muts <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb235-2" data-line-number="2"><span class="st">  </span><span class="kw">filter</span>(sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;TN28S1A&quot;</span>, <span class="st">&quot;TN20S1A&quot;</span>, <span class="st">&quot;TN17A&quot;</span>, <span class="st">&quot;TN26S2A&quot;</span>, <span class="st">&quot;TN21S1A&quot;</span>, <span class="st">&quot;BRCADT&quot;</span>, <span class="st">&quot;BRCACT&quot;</span>, <span class="st">&quot;TN27A&quot;</span>),</a>
<a class="sourceLine" id="cb235-3" data-line-number="3">         ExonicFunc.refGene <span class="op">!=</span><span class="st"> &quot;synonymous SNV&quot;</span>,</a>
<a class="sourceLine" id="cb235-4" data-line-number="4">         Polyphen2_HDIV_score <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.85</span>,</a>
<a class="sourceLine" id="cb235-5" data-line-number="5">         SIFT_score <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb235-6" data-line-number="6"><span class="st">  </span><span class="kw">group_by</span>(patient) <span class="op">%&gt;%</span><span class="st">  </span></a>
<a class="sourceLine" id="cb235-7" data-line-number="7"><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb235-8" data-line-number="8"><span class="st">  </span><span class="kw">count</span>(Gene.refGene, <span class="dt">sort =</span> T) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb235-9" data-line-number="9"><span class="st">  </span><span class="kw">filter</span>(n <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb235-10" data-line-number="10"><span class="st">  </span><span class="kw">pull</span>(Gene.refGene)</a>
<a class="sourceLine" id="cb235-11" data-line-number="11">  </a>
<a class="sourceLine" id="cb235-12" data-line-number="12">oncomap &lt;-<span class="st"> </span>TN_muts <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb235-13" data-line-number="13"><span class="st">  </span><span class="kw">filter</span>(sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;TN28S1A&quot;</span>, <span class="st">&quot;TN20S1A&quot;</span>, <span class="st">&quot;TN17A&quot;</span>, <span class="st">&quot;TN26S2A&quot;</span>, <span class="st">&quot;TN21S1A&quot;</span>, <span class="st">&quot;BRCADT&quot;</span>, <span class="st">&quot;BRCACT&quot;</span>, <span class="st">&quot;TN27A&quot;</span>),</a>
<a class="sourceLine" id="cb235-14" data-line-number="14">         ExonicFunc.refGene <span class="op">!=</span><span class="st"> &quot;synonymous SNV&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb235-15" data-line-number="15"><span class="st">  </span><span class="kw">filter</span>(Gene.refGene <span class="op">%in%</span><span class="st"> </span>genes_upset) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb235-16" data-line-number="16"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(Gene.refGene,</a>
<a class="sourceLine" id="cb235-17" data-line-number="17">             patient)) <span class="op">+</span></a>
<a class="sourceLine" id="cb235-18" data-line-number="18"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> ExonicFunc.refGene),</a>
<a class="sourceLine" id="cb235-19" data-line-number="19">             <span class="dt">size =</span> <span class="dv">5</span>,</a>
<a class="sourceLine" id="cb235-20" data-line-number="20">             <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb235-21" data-line-number="21"><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">limits =</span> genes_upset) <span class="op">+</span></a>
<a class="sourceLine" id="cb235-22" data-line-number="22"><span class="st">  </span><span class="kw">scale_y_discrete</span>(<span class="dt">limits =</span> <span class="kw">rev</span>(<span class="kw">paste0</span>(<span class="st">&quot;TN&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">8</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb235-23" data-line-number="23"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> PNWColors<span class="op">::</span><span class="kw">pnw_palette</span>(<span class="st">&quot;Sunset&quot;</span>, <span class="dv">3</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb235-24" data-line-number="24"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>,</a>
<a class="sourceLine" id="cb235-25" data-line-number="25">                                   <span class="dt">vjust =</span> <span class="fl">.5</span>,</a>
<a class="sourceLine" id="cb235-26" data-line-number="26">                                   <span class="dt">hjust =</span> <span class="dv">1</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb235-27" data-line-number="27"><span class="st">    </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;gene&quot;</span>,</a>
<a class="sourceLine" id="cb235-28" data-line-number="28">         <span class="dt">y =</span> <span class="st">&quot;&quot;</span>,</a>
<a class="sourceLine" id="cb235-29" data-line-number="29">         <span class="dt">fill =</span> <span class="st">&quot;&quot;</span>)</a>
<a class="sourceLine" id="cb235-30" data-line-number="30"></a>
<a class="sourceLine" id="cb235-31" data-line-number="31">oncomap</a></code></pre></div>
<p><img src="_main_files/figure-html/oncomap_plot-1.png" width="672" /></p>
</div>
<div id="wgd-estimation-in-mutational-timing" class="section level2">
<h2><span class="header-section-number">6.6</span> WGD estimation in mutational timing</h2>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb236-1" data-line-number="1"><span class="co"># TN5 can&#39;t be confidently estimated and for this reason is excluded</span></a>
<a class="sourceLine" id="cb236-2" data-line-number="2">wgd_time &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/timing/wgd_timing_distributions.tsv&quot;</span>), </a>
<a class="sourceLine" id="cb236-3" data-line-number="3">                       <span class="dt">header =</span> T) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb236-4" data-line-number="4"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Sample_ID =</span> <span class="kw">case_when</span>(</a>
<a class="sourceLine" id="cb236-5" data-line-number="5">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;TN28&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN1&quot;</span>,</a>
<a class="sourceLine" id="cb236-6" data-line-number="6">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;TN20&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN2&quot;</span>,</a>
<a class="sourceLine" id="cb236-7" data-line-number="7">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;TN17&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN3&quot;</span>,</a>
<a class="sourceLine" id="cb236-8" data-line-number="8">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;TN26&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN4&quot;</span>,</a>
<a class="sourceLine" id="cb236-9" data-line-number="9">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;TN21&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN5&quot;</span>,</a>
<a class="sourceLine" id="cb236-10" data-line-number="10">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;TN27&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN8&quot;</span>,</a>
<a class="sourceLine" id="cb236-11" data-line-number="11">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;pD&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN6&quot;</span>,</a>
<a class="sourceLine" id="cb236-12" data-line-number="12">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;pC&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN7&quot;</span></a>
<a class="sourceLine" id="cb236-13" data-line-number="13">  )) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb236-14" data-line-number="14"><span class="st">  </span><span class="kw">filter</span>(Sample_ID <span class="op">!=</span><span class="st"> &quot;TN5&quot;</span>)</a>
<a class="sourceLine" id="cb236-15" data-line-number="15"></a>
<a class="sourceLine" id="cb236-16" data-line-number="16">wgd_time_tidy &lt;-<span class="st"> </span>wgd_time <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb236-17" data-line-number="17"><span class="st">  </span><span class="kw">gather</span>(<span class="dt">key =</span> <span class="st">&quot;bin&quot;</span>,</a>
<a class="sourceLine" id="cb236-18" data-line-number="18">         <span class="dt">value =</span> <span class="st">&quot;prob&quot;</span>,</a>
<a class="sourceLine" id="cb236-19" data-line-number="19">         <span class="op">-</span>Distribution_Name,</a>
<a class="sourceLine" id="cb236-20" data-line-number="20">         <span class="op">-</span>Sample_ID) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb236-21" data-line-number="21"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">bin =</span> <span class="kw">str_extract</span>(bin, <span class="st">&quot;[0-9].[0-9]+&quot;</span>),</a>
<a class="sourceLine" id="cb236-22" data-line-number="22">         <span class="dt">bin =</span> <span class="kw">as.numeric</span>(bin),</a>
<a class="sourceLine" id="cb236-23" data-line-number="23">         )</a>
<a class="sourceLine" id="cb236-24" data-line-number="24"></a>
<a class="sourceLine" id="cb236-25" data-line-number="25">wgd_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(wgd_time_tidy) <span class="op">+</span></a>
<a class="sourceLine" id="cb236-26" data-line-number="26"><span class="st">  </span><span class="kw">geom_col</span>(<span class="kw">aes</span>(<span class="dt">x =</span> bin,</a>
<a class="sourceLine" id="cb236-27" data-line-number="27">               <span class="dt">y =</span> prob,</a>
<a class="sourceLine" id="cb236-28" data-line-number="28">               <span class="dt">fill =</span> Sample_ID,</a>
<a class="sourceLine" id="cb236-29" data-line-number="29">               <span class="dt">color =</span> Sample_ID)) <span class="op">+</span></a>
<a class="sourceLine" id="cb236-30" data-line-number="30"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> PNWColors<span class="op">::</span><span class="kw">pnw_palette</span>(<span class="st">&quot;Sailboat&quot;</span>, <span class="dv">8</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb236-31" data-line-number="31"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> PNWColors<span class="op">::</span><span class="kw">pnw_palette</span>(<span class="st">&quot;Sailboat&quot;</span>, <span class="dv">8</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb236-32" data-line-number="32"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="kw">vars</span>(Sample_ID), <span class="dt">nrow =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb236-33" data-line-number="33"><span class="st">  </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb236-34" data-line-number="34"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">strip.background =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb236-35" data-line-number="35">        <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb236-36" data-line-number="36"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;relative mutational time&quot;</span>,</a>
<a class="sourceLine" id="cb236-37" data-line-number="37">       <span class="dt">y =</span> <span class="st">&quot;WGD probability&quot;</span>,</a>
<a class="sourceLine" id="cb236-38" data-line-number="38">       <span class="dt">fill =</span> <span class="st">&quot;&quot;</span>,</a>
<a class="sourceLine" id="cb236-39" data-line-number="39">       <span class="dt">color =</span> <span class="st">&quot;&quot;</span>) </a>
<a class="sourceLine" id="cb236-40" data-line-number="40"></a>
<a class="sourceLine" id="cb236-41" data-line-number="41">wgd_plot</a></code></pre></div>
<p><img src="_main_files/figure-html/wgd_mut_timing-1.png" width="672" /></p>
</div>
<div id="wgd-estimations-in-copy-number-time" class="section level2">
<h2><span class="header-section-number">6.7</span> WGD estimations in copy number time</h2>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb237-1" data-line-number="1"><span class="co"># TN5 can&#39;t be confidently estimated and for this reason is excluded</span></a>
<a class="sourceLine" id="cb237-2" data-line-number="2">cn_time_wgd &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/exome/timing/cn_timing_confidence_intervals.tsv&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb237-3" data-line-number="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Sample_ID =</span> <span class="kw">case_when</span>(</a>
<a class="sourceLine" id="cb237-4" data-line-number="4">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;TN28&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN1&quot;</span>,</a>
<a class="sourceLine" id="cb237-5" data-line-number="5">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;TN20&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN2&quot;</span>,</a>
<a class="sourceLine" id="cb237-6" data-line-number="6">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;TN17&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN3&quot;</span>,</a>
<a class="sourceLine" id="cb237-7" data-line-number="7">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;TN26&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN4&quot;</span>,</a>
<a class="sourceLine" id="cb237-8" data-line-number="8">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;TN27&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN8&quot;</span>,</a>
<a class="sourceLine" id="cb237-9" data-line-number="9">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;pD&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN6&quot;</span>,</a>
<a class="sourceLine" id="cb237-10" data-line-number="10">    <span class="kw">str_detect</span>(Sample_ID, <span class="st">&quot;pC&quot;</span>) <span class="op">~</span><span class="st"> &quot;TN7&quot;</span></a>
<a class="sourceLine" id="cb237-11" data-line-number="11">  )) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb237-12" data-line-number="12"><span class="st">  </span><span class="kw">filter</span>(Sample_ID <span class="op">!=</span><span class="st"> &quot;TN5&quot;</span>) </a>
<a class="sourceLine" id="cb237-13" data-line-number="13"></a>
<a class="sourceLine" id="cb237-14" data-line-number="14">cn_wgd_plot &lt;-<span class="st"> </span>cn_time_wgd <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb237-15" data-line-number="15"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> Early_Fraction,</a>
<a class="sourceLine" id="cb237-16" data-line-number="16">             <span class="dt">x =</span> <span class="st">&quot;samp&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb237-17" data-line-number="17"><span class="st">  </span><span class="kw">geom_rect</span>(<span class="kw">aes</span>(<span class="dt">ymin =</span> <span class="dv">0</span>, <span class="dt">ymax =</span> Early_Fraction, <span class="dt">xmin =</span> <span class="op">-</span><span class="ot">Inf</span>, <span class="dt">xmax =</span> <span class="ot">Inf</span>), <span class="dt">color =</span> <span class="st">&#39;gray70&#39;</span>, <span class="dt">alpha =</span> <span class="fl">.4</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb237-18" data-line-number="18"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb237-19" data-line-number="19"><span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">ymin =</span> Early_Fraction_Low_CI,</a>
<a class="sourceLine" id="cb237-20" data-line-number="20">                    <span class="dt">ymax =</span> Early_Fraction_High_CI), </a>
<a class="sourceLine" id="cb237-21" data-line-number="21">                <span class="dt">width =</span> <span class="fl">.3</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb237-22" data-line-number="22"><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label =</span> <span class="st">&quot;WGD&quot;</span>), <span class="dt">nudge_x =</span> <span class="fl">.3</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb237-23" data-line-number="23"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</a>
<a class="sourceLine" id="cb237-24" data-line-number="24">                     <span class="dt">breaks =</span> scales<span class="op">::</span><span class="kw">pretty_breaks</span>(<span class="dt">n =</span> <span class="dv">10</span>),</a>
<a class="sourceLine" id="cb237-25" data-line-number="25">                     <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb237-26" data-line-number="26"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="kw">vars</span>(Sample_ID), <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>, <span class="dt">nrow =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb237-27" data-line-number="27"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.line.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb237-28" data-line-number="28">        <span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb237-29" data-line-number="29">        <span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb237-30" data-line-number="30">        <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb237-31" data-line-number="31">        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">14</span>,</a>
<a class="sourceLine" id="cb237-32" data-line-number="32">                                   <span class="dt">angle =</span> <span class="dv">90</span>,</a>
<a class="sourceLine" id="cb237-33" data-line-number="33">                                   <span class="dt">vjust =</span> <span class="fl">.5</span>,</a>
<a class="sourceLine" id="cb237-34" data-line-number="34">                                   <span class="dt">hjust =</span> <span class="dv">1</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb237-35" data-line-number="35"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">y =</span> <span class="st">&quot;WGD relative copy number time&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb237-36" data-line-number="36"><span class="st">  </span><span class="kw">coord_flip</span>() </a>
<a class="sourceLine" id="cb237-37" data-line-number="37"></a>
<a class="sourceLine" id="cb237-38" data-line-number="38">cn_wgd_plot</a></code></pre></div>
<p><img src="_main_files/figure-html/cn_timing-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="act-10x-cna.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cell-lines.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
