<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Single-Cell Trees | ACT Paper</title>
  <meta name="description" content="4 Single-Cell Trees | ACT Paper" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Single-Cell Trees | ACT Paper" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Single-Cell Trees | ACT Paper" />
  
  
  

<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tumors.html"/>
<link rel="next" href="act-10x-cna.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="instructions.html"><a href="instructions.html"><i class="fa fa-check"></i><b>1</b> Instructions</a></li>
<li class="chapter" data-level="2" data-path="technical-properties.html"><a href="technical-properties.html"><i class="fa fa-check"></i><b>2</b> Technical properties</a><ul>
<li class="chapter" data-level="2.1" data-path="technical-properties.html"><a href="technical-properties.html#breadth-of-coverage"><i class="fa fa-check"></i><b>2.1</b> Breadth of coverage</a></li>
<li class="chapter" data-level="2.2" data-path="technical-properties.html"><a href="technical-properties.html#overdispersion"><i class="fa fa-check"></i><b>2.2</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="tumors.html"><a href="tumors.html"><i class="fa fa-check"></i><b>3</b> Tumors</a><ul>
<li class="chapter" data-level="3.1" data-path="tumors.html"><a href="tumors.html#tn1"><i class="fa fa-check"></i><b>3.1</b> TN1</a></li>
<li class="chapter" data-level="3.2" data-path="tumors.html"><a href="tumors.html#tn2"><i class="fa fa-check"></i><b>3.2</b> TN2</a></li>
<li class="chapter" data-level="3.3" data-path="tumors.html"><a href="tumors.html#tn3"><i class="fa fa-check"></i><b>3.3</b> TN3</a></li>
<li class="chapter" data-level="3.4" data-path="tumors.html"><a href="tumors.html#tn4"><i class="fa fa-check"></i><b>3.4</b> TN4</a></li>
<li class="chapter" data-level="3.5" data-path="tumors.html"><a href="tumors.html#tn5"><i class="fa fa-check"></i><b>3.5</b> TN5</a></li>
<li class="chapter" data-level="3.6" data-path="tumors.html"><a href="tumors.html#tn6"><i class="fa fa-check"></i><b>3.6</b> TN6</a></li>
<li class="chapter" data-level="3.7" data-path="tumors.html"><a href="tumors.html#tn7"><i class="fa fa-check"></i><b>3.7</b> TN7</a></li>
<li class="chapter" data-level="3.8" data-path="tumors.html"><a href="tumors.html#tn8"><i class="fa fa-check"></i><b>3.8</b> TN8</a></li>
<li class="chapter" data-level="3.9" data-path="tumors.html"><a href="tumors.html#clones-barplot"><i class="fa fa-check"></i><b>3.9</b> Clones barplot</a></li>
<li class="chapter" data-level="3.10" data-path="tumors.html"><a href="tumors.html#shannon-diversity"><i class="fa fa-check"></i><b>3.10</b> Shannon diversity</a></li>
<li class="chapter" data-level="3.11" data-path="tumors.html"><a href="tumors.html#gainslosses-plot"><i class="fa fa-check"></i><b>3.11</b> Gains/Losses plot</a></li>
<li class="chapter" data-level="3.12" data-path="tumors.html"><a href="tumors.html#genomic-classes-plots"><i class="fa fa-check"></i><b>3.12</b> Genomic classes plots</a><ul>
<li class="chapter" data-level="3.12.1" data-path="tumors.html"><a href="tumors.html#counts"><i class="fa fa-check"></i><b>3.12.1</b> Counts</a></li>
<li class="chapter" data-level="3.12.2" data-path="tumors.html"><a href="tumors.html#segment-length"><i class="fa fa-check"></i><b>3.12.2</b> Segment length</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="single-cell-trees.html"><a href="single-cell-trees.html"><i class="fa fa-check"></i><b>4</b> Single-Cell Trees</a></li>
<li class="chapter" data-level="5" data-path="act-10x-cna.html"><a href="act-10x-cna.html"><i class="fa fa-check"></i><b>5</b> ACT &amp; 10X CNA</a><ul>
<li class="chapter" data-level="5.1" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn1-co-clustering-umaps"><i class="fa fa-check"></i><b>5.1</b> TN1 Co-clustering umaps</a></li>
<li class="chapter" data-level="5.2" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn1-gene-frequency-barplot"><i class="fa fa-check"></i><b>5.2</b> TN1 Gene Frequency Barplot</a></li>
<li class="chapter" data-level="5.3" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn3-co-clustering-umaps"><i class="fa fa-check"></i><b>5.3</b> TN3 Co-clustering umaps</a></li>
<li class="chapter" data-level="5.4" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn3-gene-frequency-barplot"><i class="fa fa-check"></i><b>5.4</b> TN3 Gene Frequency Barplot</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html"><i class="fa fa-check"></i><b>6</b> Exome and WGD</a><ul>
<li class="chapter" data-level="6.1" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#data"><i class="fa fa-check"></i><b>6.1</b> Data</a></li>
<li class="chapter" data-level="6.2" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#filtering"><i class="fa fa-check"></i><b>6.2</b> Filtering</a></li>
<li class="chapter" data-level="6.3" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#mut-timer-information"><i class="fa fa-check"></i><b>6.3</b> Mut timer information</a></li>
<li class="chapter" data-level="6.4" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#mutation-burden"><i class="fa fa-check"></i><b>6.4</b> Mutation burden</a></li>
<li class="chapter" data-level="6.5" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#oncomap"><i class="fa fa-check"></i><b>6.5</b> Oncomap</a></li>
<li class="chapter" data-level="6.6" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#wgd-estimation-in-mutational-timing"><i class="fa fa-check"></i><b>6.6</b> WGD estimation in mutational timing</a></li>
<li class="chapter" data-level="6.7" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#wgd-estimations-in-copy-number-time"><i class="fa fa-check"></i><b>6.7</b> WGD estimations in copy number time</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cell-lines.html"><a href="cell-lines.html"><i class="fa fa-check"></i><b>7</b> Cell lines</a><ul>
<li class="chapter" data-level="7.1" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231"><i class="fa fa-check"></i><b>7.1</b> MDA-MB-231</a></li>
<li class="chapter" data-level="7.2" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231-ex1"><i class="fa fa-check"></i><b>7.2</b> MDA-MB-231-EX1</a></li>
<li class="chapter" data-level="7.3" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231-ex2"><i class="fa fa-check"></i><b>7.3</b> MDA-MB-231-EX2</a></li>
<li class="chapter" data-level="7.4" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-453"><i class="fa fa-check"></i><b>7.4</b> MDA-MB-453</a></li>
<li class="chapter" data-level="7.5" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-157"><i class="fa fa-check"></i><b>7.5</b> MDA-MB-157</a></li>
<li class="chapter" data-level="7.6" data-path="cell-lines.html"><a href="cell-lines.html#bt20"><i class="fa fa-check"></i><b>7.6</b> BT20</a></li>
<li class="chapter" data-level="7.7" data-path="cell-lines.html"><a href="cell-lines.html#clones-barplot-1"><i class="fa fa-check"></i><b>7.7</b> Clones barplot</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html"><i class="fa fa-check"></i><b>8</b> MDA-231 DNA &amp; RNA</a><ul>
<li class="chapter" data-level="8.1" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#co-clustering"><i class="fa fa-check"></i><b>8.1</b> Co-clustering</a></li>
<li class="chapter" data-level="8.2" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#single-cell-bulk-heatmap"><i class="fa fa-check"></i><b>8.2</b> Single-cell + Bulk Heatmap</a></li>
<li class="chapter" data-level="8.3" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#bulk-heatmap"><i class="fa fa-check"></i><b>8.3</b> Bulk Heatmap</a></li>
<li class="chapter" data-level="8.4" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#dna-rna"><i class="fa fa-check"></i><b>8.4</b> DNA &amp; RNA</a></li>
<li class="chapter" data-level="8.5" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#moving-averages"><i class="fa fa-check"></i><b>8.5</b> Moving averages</a></li>
<li class="chapter" data-level="8.6" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#de"><i class="fa fa-check"></i><b>8.6</b> DE</a></li>
<li class="chapter" data-level="8.7" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#rna-pca"><i class="fa fa-check"></i><b>8.7</b> RNA PCA</a></li>
<li class="chapter" data-level="8.8" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#fgsea"><i class="fa fa-check"></i><b>8.8</b> FGSEA</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>9</b> Session Info</a><ul>
<li class="chapter" data-level="9.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>9.1</b> Dockerfile</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="instructions-1.html"><a href="instructions-1.html"><i class="fa fa-check"></i><b>10</b> Instructions</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ACT Paper</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="single-cell-trees" class="section level1">
<h1><span class="header-section-number">4</span> Single-Cell Trees</h1>
<p>Function used to generate the trees using <code>ape::fastme.bal()</code> is <code>make_single_cell_tree</code>. Trees generated from the following function are provided in the directory <code>extdata/trees</code></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">make_single_cell_tree &lt;-<span class="st"> </span><span class="cf">function</span>(tumor_log_segratio,</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">                                  <span class="dt">ploidy_VAL_diploid =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">                                  tumor_ploidy_VAL,</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">                                  <span class="dt">tree_fun =</span> <span class="st">&quot;fastme.bal&quot;</span>,</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">                                  <span class="dt">dist_metric =</span> <span class="st">&quot;manhattan&quot;</span>) {</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">  </a>
<a class="sourceLine" id="cb2-7" data-line-number="7">  <span class="st">&#39;%!in%&#39;</span> &lt;-<span class="st"> </span><span class="cf">function</span>(x,y)<span class="op">!</span>(<span class="st">&#39;%in%&#39;</span>(x,y))</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">  </a>
<a class="sourceLine" id="cb2-9" data-line-number="9">  <span class="cf">if</span> (tree_fun <span class="op">%!in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;nj&quot;</span>, <span class="st">&quot;fastme.bal&quot;</span>)) {</a>
<a class="sourceLine" id="cb2-10" data-line-number="10">    <span class="kw">stop</span>(<span class="st">&quot;Please input a valid tree function. Accepted values are &#39;nj&#39; or &#39;fastme.bal&#39;&quot;</span>)</a>
<a class="sourceLine" id="cb2-11" data-line-number="11">  }</a>
<a class="sourceLine" id="cb2-12" data-line-number="12">  </a>
<a class="sourceLine" id="cb2-13" data-line-number="13">  normal_long &lt;-<span class="st"> </span><span class="kw">tibble</span>(</a>
<a class="sourceLine" id="cb2-14" data-line-number="14">    <span class="dt">C1 =</span> <span class="kw">rep.int</span>(<span class="fl">1e-3</span>, <span class="kw">nrow</span>(tumor_log_segratio)),</a>
<a class="sourceLine" id="cb2-15" data-line-number="15">    <span class="dt">C2 =</span> <span class="kw">rep.int</span>(<span class="fl">1e-3</span>, <span class="kw">nrow</span>(tumor_log_segratio)),</a>
<a class="sourceLine" id="cb2-16" data-line-number="16">    <span class="dt">C3 =</span> <span class="kw">rep.int</span>(<span class="fl">1e-3</span>, <span class="kw">nrow</span>(tumor_log_segratio)),</a>
<a class="sourceLine" id="cb2-17" data-line-number="17">    <span class="dt">C4 =</span> <span class="kw">rep.int</span>(<span class="fl">1e-3</span>, <span class="kw">nrow</span>(tumor_log_segratio))</a>
<a class="sourceLine" id="cb2-18" data-line-number="18">  )</a>
<a class="sourceLine" id="cb2-19" data-line-number="19">  normal_long &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">t</span>(normal_long))</a>
<a class="sourceLine" id="cb2-20" data-line-number="20">  n_ploidy &lt;-<span class="st"> </span><span class="kw">ploidy_scale</span>(ploidy_VAL_diploid, normal_long)</a>
<a class="sourceLine" id="cb2-21" data-line-number="21">  <span class="kw">rownames</span>(n_ploidy) &lt;-<span class="st"> </span><span class="kw">rownames</span>(normal_long)</a>
<a class="sourceLine" id="cb2-22" data-line-number="22">  </a>
<a class="sourceLine" id="cb2-23" data-line-number="23">  <span class="co"># tumor_long &lt;- create_popseg_long(popseg_tumor) # tumor cells only</span></a>
<a class="sourceLine" id="cb2-24" data-line-number="24">  t_ploidy &lt;-<span class="st"> </span><span class="kw">ploidy_scale</span>(tumor_ploidy_VAL, tumor_log_segratio)</a>
<a class="sourceLine" id="cb2-25" data-line-number="25"></a>
<a class="sourceLine" id="cb2-26" data-line-number="26">    <span class="co"># combining normal and tumor cells</span></a>
<a class="sourceLine" id="cb2-27" data-line-number="27">  combined &lt;-<span class="st"> </span><span class="kw">rbind</span>(n_ploidy, t_ploidy)</a>
<a class="sourceLine" id="cb2-28" data-line-number="28"></a>
<a class="sourceLine" id="cb2-29" data-line-number="29">  <span class="cf">if</span> (tree_fun <span class="op">==</span><span class="st"> &quot;nj&quot;</span>) {</a>
<a class="sourceLine" id="cb2-30" data-line-number="30">    tree &lt;-<span class="st"> </span>ape<span class="op">::</span><span class="kw">nj</span>(amap<span class="op">::</span><span class="kw">Dist</span>(combined, <span class="dt">method=</span>dist_metric, <span class="dt">nbproc =</span> <span class="dv">40</span>))</a>
<a class="sourceLine" id="cb2-31" data-line-number="31">    <span class="kw">return</span>(tree)</a>
<a class="sourceLine" id="cb2-32" data-line-number="32">  }</a>
<a class="sourceLine" id="cb2-33" data-line-number="33">  </a>
<a class="sourceLine" id="cb2-34" data-line-number="34">  <span class="cf">if</span> (tree_fun <span class="op">==</span><span class="st"> &quot;fastme.bal&quot;</span>) {</a>
<a class="sourceLine" id="cb2-35" data-line-number="35">    tree &lt;-<span class="st"> </span>ape<span class="op">::</span><span class="kw">fastme.bal</span>(amap<span class="op">::</span><span class="kw">Dist</span>(combined, <span class="dt">method=</span>dist_metric, <span class="dt">nbproc =</span> <span class="dv">40</span>))</a>
<a class="sourceLine" id="cb2-36" data-line-number="36">    <span class="kw">return</span>(tree)</a>
<a class="sourceLine" id="cb2-37" data-line-number="37">  } </a>
<a class="sourceLine" id="cb2-38" data-line-number="38">  </a>
<a class="sourceLine" id="cb2-39" data-line-number="39">}</a></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">tn1_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/trees/tn1_sc_tree.tree&quot;</span>))</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">tn2_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/trees/tn2_sc_tree.tree&quot;</span>))</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">tn3_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/trees/tn3_sc_tree.tree&quot;</span>))</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">tn4_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/trees/tn4_sc_tree.tree&quot;</span>))</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">tn5_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/trees/tn5_sc_tree.tree&quot;</span>))</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">tn6_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/trees/tn6_sc_tree.tree&quot;</span>))</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">tn7_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/trees/tn7_sc_tree.tree&quot;</span>))</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">tn8_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/trees/tn8_sc_tree.tree&quot;</span>))</a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">source</span>(<span class="kw">here</span>(<span class="st">&quot;R/plot_sctree.R&quot;</span>))</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="kw">plot_sctree</span>(tn1_tree, <span class="dt">title =</span> <span class="st">&quot;TN1&quot;</span>)</a></code></pre></div>
<pre><code>## Warning: `data_frame()` is deprecated as of tibble 1.1.0.
## Please use `tibble()` instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<pre><code>## Warning: `mutate_()` is deprecated as of dplyr 0.7.0.
## Please use `mutate()` instead.
## See vignette(&#39;programming&#39;) for more help
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<p><img src="_main_files/figure-html/plots-1.png" width="672" /></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">plot_sctree</span>(tn2_tree, <span class="dt">title =</span> <span class="st">&quot;TN2&quot;</span>)</a></code></pre></div>
<p><img src="_main_files/figure-html/plots-2.png" width="672" /></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">plot_sctree</span>(tn3_tree, <span class="dt">title =</span> <span class="st">&quot;TN3&quot;</span>)</a></code></pre></div>
<p><img src="_main_files/figure-html/plots-3.png" width="672" /></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">plot_sctree</span>(tn4_tree, <span class="dt">title =</span> <span class="st">&quot;TN4&quot;</span>, <span class="dt">anno_y =</span> <span class="dv">1450</span>)</a></code></pre></div>
<p><img src="_main_files/figure-html/plots-4.png" width="672" /></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">plot_sctree</span>(tn5_tree, <span class="dt">title =</span> <span class="st">&quot;TN5&quot;</span>, <span class="dt">anno_y =</span> <span class="dv">1350</span>)</a></code></pre></div>
<p><img src="_main_files/figure-html/plots-5.png" width="672" /></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">plot_sctree</span>(tn6_tree, <span class="dt">title =</span> <span class="st">&quot;TN6&quot;</span>, <span class="dt">anno_y =</span> <span class="dv">1550</span>)</a></code></pre></div>
<p><img src="_main_files/figure-html/plots-6.png" width="672" /></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">plot_sctree</span>(tn7_tree, <span class="dt">title =</span> <span class="st">&quot;TN7&quot;</span>, <span class="dt">anno_y =</span> <span class="dv">1550</span>)</a></code></pre></div>
<p><img src="_main_files/figure-html/plots-7.png" width="672" /></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">plot_sctree</span>(tn8_tree, <span class="dt">title =</span> <span class="st">&quot;TN8&quot;</span>)</a></code></pre></div>
<p><img src="_main_files/figure-html/plots-8.png" width="672" /></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">tn1_dist_nodes &lt;-<span class="st"> </span><span class="kw">calc_sctree_dists</span>(tn1_tree) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN1&quot;</span>) </a>
<a class="sourceLine" id="cb14-2" data-line-number="2">tn2_dist_nodes &lt;-<span class="st"> </span><span class="kw">calc_sctree_dists</span>(tn2_tree) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN2&quot;</span>)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">tn3_dist_nodes &lt;-<span class="st"> </span><span class="kw">calc_sctree_dists</span>(tn3_tree) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN3&quot;</span>)</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">tn4_dist_nodes &lt;-<span class="st"> </span><span class="kw">calc_sctree_dists</span>(tn4_tree) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN4&quot;</span>)</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">tn5_dist_nodes &lt;-<span class="st"> </span><span class="kw">calc_sctree_dists</span>(tn5_tree) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN5&quot;</span>)</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">tn6_dist_nodes &lt;-<span class="st"> </span><span class="kw">calc_sctree_dists</span>(tn6_tree) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN6&quot;</span>)</a>
<a class="sourceLine" id="cb14-7" data-line-number="7">tn7_dist_nodes &lt;-<span class="st"> </span><span class="kw">calc_sctree_dists</span>(tn7_tree) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN7&quot;</span>)</a>
<a class="sourceLine" id="cb14-8" data-line-number="8">tn8_dist_nodes &lt;-<span class="st"> </span><span class="kw">calc_sctree_dists</span>(tn8_tree) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN8&quot;</span>)</a>
<a class="sourceLine" id="cb14-9" data-line-number="9"></a>
<a class="sourceLine" id="cb14-10" data-line-number="10">tumors_dist_nodes &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(</a>
<a class="sourceLine" id="cb14-11" data-line-number="11">  tn1_dist_nodes,</a>
<a class="sourceLine" id="cb14-12" data-line-number="12">  tn2_dist_nodes,</a>
<a class="sourceLine" id="cb14-13" data-line-number="13">  tn3_dist_nodes,</a>
<a class="sourceLine" id="cb14-14" data-line-number="14">  tn4_dist_nodes,</a>
<a class="sourceLine" id="cb14-15" data-line-number="15">  tn5_dist_nodes,</a>
<a class="sourceLine" id="cb14-16" data-line-number="16">  tn6_dist_nodes,</a>
<a class="sourceLine" id="cb14-17" data-line-number="17">  tn7_dist_nodes,</a>
<a class="sourceLine" id="cb14-18" data-line-number="18">  tn8_dist_nodes</a>
<a class="sourceLine" id="cb14-19" data-line-number="19">)</a>
<a class="sourceLine" id="cb14-20" data-line-number="20"></a>
<a class="sourceLine" id="cb14-21" data-line-number="21">sc_trees_dist_plots &lt;-<span class="st"> </span>tumors_dist_nodes <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-22" data-line-number="22"><span class="st">  </span><span class="kw">gather</span>(<span class="dt">key =</span> <span class="st">&quot;evolution&quot;</span>,</a>
<a class="sourceLine" id="cb14-23" data-line-number="23">         <span class="dt">value =</span> <span class="st">&quot;manhattan_dist&quot;</span>, </a>
<a class="sourceLine" id="cb14-24" data-line-number="24">         <span class="op">-</span>sample,</a>
<a class="sourceLine" id="cb14-25" data-line-number="25">         <span class="op">-</span>truncal_node) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-26" data-line-number="26"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> manhattan_dist,</a>
<a class="sourceLine" id="cb14-27" data-line-number="27">                 <span class="dt">x =</span> <span class="kw">fct_relevel</span>(sample, <span class="kw">rev</span>(gtools<span class="op">::</span><span class="kw">mixedsort</span>(tumors_dist_nodes<span class="op">$</span>sample))))) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-28" data-line-number="28"><span class="st">  </span><span class="kw">geom_linerange</span>(<span class="kw">aes</span>(<span class="dt">ymin =</span> <span class="dv">0</span>, <span class="dt">ymax =</span> manhattan_dist, <span class="dt">group =</span> evolution),</a>
<a class="sourceLine" id="cb14-29" data-line-number="29">                 <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="dt">width =</span> <span class="fl">.5</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-30" data-line-number="30"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">color =</span> evolution,</a>
<a class="sourceLine" id="cb14-31" data-line-number="31">             <span class="dt">group =</span> evolution),</a>
<a class="sourceLine" id="cb14-32" data-line-number="32">             <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="dt">width =</span> <span class="fl">.5</span>),</a>
<a class="sourceLine" id="cb14-33" data-line-number="33">             <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-34" data-line-number="34"><span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb14-35" data-line-number="35"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-36" data-line-number="36"><span class="st">  </span><span class="kw">scale_color_paletteer_d</span>(<span class="st">&quot;ggthemes::hc_default&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-37" data-line-number="37"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">y =</span> <span class="st">&quot;manhattan distance&quot;</span>,</a>
<a class="sourceLine" id="cb14-38" data-line-number="38">       <span class="dt">color =</span> <span class="st">&quot;&quot;</span>,</a>
<a class="sourceLine" id="cb14-39" data-line-number="39">       <span class="dt">x =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-40" data-line-number="40"><span class="st">  </span><span class="kw">theme_cowplot</span>()</a>
<a class="sourceLine" id="cb14-41" data-line-number="41"></a>
<a class="sourceLine" id="cb14-42" data-line-number="42">sc_trees_dist_plots</a></code></pre></div>
<p><img src="_main_files/figure-html/lolliplot-1.png" width="672" /></p>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="tumors.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="act-10x-cna.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
