<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8 MDA-231 DNA &amp; RNA | ACT Paper</title>
  <meta name="description" content="8 MDA-231 DNA &amp; RNA | ACT Paper" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="8 MDA-231 DNA &amp; RNA | ACT Paper" />
  <meta property="og:type" content="book" />





  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8 MDA-231 DNA &amp; RNA | ACT Paper" />




<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />


<link rel="prev" href="cell-lines.html"/>
<link rel="next" href="session-info.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Instructions</a></li>
<li class="chapter" data-level="2" data-path="technical-properties.html"><a href="technical-properties.html"><i class="fa fa-check"></i><b>2</b> Technical properties</a><ul>
<li class="chapter" data-level="2.1" data-path="technical-properties.html"><a href="technical-properties.html#breadth-of-coverage"><i class="fa fa-check"></i><b>2.1</b> Breadth of coverage</a></li>
<li class="chapter" data-level="2.2" data-path="technical-properties.html"><a href="technical-properties.html#overdispersion"><i class="fa fa-check"></i><b>2.2</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="tumors.html"><a href="tumors.html"><i class="fa fa-check"></i><b>3</b> Tumors</a><ul>
<li class="chapter" data-level="3.1" data-path="tumors.html"><a href="tumors.html#tn1"><i class="fa fa-check"></i><b>3.1</b> TN1</a></li>
<li class="chapter" data-level="3.2" data-path="tumors.html"><a href="tumors.html#tn2"><i class="fa fa-check"></i><b>3.2</b> TN2</a></li>
<li class="chapter" data-level="3.3" data-path="tumors.html"><a href="tumors.html#tn3"><i class="fa fa-check"></i><b>3.3</b> TN3</a></li>
<li class="chapter" data-level="3.4" data-path="tumors.html"><a href="tumors.html#tn4"><i class="fa fa-check"></i><b>3.4</b> TN4</a></li>
<li class="chapter" data-level="3.5" data-path="tumors.html"><a href="tumors.html#tn5"><i class="fa fa-check"></i><b>3.5</b> TN5</a></li>
<li class="chapter" data-level="3.6" data-path="tumors.html"><a href="tumors.html#tn6"><i class="fa fa-check"></i><b>3.6</b> TN6</a></li>
<li class="chapter" data-level="3.7" data-path="tumors.html"><a href="tumors.html#tn7"><i class="fa fa-check"></i><b>3.7</b> TN7</a></li>
<li class="chapter" data-level="3.8" data-path="tumors.html"><a href="tumors.html#tn8"><i class="fa fa-check"></i><b>3.8</b> TN8</a></li>
<li class="chapter" data-level="3.9" data-path="tumors.html"><a href="tumors.html#clones-barplot"><i class="fa fa-check"></i><b>3.9</b> Clones barplot</a></li>
<li class="chapter" data-level="3.10" data-path="tumors.html"><a href="tumors.html#shannon-diversity"><i class="fa fa-check"></i><b>3.10</b> Shannon diversity</a></li>
<li class="chapter" data-level="3.11" data-path="tumors.html"><a href="tumors.html#gainslosses-plot"><i class="fa fa-check"></i><b>3.11</b> Gains/Losses plot</a></li>
<li class="chapter" data-level="3.12" data-path="tumors.html"><a href="tumors.html#genomic-classes-plots"><i class="fa fa-check"></i><b>3.12</b> Genomic classes plots</a><ul>
<li class="chapter" data-level="3.12.1" data-path="tumors.html"><a href="tumors.html#counts"><i class="fa fa-check"></i><b>3.12.1</b> Counts</a></li>
<li class="chapter" data-level="3.12.2" data-path="tumors.html"><a href="tumors.html#segment-length"><i class="fa fa-check"></i><b>3.12.2</b> Segment length</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="single-cell-trees.html"><a href="single-cell-trees.html"><i class="fa fa-check"></i><b>4</b> Single-Cell Trees</a></li>
<li class="chapter" data-level="5" data-path="act-10x-cna.html"><a href="act-10x-cna.html"><i class="fa fa-check"></i><b>5</b> ACT &amp; 10X CNA</a><ul>
<li class="chapter" data-level="5.1" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn1-co-clustering-umaps"><i class="fa fa-check"></i><b>5.1</b> TN1 Co-clustering umaps</a></li>
<li class="chapter" data-level="5.2" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn1-gene-frequency-barplot"><i class="fa fa-check"></i><b>5.2</b> TN1 Gene Frequency Barplot</a></li>
<li class="chapter" data-level="5.3" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn3-co-clustering-umaps"><i class="fa fa-check"></i><b>5.3</b> TN3 Co-clustering umaps</a></li>
<li class="chapter" data-level="5.4" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn3-gene-frequency-barplot"><i class="fa fa-check"></i><b>5.4</b> TN3 Gene Frequency Barplot</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html"><i class="fa fa-check"></i><b>6</b> Exome and WGD</a><ul>
<li class="chapter" data-level="6.1" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#data"><i class="fa fa-check"></i><b>6.1</b> Data</a></li>
<li class="chapter" data-level="6.2" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#filtering"><i class="fa fa-check"></i><b>6.2</b> Filtering</a></li>
<li class="chapter" data-level="6.3" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#mut-timer-information"><i class="fa fa-check"></i><b>6.3</b> Mut timer information</a></li>
<li class="chapter" data-level="6.4" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#mutation-burden"><i class="fa fa-check"></i><b>6.4</b> Mutation burden</a></li>
<li class="chapter" data-level="6.5" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#oncomap"><i class="fa fa-check"></i><b>6.5</b> Oncomap</a></li>
<li class="chapter" data-level="6.6" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#wgd-estimation-in-mutational-timing"><i class="fa fa-check"></i><b>6.6</b> WGD estimation in mutational timing</a></li>
<li class="chapter" data-level="6.7" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#wgd-estimations-in-copy-number-time"><i class="fa fa-check"></i><b>6.7</b> WGD estimations in copy number time</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cell-lines.html"><a href="cell-lines.html"><i class="fa fa-check"></i><b>7</b> Cell lines</a><ul>
<li class="chapter" data-level="7.1" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231"><i class="fa fa-check"></i><b>7.1</b> MDA-MB-231</a></li>
<li class="chapter" data-level="7.2" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231-ex1"><i class="fa fa-check"></i><b>7.2</b> MDA-MB-231-EX1</a></li>
<li class="chapter" data-level="7.3" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231-ex2"><i class="fa fa-check"></i><b>7.3</b> MDA-MB-231-EX2</a></li>
<li class="chapter" data-level="7.4" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-453"><i class="fa fa-check"></i><b>7.4</b> MDA-MB-453</a></li>
<li class="chapter" data-level="7.5" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-157"><i class="fa fa-check"></i><b>7.5</b> MDA-MB-157</a></li>
<li class="chapter" data-level="7.6" data-path="cell-lines.html"><a href="cell-lines.html#bt20"><i class="fa fa-check"></i><b>7.6</b> BT20</a></li>
<li class="chapter" data-level="7.7" data-path="cell-lines.html"><a href="cell-lines.html#clones-barplot-1"><i class="fa fa-check"></i><b>7.7</b> Clones barplot</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html"><i class="fa fa-check"></i><b>8</b> MDA-231 DNA &amp; RNA</a><ul>
<li class="chapter" data-level="8.1" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#co-clustering"><i class="fa fa-check"></i><b>8.1</b> Co-clustering</a></li>
<li class="chapter" data-level="8.2" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#single-cell-bulk-heatmap"><i class="fa fa-check"></i><b>8.2</b> Single-cell + Bulk Heatmap</a></li>
<li class="chapter" data-level="8.3" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#bulk-heatmap"><i class="fa fa-check"></i><b>8.3</b> Bulk Heatmap</a></li>
<li class="chapter" data-level="8.4" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#dna-rna"><i class="fa fa-check"></i><b>8.4</b> DNA &amp; RNA</a></li>
<li class="chapter" data-level="8.5" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#moving-averages"><i class="fa fa-check"></i><b>8.5</b> Moving averages</a></li>
<li class="chapter" data-level="8.6" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#de"><i class="fa fa-check"></i><b>8.6</b> DE</a></li>
<li class="chapter" data-level="8.7" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#rna-pca"><i class="fa fa-check"></i><b>8.7</b> RNA PCA</a></li>
<li class="chapter" data-level="8.8" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#fgsea"><i class="fa fa-check"></i><b>8.8</b> FGSEA</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>9</b> Session Info</a><ul>
<li class="chapter" data-level="9.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>9.1</b> Dockerfile</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="instructions-1.html"><a href="instructions-1.html"><i class="fa fa-check"></i><b>10</b> Instructions</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ACT Paper</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mda-231-dna-rna" class="section level1">
<h1><span class="header-section-number">8</span> MDA-231 DNA &amp; RNA</h1>
<p>Transcript abundances for expanded clones triplicates were quantified by Salmon (v.0.14) with GENCODE transcript v30 and options -l A -1 read1 -2 read2 -p 40 –validateMappings –seqBias –gcBias. Quantified transcripts were imported into R with ‘tximport’ (v 1.14). Expanded clones e7, e39 and e71 had one technical replicate excluded due to poor RNA quality</p>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb322-1" data-line-number="1"><span class="kw">source</span>(<span class="st">&quot;R/setup.R&quot;</span>)</a>
<a class="sourceLine" id="cb322-2" data-line-number="2"><span class="kw">source</span>(<span class="st">&quot;R/run_umap.R&quot;</span>)</a>
<a class="sourceLine" id="cb322-3" data-line-number="3"><span class="kw">source</span>(<span class="st">&quot;R/run_clustering.R&quot;</span>)</a>
<a class="sourceLine" id="cb322-4" data-line-number="4"><span class="kw">source</span>(<span class="st">&quot;R/order_dataset.R&quot;</span>)</a>
<a class="sourceLine" id="cb322-5" data-line-number="5"><span class="kw">source</span>(<span class="st">&quot;R/plot_umap.R&quot;</span>)</a>
<a class="sourceLine" id="cb322-6" data-line-number="6"><span class="kw">source</span>(<span class="st">&quot;R/calculate_consensus.R&quot;</span>)</a>
<a class="sourceLine" id="cb322-7" data-line-number="7"><span class="kw">source</span>(<span class="st">&quot;R/consensus_genomic_classes.R&quot;</span>)</a>
<a class="sourceLine" id="cb322-8" data-line-number="8"><span class="kw">source</span>(<span class="st">&quot;R/run_me_tree.R&quot;</span>)</a>
<a class="sourceLine" id="cb322-9" data-line-number="9"><span class="kw">source</span>(<span class="st">&quot;R/plot_heatmap.R&quot;</span>)</a>
<a class="sourceLine" id="cb322-10" data-line-number="10"><span class="kw">source</span>(<span class="st">&quot;R/plot_moving_averages.R&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb323-1" data-line-number="1"><span class="co"># ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Sat Mar  6 17:34:47 2021</span></a>
<a class="sourceLine" id="cb323-2" data-line-number="2"><span class="co"># RNA</span></a>
<a class="sourceLine" id="cb323-3" data-line-number="3"><span class="co"># ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Sat Mar  6 17:34:51 2021</span></a>
<a class="sourceLine" id="cb323-4" data-line-number="4"></a>
<a class="sourceLine" id="cb323-5" data-line-number="5"><span class="co"># reading RNA data</span></a>
<a class="sourceLine" id="cb323-6" data-line-number="6">dds &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/rna/mdamb231_rna_dds.rds&quot;</span>))</a>
<a class="sourceLine" id="cb323-7" data-line-number="7"></a>
<a class="sourceLine" id="cb323-8" data-line-number="8"><span class="co"># Quality filtering</span></a>
<a class="sourceLine" id="cb323-9" data-line-number="9">keep &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">counts</span>(dds) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">5</span>) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">3</span></a>
<a class="sourceLine" id="cb323-10" data-line-number="10">dds &lt;-<span class="st"> </span>dds[keep,]</a>
<a class="sourceLine" id="cb323-11" data-line-number="11"></a>
<a class="sourceLine" id="cb323-12" data-line-number="12"><span class="co"># computation of size factors which normalize for </span></a>
<a class="sourceLine" id="cb323-13" data-line-number="13"><span class="co"># differences in sequencing depth among samples</span></a>
<a class="sourceLine" id="cb323-14" data-line-number="14">dds &lt;-<span class="st"> </span><span class="kw">estimateSizeFactors</span>(dds)</a></code></pre></div>
<pre><code>## using &#39;avgTxLength&#39; from assays(dds), correcting for library size</code></pre>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb325-1" data-line-number="1"><span class="kw">boxplot</span>(<span class="kw">log10</span>(<span class="kw">counts</span>(dds,<span class="dt">normalized=</span><span class="ot">TRUE</span>)<span class="op">+</span><span class="dv">1</span>), <span class="dt">las =</span> <span class="dv">2</span>)</a></code></pre></div>
<p><img src="_main_files/figure-html/data-1.png" width="672" /></p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb326-1" data-line-number="1"><span class="co"># variance-stabilizing transformation</span></a>
<a class="sourceLine" id="cb326-2" data-line-number="2">vsd &lt;-<span class="st"> </span><span class="kw">vst</span>(dds)</a>
<a class="sourceLine" id="cb326-3" data-line-number="3"></a>
<a class="sourceLine" id="cb326-4" data-line-number="4"><span class="co"># ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Sat Mar  6 17:34:56 2021</span></a>
<a class="sourceLine" id="cb326-5" data-line-number="5"><span class="co"># DNA</span></a>
<a class="sourceLine" id="cb326-6" data-line-number="6"><span class="co"># ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Sat Mar  6 17:35:01 2021</span></a>
<a class="sourceLine" id="cb326-7" data-line-number="7"></a>
<a class="sourceLine" id="cb326-8" data-line-number="8">mda_ploidy &lt;-<span class="st"> </span><span class="fl">2.41</span></a>
<a class="sourceLine" id="cb326-9" data-line-number="9"></a>
<a class="sourceLine" id="cb326-10" data-line-number="10"><span class="co"># data from DNA bulk expanded clones </span></a>
<a class="sourceLine" id="cb326-11" data-line-number="11"><span class="co"># and from mda-mb-231 single-cell were merged to the same dataframe</span></a>
<a class="sourceLine" id="cb326-12" data-line-number="12"></a>
<a class="sourceLine" id="cb326-13" data-line-number="13">mdamb231_sc_bulk_cna_popseg &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/popseg/mdamb231_sc_bulk_cna_popseg.rds&quot;</span>))</a>
<a class="sourceLine" id="cb326-14" data-line-number="14">mdamb231_sc_bulk_cna_ml &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/merge_levels/mdamb231_sc_bulk_cna_ml.rds&quot;</span>))</a></code></pre></div>
<div id="co-clustering" class="section level2">
<h2><span class="header-section-number">8.1</span> Co-clustering</h2>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb327-1" data-line-number="1">mdamb231_sc_bulk_umap &lt;-<span class="st"> </span><span class="kw">run_umap</span>(mdamb231_sc_bulk_cna_ml,</a>
<a class="sourceLine" id="cb327-2" data-line-number="2">                                  <span class="dt">umap_n_neighbors =</span> <span class="dv">25</span>,</a>
<a class="sourceLine" id="cb327-3" data-line-number="3">                                  <span class="dt">seed =</span> <span class="dv">5</span>)</a></code></pre></div>
<pre><code>## Constructing UMAP embedding.</code></pre>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb329-1" data-line-number="1">mdamb231_sc_bulk_clustering &lt;-<span class="st"> </span><span class="kw">run_clustering</span>(mdamb231_sc_bulk_umap,</a>
<a class="sourceLine" id="cb329-2" data-line-number="2">                                 <span class="dt">k_snn_major =</span> <span class="dv">43</span>,</a>
<a class="sourceLine" id="cb329-3" data-line-number="3">                                 <span class="dt">k_snn_minor =</span> <span class="dv">14</span>) </a></code></pre></div>
<pre><code>## Building SNN graph.</code></pre>
<pre><code>## Running hdbscan.</code></pre>
<pre><code>##  cluster   n    percent
##       c1 176 0.19599109
##      c10 203 0.22605791
##      c11  36 0.04008909
##      c12  14 0.01559020
##      c13 197 0.21937639
##       c2  28 0.03118040
##       c3  59 0.06570156
##       c4  52 0.05790646
##       c5  32 0.03563474
##       c6  42 0.04677060
##       c7  20 0.02227171
##       c8  16 0.01781737
##       c9  23 0.02561247</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb334-1" data-line-number="1">mdamb231_sc_bulk_clustering_umap &lt;-<span class="st"> </span><span class="kw">left_join</span>(mdamb231_sc_bulk_umap,</a>
<a class="sourceLine" id="cb334-2" data-line-number="2">                                              mdamb231_sc_bulk_clustering, </a>
<a class="sourceLine" id="cb334-3" data-line-number="3">                                              <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;cell&quot;</span> =<span class="st"> &quot;cells&quot;</span>))</a>
<a class="sourceLine" id="cb334-4" data-line-number="4"></a>
<a class="sourceLine" id="cb334-5" data-line-number="5">mdamb231_sc_bulk_clustering_umap &lt;-<span class="st"> </span>mdamb231_sc_bulk_clustering_umap <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb334-6" data-line-number="6"><span class="st">  </span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb334-7" data-line-number="7">    <span class="dt">type =</span> <span class="kw">case_when</span>(<span class="kw">str_detect</span>(cell, <span class="st">&quot;gDNA&quot;</span>) <span class="op">~</span><span class="st"> &quot;bulk&quot;</span>,</a>
<a class="sourceLine" id="cb334-8" data-line-number="8">                     <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> </span>subclones),</a>
<a class="sourceLine" id="cb334-9" data-line-number="9">    <span class="dt">shape =</span> <span class="kw">case_when</span>(<span class="kw">str_detect</span>(cell, <span class="st">&quot;gDNA&quot;</span>) <span class="op">~</span><span class="st"> </span><span class="dv">8</span>,</a>
<a class="sourceLine" id="cb334-10" data-line-number="10">                      <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> </span><span class="dv">20</span>),</a>
<a class="sourceLine" id="cb334-11" data-line-number="11">    <span class="dt">size =</span> <span class="kw">case_when</span>(<span class="kw">str_detect</span>(cell, <span class="st">&quot;gDNA&quot;</span>) <span class="op">~</span><span class="st"> </span><span class="fl">1.2</span>,</a>
<a class="sourceLine" id="cb334-12" data-line-number="12">                     <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> </span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb334-13" data-line-number="13">  )</a>
<a class="sourceLine" id="cb334-14" data-line-number="14"></a>
<a class="sourceLine" id="cb334-15" data-line-number="15">p1 &lt;-<span class="st"> </span>mdamb231_sc_bulk_clustering_umap <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(type)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb334-16" data-line-number="16"><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb334-17" data-line-number="17"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(V1,V2, <span class="dt">color =</span> superclones), <span class="dt">size =</span> <span class="dv">6</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb334-18" data-line-number="18"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(V1,V2, <span class="dt">color =</span> type, <span class="dt">shape =</span> shape, <span class="dt">size =</span> size), <span class="dt">stroke =</span> <span class="fl">1.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb334-19" data-line-number="19"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(colors_vector<span class="op">$</span>superclones, colors_vector<span class="op">$</span>subclones, <span class="st">&quot;bulk&quot;</span> =<span class="st"> &quot;black&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb334-20" data-line-number="20"><span class="st">  </span><span class="kw">scale_shape_identity</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb334-21" data-line-number="21"><span class="st">  </span><span class="kw">scale_size_identity</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb334-22" data-line-number="22"><span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb334-23" data-line-number="23"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.title.x=</span><span class="kw">element_text</span>(<span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>, <span class="dt">size =</span> <span class="dv">20</span>),</a>
<a class="sourceLine" id="cb334-24" data-line-number="24">        <span class="dt">axis.text.x=</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb334-25" data-line-number="25">        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb334-26" data-line-number="26">        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>, <span class="dt">size =</span> <span class="dv">20</span>),</a>
<a class="sourceLine" id="cb334-27" data-line-number="27">        <span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb334-28" data-line-number="28">        <span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb334-29" data-line-number="29">        <span class="dt">axis.line =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb334-30" data-line-number="30">        <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,</a>
<a class="sourceLine" id="cb334-31" data-line-number="31">        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, </a>
<a class="sourceLine" id="cb334-32" data-line-number="32">                                    <span class="dt">fill =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb334-33" data-line-number="33">                                    <span class="dt">size =</span> <span class="dv">2</span>),</a>
<a class="sourceLine" id="cb334-34" data-line-number="34">        <span class="co"># legend.position = &quot;null&quot;</span></a>
<a class="sourceLine" id="cb334-35" data-line-number="35">  )  <span class="op">+</span></a>
<a class="sourceLine" id="cb334-36" data-line-number="36"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">color =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb334-37" data-line-number="37"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;umap1&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb334-38" data-line-number="38"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;umap2&quot;</span>) </a>
<a class="sourceLine" id="cb334-39" data-line-number="39"></a>
<a class="sourceLine" id="cb334-40" data-line-number="40">p1</a></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb335-1" data-line-number="1"><span class="co"># this setup will be used to plot the heatmap later</span></a>
<a class="sourceLine" id="cb335-2" data-line-number="2"></a>
<a class="sourceLine" id="cb335-3" data-line-number="3">mda_ordered &lt;-<span class="st"> </span><span class="kw">order_dataset</span>(<span class="dt">popseg_long =</span> mdamb231_sc_bulk_cna_ml,</a>
<a class="sourceLine" id="cb335-4" data-line-number="4">                             <span class="dt">clustering =</span> mdamb231_sc_bulk_clustering)</a>
<a class="sourceLine" id="cb335-5" data-line-number="5"></a>
<a class="sourceLine" id="cb335-6" data-line-number="6">mda_cocluster_consensus &lt;-<span class="st"> </span><span class="kw">calculate_consensus</span>(<span class="dt">df =</span> mda_ordered<span class="op">$</span>dataset_ordered,</a>
<a class="sourceLine" id="cb335-7" data-line-number="7">                                     <span class="dt">clusters =</span> mda_ordered<span class="op">$</span>clustering_ordered<span class="op">$</span>subclones)</a>
<a class="sourceLine" id="cb335-8" data-line-number="8"></a>
<a class="sourceLine" id="cb335-9" data-line-number="9">mda_cocluster_me_tree &lt;-<span class="st"> </span><span class="kw">run_me_tree</span>(<span class="dt">consensus_df =</span> mda_cocluster_consensus,</a>
<a class="sourceLine" id="cb335-10" data-line-number="10">                                     <span class="dt">clusters =</span> mdamb231_sc_bulk_clustering,</a>
<a class="sourceLine" id="cb335-11" data-line-number="11">                                     <span class="dt">ploidy_VAL =</span> mda_ploidy)</a></code></pre></div>
<p><img src="_main_files/figure-html/more_setup-1.png" width="672" /></p>
</div>
<div id="single-cell-bulk-heatmap" class="section level2">
<h2><span class="header-section-number">8.2</span> Single-cell + Bulk Heatmap</h2>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb336-1" data-line-number="1"><span class="kw">plot_heatmap</span>(<span class="dt">df =</span> mda_ordered<span class="op">$</span>dataset_ordered,</a>
<a class="sourceLine" id="cb336-2" data-line-number="2">             <span class="dt">ploidy_VAL =</span> mda_ploidy,</a>
<a class="sourceLine" id="cb336-3" data-line-number="3">             <span class="dt">ploidy_trunc =</span> <span class="dv">2</span><span class="op">*</span>(<span class="kw">round</span>(mda_ploidy))<span class="op">+</span><span class="dv">2</span>,</a>
<a class="sourceLine" id="cb336-4" data-line-number="4">             <span class="dt">clusters =</span> mda_ordered<span class="op">$</span>clustering_ordered,</a>
<a class="sourceLine" id="cb336-5" data-line-number="5">             <span class="dt">genomic_classes =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb336-6" data-line-number="6">             <span class="dt">keep_gene =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb336-7" data-line-number="7">             <span class="dt">tree_order =</span> mda_cocluster_me_tree<span class="op">$</span>cs_tree_order,</a>
<a class="sourceLine" id="cb336-8" data-line-number="8">             <span class="dt">show_legend =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb336-9" data-line-number="9">             <span class="dt">mda_cocluster =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="_main_files/figure-html/sc_bulk_heat-1.png" width="672" /></p>
</div>
<div id="bulk-heatmap" class="section level2">
<h2><span class="header-section-number">8.3</span> Bulk Heatmap</h2>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb337-1" data-line-number="1">mdamb231_bulk_cna_long &lt;-<span class="st"> </span>mda_ordered<span class="op">$</span>dataset_ordered[<span class="kw">str_detect</span>(<span class="kw">rownames</span>(mda_ordered<span class="op">$</span>dataset_ordered), <span class="st">&quot;gDNA&quot;</span>),]</a>
<a class="sourceLine" id="cb337-2" data-line-number="2"></a>
<a class="sourceLine" id="cb337-3" data-line-number="3"><span class="co"># clustering info</span></a>
<a class="sourceLine" id="cb337-4" data-line-number="4">mdamb231_bulk_cl_info &lt;-<span class="st"> </span>mda_ordered<span class="op">$</span>clustering_ordered <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb337-5" data-line-number="5"><span class="st">  </span><span class="kw">filter</span>(<span class="kw">str_detect</span>(cells, <span class="st">&quot;gDNA&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb337-6" data-line-number="6"><span class="st">  </span><span class="kw">arrange</span>(superclones, subclones)</a>
<a class="sourceLine" id="cb337-7" data-line-number="7"></a>
<a class="sourceLine" id="cb337-8" data-line-number="8"><span class="co"># matching order</span></a>
<a class="sourceLine" id="cb337-9" data-line-number="9">mdamb231_bulk_ordered &lt;-<span class="st"> </span>mdamb231_bulk_cna_long[mdamb231_bulk_cl_info<span class="op">$</span>cells,]</a>
<a class="sourceLine" id="cb337-10" data-line-number="10"></a>
<a class="sourceLine" id="cb337-11" data-line-number="11"><span class="kw">plot_heatmap</span>(<span class="dt">df =</span> mdamb231_bulk_ordered,</a>
<a class="sourceLine" id="cb337-12" data-line-number="12">             <span class="dt">ploidy_VAL =</span> mda_ploidy,</a>
<a class="sourceLine" id="cb337-13" data-line-number="13">             <span class="dt">ploidy_trunc =</span> <span class="dv">2</span><span class="op">*</span>(<span class="kw">round</span>(mda_ploidy))<span class="op">+</span><span class="dv">2</span>,</a>
<a class="sourceLine" id="cb337-14" data-line-number="14">             <span class="dt">clusters =</span> mdamb231_bulk_cl_info,</a>
<a class="sourceLine" id="cb337-15" data-line-number="15">             <span class="dt">genomic_classes =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb337-16" data-line-number="16">             <span class="dt">keep_gene =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb337-17" data-line-number="17">             <span class="dt">tree_order =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb337-18" data-line-number="18">             <span class="dt">show_legend =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb337-19" data-line-number="19">             <span class="dt">mda_cocluster =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="_main_files/figure-html/bulk_heatmap-1.png" width="672" /></p>
</div>
<div id="dna-rna" class="section level2">
<h2><span class="header-section-number">8.4</span> DNA &amp; RNA</h2>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb338-1" data-line-number="1"><span class="co"># Biomart</span></a>
<a class="sourceLine" id="cb338-2" data-line-number="2">grch37 &lt;-</a>
<a class="sourceLine" id="cb338-3" data-line-number="3"><span class="st">  </span><span class="kw">useMart</span>(</a>
<a class="sourceLine" id="cb338-4" data-line-number="4">    <span class="dt">biomart =</span> <span class="st">&quot;ENSEMBL_MART_ENSEMBL&quot;</span>,</a>
<a class="sourceLine" id="cb338-5" data-line-number="5">    <span class="dt">host =</span> <span class="st">&quot;grch37.ensembl.org&quot;</span>,</a>
<a class="sourceLine" id="cb338-6" data-line-number="6">    <span class="dt">path =</span> <span class="st">&quot;/biomart/martservice&quot;</span>,</a>
<a class="sourceLine" id="cb338-7" data-line-number="7">    <span class="dt">dataset =</span> <span class="st">&quot;hsapiens_gene_ensembl&quot;</span></a>
<a class="sourceLine" id="cb338-8" data-line-number="8">  )</a>
<a class="sourceLine" id="cb338-9" data-line-number="9"></a>
<a class="sourceLine" id="cb338-10" data-line-number="10"><span class="co"># hg19 chromosome positions from http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes</span></a>
<a class="sourceLine" id="cb338-11" data-line-number="11">hg19_chrom_sizes &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/lib/hg19.chrom.sizes.rds&quot;</span>))</a>
<a class="sourceLine" id="cb338-12" data-line-number="12"></a>
<a class="sourceLine" id="cb338-13" data-line-number="13"><span class="co"># transforming into a named vector and sorting by chrom</span></a>
<a class="sourceLine" id="cb338-14" data-line-number="14">hg19_chrom_sizes &lt;-<span class="st"> </span>hg19_chrom_sizes[<span class="dv">1</span><span class="op">:</span><span class="dv">23</span>,]</a>
<a class="sourceLine" id="cb338-15" data-line-number="15">hg19_chrom_sizes &lt;-<span class="st"> </span><span class="kw">deframe</span>(hg19_chrom_sizes)</a>
<a class="sourceLine" id="cb338-16" data-line-number="16">hg19_chrom_sizes &lt;-<span class="st"> </span>hg19_chrom_sizes[gtools<span class="op">::</span><span class="kw">mixedsort</span>(<span class="kw">names</span>(hg19_chrom_sizes))]</a>
<a class="sourceLine" id="cb338-17" data-line-number="17"></a>
<a class="sourceLine" id="cb338-18" data-line-number="18"><span class="co"># obtaining the position for every ensembl gene id </span></a>
<a class="sourceLine" id="cb338-19" data-line-number="19">bms &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb338-20" data-line-number="20"></a>
<a class="sourceLine" id="cb338-21" data-line-number="21"><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(hg19_chrom_sizes)) {</a>
<a class="sourceLine" id="cb338-22" data-line-number="22">  chrom &lt;-<span class="st"> </span><span class="kw">str_remove</span>(<span class="kw">names</span>(hg19_chrom_sizes)[i], <span class="st">&quot;chr&quot;</span>)</a>
<a class="sourceLine" id="cb338-23" data-line-number="23">  </a>
<a class="sourceLine" id="cb338-24" data-line-number="24">  <span class="cf">if</span> (chrom <span class="op">!=</span><span class="st"> &quot;X&quot;</span>)</a>
<a class="sourceLine" id="cb338-25" data-line-number="25">    chrom &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(chrom)</a>
<a class="sourceLine" id="cb338-26" data-line-number="26">  </a>
<a class="sourceLine" id="cb338-27" data-line-number="27">  length_chrom &lt;-<span class="st"> </span>hg19_chrom_sizes[i]</a>
<a class="sourceLine" id="cb338-28" data-line-number="28">  </a>
<a class="sourceLine" id="cb338-29" data-line-number="29">  bms[[i]] &lt;-</a>
<a class="sourceLine" id="cb338-30" data-line-number="30"><span class="st">    </span><span class="kw">getBM</span>(</a>
<a class="sourceLine" id="cb338-31" data-line-number="31">      <span class="kw">c</span>(</a>
<a class="sourceLine" id="cb338-32" data-line-number="32">        <span class="st">&quot;ensembl_gene_id&quot;</span>,</a>
<a class="sourceLine" id="cb338-33" data-line-number="33">        <span class="st">&quot;hgnc_symbol&quot;</span>,</a>
<a class="sourceLine" id="cb338-34" data-line-number="34">        <span class="st">&quot;start_position&quot;</span>,</a>
<a class="sourceLine" id="cb338-35" data-line-number="35">        <span class="st">&quot;end_position&quot;</span></a>
<a class="sourceLine" id="cb338-36" data-line-number="36">      ),</a>
<a class="sourceLine" id="cb338-37" data-line-number="37">      <span class="dt">filters =</span> <span class="kw">c</span>(<span class="st">&quot;chromosome_name&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>),</a>
<a class="sourceLine" id="cb338-38" data-line-number="38">      <span class="dt">values =</span> <span class="kw">list</span>(chrom,</a>
<a class="sourceLine" id="cb338-39" data-line-number="39">                    <span class="dv">0</span>, length_chrom),</a>
<a class="sourceLine" id="cb338-40" data-line-number="40">      <span class="dt">mart =</span> grch37</a>
<a class="sourceLine" id="cb338-41" data-line-number="41">    )</a>
<a class="sourceLine" id="cb338-42" data-line-number="42">  </a>
<a class="sourceLine" id="cb338-43" data-line-number="43">  bms[[i]] &lt;-<span class="st"> </span>bms[[i]] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(start_position)</a>
<a class="sourceLine" id="cb338-44" data-line-number="44">  </a>
<a class="sourceLine" id="cb338-45" data-line-number="45">  <span class="kw">names</span>(bms)[[i]] &lt;-<span class="st"> </span><span class="kw">names</span>(hg19_chrom_sizes)[i]</a>
<a class="sourceLine" id="cb338-46" data-line-number="46">}</a>
<a class="sourceLine" id="cb338-47" data-line-number="47"></a>
<a class="sourceLine" id="cb338-48" data-line-number="48">bms_df &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(bms, <span class="dt">.id =</span> <span class="st">&quot;chr&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb338-49" data-line-number="49"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">gene =</span> <span class="st">&quot;hgnc_symbol&quot;</span>,</a>
<a class="sourceLine" id="cb338-50" data-line-number="50">                <span class="dt">gene_id =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>)</a>
<a class="sourceLine" id="cb338-51" data-line-number="51"></a>
<a class="sourceLine" id="cb338-52" data-line-number="52"><span class="co"># obtaining the count matrix of the MDAMB231 bulk RNAseq</span></a>
<a class="sourceLine" id="cb338-53" data-line-number="53">cnt &lt;-<span class="st"> </span><span class="kw">assay</span>(vsd)</a>
<a class="sourceLine" id="cb338-54" data-line-number="54"></a>
<a class="sourceLine" id="cb338-55" data-line-number="55"><span class="co"># gene id clean up</span></a>
<a class="sourceLine" id="cb338-56" data-line-number="56"><span class="kw">rownames</span>(cnt) &lt;-<span class="st"> </span><span class="kw">str_extract</span>(<span class="kw">rownames</span>(cnt), <span class="st">&quot;[A-Z]+[0-9]+&quot;</span>)</a>
<a class="sourceLine" id="cb338-57" data-line-number="57"></a>
<a class="sourceLine" id="cb338-58" data-line-number="58"><span class="co"># averaging the expression of the RNA triplicates and calculating a zscore</span></a>
<a class="sourceLine" id="cb338-59" data-line-number="59">cnt_long &lt;-<span class="st"> </span>cnt <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb338-60" data-line-number="60"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb338-61" data-line-number="61"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;gene_id&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb338-62" data-line-number="62"><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">gather</span>(<span class="dt">key =</span> <span class="st">&quot;sample&quot;</span>,</a>
<a class="sourceLine" id="cb338-63" data-line-number="63">                <span class="dt">value =</span> <span class="st">&quot;count&quot;</span>,</a>
<a class="sourceLine" id="cb338-64" data-line-number="64">                <span class="op">-</span>gene_id) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb338-65" data-line-number="65"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">str_remove</span>(sample, <span class="st">&quot;[A-C]$&quot;</span>))</a>
<a class="sourceLine" id="cb338-66" data-line-number="66"></a>
<a class="sourceLine" id="cb338-67" data-line-number="67">cnt_long_avg &lt;-<span class="st"> </span>cnt_long <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb338-68" data-line-number="68"><span class="st">  </span><span class="kw">group_by</span>(gene_id, sample) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb338-69" data-line-number="69"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">count_avg =</span> <span class="kw">mean</span>(count)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb338-70" data-line-number="70"><span class="st">  </span><span class="kw">ungroup</span>()  <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb338-71" data-line-number="71"><span class="st">  </span><span class="kw">group_by</span>(gene_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb338-72" data-line-number="72"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">z_score =</span> (count_avg<span class="op">-</span><span class="kw">mean</span>(count_avg))<span class="op">/</span><span class="kw">sd</span>(count_avg)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb338-73" data-line-number="73"><span class="st">  </span><span class="kw">ungroup</span>() </a>
<a class="sourceLine" id="cb338-74" data-line-number="74"></a>
<a class="sourceLine" id="cb338-75" data-line-number="75"><span class="co"># wide df will be used later for moving average</span></a>
<a class="sourceLine" id="cb338-76" data-line-number="76">cnt_avg &lt;-<span class="st"> </span>tidyr<span class="op">::</span><span class="kw">pivot_wider</span>(cnt_long_avg,</a>
<a class="sourceLine" id="cb338-77" data-line-number="77">                              <span class="dt">names_from =</span> sample,</a>
<a class="sourceLine" id="cb338-78" data-line-number="78">                              <span class="dt">values_from =</span> z_score,</a>
<a class="sourceLine" id="cb338-79" data-line-number="79">                              <span class="dt">id_cols =</span> gene_id)</a>
<a class="sourceLine" id="cb338-80" data-line-number="80"></a>
<a class="sourceLine" id="cb338-81" data-line-number="81"></a>
<a class="sourceLine" id="cb338-82" data-line-number="82">cnt_avg &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(cnt_avg)</a>
<a class="sourceLine" id="cb338-83" data-line-number="83"><span class="kw">rownames</span>(cnt_avg) &lt;-<span class="st"> </span>cnt_avg<span class="op">$</span>gene_id</a>
<a class="sourceLine" id="cb338-84" data-line-number="84">cnt_avg &lt;-<span class="st"> </span>cnt_avg <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb338-85" data-line-number="85"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>gene_id)</a></code></pre></div>
</div>
<div id="moving-averages" class="section level2">
<h2><span class="header-section-number">8.5</span> Moving averages</h2>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb339-1" data-line-number="1"><span class="co"># adding segments from population segmentation information</span></a>
<a class="sourceLine" id="cb339-2" data-line-number="2">mdamb231_bulk_cna_id &lt;-<span class="st"> </span>mdamb231_sc_bulk_cna_popseg <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-3" data-line-number="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">seg_index =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(mdamb231_sc_bulk_cna_popseg))</a>
<a class="sourceLine" id="cb339-4" data-line-number="4"></a>
<a class="sourceLine" id="cb339-5" data-line-number="5"><span class="co"># clustering information</span></a>
<a class="sourceLine" id="cb339-6" data-line-number="6">cl_info &lt;-<span class="st"> </span>mdamb231_bulk_cl_info <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-7" data-line-number="7"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span><span class="kw">str_remove</span>(cells, <span class="st">&quot;gDNA&quot;</span>),</a>
<a class="sourceLine" id="cb339-8" data-line-number="8">         <span class="dt">sample =</span> <span class="kw">str_extract</span>(sample, <span class="st">&quot;MDAMB231C[0-9]+&quot;</span>))</a>
<a class="sourceLine" id="cb339-9" data-line-number="9"></a>
<a class="sourceLine" id="cb339-10" data-line-number="10">seg_ids_long &lt;-<span class="st"> </span><span class="kw">rep.int</span>(mdamb231_bulk_cna_id<span class="op">$</span>seg_index, mdamb231_sc_bulk_cna_popseg<span class="op">$</span>n.probes)</a>
<a class="sourceLine" id="cb339-11" data-line-number="11"></a>
<a class="sourceLine" id="cb339-12" data-line-number="12"><span class="co"># bulk integer df</span></a>
<a class="sourceLine" id="cb339-13" data-line-number="13">mdamb231_bulk_cna_long_int &lt;-<span class="st"> </span><span class="kw">ploidy_scale</span>(<span class="dt">ploidy_VAL =</span> <span class="fl">2.41</span>,</a>
<a class="sourceLine" id="cb339-14" data-line-number="14">                                           <span class="dt">df =</span> mdamb231_bulk_ordered)</a>
<a class="sourceLine" id="cb339-15" data-line-number="15"></a>
<a class="sourceLine" id="cb339-16" data-line-number="16"><span class="kw">rownames</span>(mdamb231_bulk_cna_long_int) &lt;-<span class="st"> </span><span class="kw">str_remove</span>(<span class="kw">rownames</span>(mdamb231_bulk_cna_long_int), <span class="st">&quot;gDNA&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-17" data-line-number="17"><span class="st">  </span><span class="kw">str_remove</span>(<span class="st">&quot;_S[0-9]+&quot;</span>)</a>
<a class="sourceLine" id="cb339-18" data-line-number="18"></a>
<a class="sourceLine" id="cb339-19" data-line-number="19">blk_long &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">t</span>(mdamb231_bulk_cna_long_int))</a>
<a class="sourceLine" id="cb339-20" data-line-number="20"></a>
<a class="sourceLine" id="cb339-21" data-line-number="21"><span class="co"># adding pipeline information</span></a>
<a class="sourceLine" id="cb339-22" data-line-number="22">blk_long<span class="op">$</span>chr &lt;-<span class="st"> </span>bins_in_cna_pipeline<span class="op">$</span>chr[<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(blk_long)]</a>
<a class="sourceLine" id="cb339-23" data-line-number="23">blk_long<span class="op">$</span>start &lt;-<span class="st"> </span>bins_in_cna_pipeline<span class="op">$</span>start[<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(blk_long)]</a>
<a class="sourceLine" id="cb339-24" data-line-number="24">blk_long<span class="op">$</span>end &lt;-<span class="st"> </span>bins_in_cna_pipeline<span class="op">$</span>end[<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(blk_long)]</a>
<a class="sourceLine" id="cb339-25" data-line-number="25">blk_long<span class="op">$</span>abspos &lt;-<span class="st"> </span>bins_in_cna_pipeline<span class="op">$</span>abspos[<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(blk_long)]</a>
<a class="sourceLine" id="cb339-26" data-line-number="26">blk_long<span class="op">$</span>seg_index &lt;-<span class="st"> </span>seg_ids_long</a>
<a class="sourceLine" id="cb339-27" data-line-number="27"></a>
<a class="sourceLine" id="cb339-28" data-line-number="28"><span class="co"># creating a long file with a position vector according to the number of rows</span></a>
<a class="sourceLine" id="cb339-29" data-line-number="29">blk_long_g &lt;-<span class="st"> </span>blk_long <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-30" data-line-number="30"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">pos =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(blk_long)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-31" data-line-number="31"><span class="st">  </span><span class="kw">gather</span>(<span class="dt">key =</span> <span class="st">&quot;sample&quot;</span>,</a>
<a class="sourceLine" id="cb339-32" data-line-number="32">         <span class="dt">value =</span> <span class="st">&quot;cn&quot;</span>,</a>
<a class="sourceLine" id="cb339-33" data-line-number="33">         <span class="op">-</span><span class="kw">c</span>(chr, start, end, abspos, pos, seg_index))</a>
<a class="sourceLine" id="cb339-34" data-line-number="34"></a>
<a class="sourceLine" id="cb339-35" data-line-number="35"><span class="co"># obataining gene positions Grange</span></a>
<a class="sourceLine" id="cb339-36" data-line-number="36">txdb &lt;-<span class="st"> </span>Homo.sapiens</a>
<a class="sourceLine" id="cb339-37" data-line-number="37"></a>
<a class="sourceLine" id="cb339-38" data-line-number="38">hg19_genes &lt;-<span class="st"> </span>GenomicFeatures<span class="op">::</span><span class="kw">genes</span>(txdb, <span class="dt">columns =</span> <span class="st">&quot;SYMBOL&quot;</span>)</a>
<a class="sourceLine" id="cb339-39" data-line-number="39">hg19_genes_df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(hg19_genes) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-40" data-line-number="40"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">SYMBOL =</span> <span class="kw">as.character</span>(SYMBOL))</a>
<a class="sourceLine" id="cb339-41" data-line-number="41"></a>
<a class="sourceLine" id="cb339-42" data-line-number="42">bins_gr &lt;-<span class="st"> </span>bins_in_cna_pipeline  <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-43" data-line-number="43"><span class="st">  </span><span class="kw">makeGRangesFromDataFrame</span>(<span class="dt">keep.extra.columns =</span> T, <span class="dt">ignore.strand =</span> T)  </a>
<a class="sourceLine" id="cb339-44" data-line-number="44"></a>
<a class="sourceLine" id="cb339-45" data-line-number="45">olaps &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(hg19_genes, bins_gr)</a>
<a class="sourceLine" id="cb339-46" data-line-number="46"></a>
<a class="sourceLine" id="cb339-47" data-line-number="47"><span class="co"># finding which index (bin in the 200kb pipeline the gene is located) </span></a>
<a class="sourceLine" id="cb339-48" data-line-number="48"><span class="co"># saving to a data frame</span></a>
<a class="sourceLine" id="cb339-49" data-line-number="49">mk_df &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">gene =</span> hg19_genes_df<span class="op">$</span>SYMBOL[<span class="kw">queryHits</span>(olaps)],</a>
<a class="sourceLine" id="cb339-50" data-line-number="50">                <span class="dt">pos =</span> <span class="kw">subjectHits</span>(olaps)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-51" data-line-number="51"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">distinct</span>(gene, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb339-52" data-line-number="52"></a>
<a class="sourceLine" id="cb339-53" data-line-number="53">mk_df_j &lt;-<span class="st"> </span><span class="kw">inner_join</span>(mk_df, bms_df) </a>
<a class="sourceLine" id="cb339-54" data-line-number="54"></a>
<a class="sourceLine" id="cb339-55" data-line-number="55"><span class="co"># all triplicates in long form</span></a>
<a class="sourceLine" id="cb339-56" data-line-number="56">cnt_long_trip &lt;-<span class="st"> </span>cnt <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-57" data-line-number="57"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-58" data-line-number="58"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;gene_id&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-59" data-line-number="59"><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">gather</span>(<span class="dt">key =</span> <span class="st">&quot;sample_trip&quot;</span>,</a>
<a class="sourceLine" id="cb339-60" data-line-number="60">                <span class="dt">value =</span> <span class="st">&quot;count&quot;</span>,</a>
<a class="sourceLine" id="cb339-61" data-line-number="61">                <span class="op">-</span>gene_id) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-62" data-line-number="62"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample=</span> <span class="kw">str_remove</span>(sample_trip, <span class="st">&quot;[A-C]$&quot;</span>))</a>
<a class="sourceLine" id="cb339-63" data-line-number="63"></a>
<a class="sourceLine" id="cb339-64" data-line-number="64">cnt_long_all_zc &lt;-<span class="st"> </span>cnt_long_trip <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb339-65" data-line-number="65"><span class="st">  </span><span class="kw">group_by</span>(gene_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb339-66" data-line-number="66"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">z_score =</span> (count<span class="op">-</span><span class="kw">mean</span>(count))<span class="op">/</span><span class="kw">sd</span>(count)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb339-67" data-line-number="67"><span class="st">  </span><span class="kw">ungroup</span>()</a>
<a class="sourceLine" id="cb339-68" data-line-number="68"></a>
<a class="sourceLine" id="cb339-69" data-line-number="69">cnt_long_gene_trip &lt;-<span class="st"> </span><span class="kw">inner_join</span>(cnt_long_all_zc, mk_df_j)</a>
<a class="sourceLine" id="cb339-70" data-line-number="70"></a>
<a class="sourceLine" id="cb339-71" data-line-number="71">cnt_long_gene_trip &lt;-<span class="st"> </span><span class="kw">inner_join</span>(cnt_long_gene_trip, cl_info) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-72" data-line-number="72"><span class="st">  </span><span class="kw">arrange</span>(chr, start_position)</a>
<a class="sourceLine" id="cb339-73" data-line-number="73"></a>
<a class="sourceLine" id="cb339-74" data-line-number="74"><span class="co"># joining the copy number long data with the gene positions on their respective bins</span></a>
<a class="sourceLine" id="cb339-75" data-line-number="75">blk_long_gj &lt;-<span class="st"> </span><span class="kw">inner_join</span>(blk_long_g, mk_df)</a>
<a class="sourceLine" id="cb339-76" data-line-number="76"></a>
<a class="sourceLine" id="cb339-77" data-line-number="77"><span class="co"># adding the ensembl gene id information and the cluster information</span></a>
<a class="sourceLine" id="cb339-78" data-line-number="78">blk_long_gj &lt;-<span class="st"> </span><span class="kw">inner_join</span>(blk_long_gj, bms_df) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-79" data-line-number="79"><span class="st">  </span><span class="kw">arrange</span>(pos) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-80" data-line-number="80"><span class="st">  </span><span class="kw">left_join</span>(cl_info)</a>
<a class="sourceLine" id="cb339-81" data-line-number="81"></a>
<a class="sourceLine" id="cb339-82" data-line-number="82"><span class="co"># full dna rna table with counts merged by sample and ensemblgene_id</span></a>
<a class="sourceLine" id="cb339-83" data-line-number="83">dna_rna &lt;-<span class="st"> </span><span class="kw">inner_join</span>(blk_long_gj,</a>
<a class="sourceLine" id="cb339-84" data-line-number="84">                      cnt_long_avg)</a>
<a class="sourceLine" id="cb339-85" data-line-number="85"></a>
<a class="sourceLine" id="cb339-86" data-line-number="86"></a>
<a class="sourceLine" id="cb339-87" data-line-number="87"><span class="co"># jitter to be able to visualize all the copy number tracks</span></a>
<a class="sourceLine" id="cb339-88" data-line-number="88">jitter &lt;-<span class="st"> </span><span class="kw">seq</span>(<span class="op">-</span><span class="fl">0.14</span>, <span class="fl">0.17</span>, <span class="fl">0.03</span>) </a>
<a class="sourceLine" id="cb339-89" data-line-number="89"></a>
<a class="sourceLine" id="cb339-90" data-line-number="90">cls &lt;-<span class="st"> </span>gtools<span class="op">::</span><span class="kw">mixedsort</span>(<span class="kw">unique</span>(<span class="kw">levels</span>(<span class="kw">droplevels</span>(<span class="kw">as.factor</span>(blk_long_gj<span class="op">$</span>subclones)))))</a>
<a class="sourceLine" id="cb339-91" data-line-number="91"></a>
<a class="sourceLine" id="cb339-92" data-line-number="92">blk_long_gj_jit &lt;-<span class="st"> </span>blk_long_gj <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb339-93" data-line-number="93"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">cn =</span> <span class="kw">case_when</span>(</a>
<a class="sourceLine" id="cb339-94" data-line-number="94">    subclones <span class="op">==</span><span class="st"> </span>cls[<span class="dv">1</span>] <span class="op">~</span><span class="st"> </span>cn <span class="op">+</span><span class="st"> </span>jitter[<span class="dv">1</span>],</a>
<a class="sourceLine" id="cb339-95" data-line-number="95">    subclones <span class="op">==</span><span class="st"> </span>cls[<span class="dv">2</span>] <span class="op">~</span><span class="st"> </span>cn <span class="op">+</span><span class="st"> </span>jitter[<span class="dv">2</span>],</a>
<a class="sourceLine" id="cb339-96" data-line-number="96">    subclones <span class="op">==</span><span class="st"> </span>cls[<span class="dv">3</span>] <span class="op">~</span><span class="st"> </span>cn <span class="op">+</span><span class="st"> </span>jitter[<span class="dv">3</span>],</a>
<a class="sourceLine" id="cb339-97" data-line-number="97">    subclones <span class="op">==</span><span class="st"> </span>cls[<span class="dv">4</span>] <span class="op">~</span><span class="st"> </span>cn <span class="op">+</span><span class="st"> </span>jitter[<span class="dv">4</span>],</a>
<a class="sourceLine" id="cb339-98" data-line-number="98">    subclones <span class="op">==</span><span class="st"> </span>cls[<span class="dv">5</span>] <span class="op">~</span><span class="st"> </span>cn <span class="op">+</span><span class="st"> </span>jitter[<span class="dv">5</span>],</a>
<a class="sourceLine" id="cb339-99" data-line-number="99">    subclones <span class="op">==</span><span class="st"> </span>cls[<span class="dv">6</span>] <span class="op">~</span><span class="st"> </span>cn <span class="op">+</span><span class="st"> </span>jitter[<span class="dv">6</span>],</a>
<a class="sourceLine" id="cb339-100" data-line-number="100">    subclones <span class="op">==</span><span class="st"> </span>cls[<span class="dv">7</span>] <span class="op">~</span><span class="st"> </span>cn <span class="op">+</span><span class="st"> </span>jitter[<span class="dv">7</span>],</a>
<a class="sourceLine" id="cb339-101" data-line-number="101">    subclones <span class="op">==</span><span class="st"> </span>cls[<span class="dv">8</span>] <span class="op">~</span><span class="st"> </span>cn <span class="op">+</span><span class="st"> </span>jitter[<span class="dv">8</span>],</a>
<a class="sourceLine" id="cb339-102" data-line-number="102">    subclones <span class="op">==</span><span class="st"> </span>cls[<span class="dv">9</span>] <span class="op">~</span><span class="st"> </span>cn <span class="op">+</span><span class="st"> </span>jitter[<span class="dv">9</span>],</a>
<a class="sourceLine" id="cb339-103" data-line-number="103">    subclones <span class="op">==</span><span class="st"> </span>cls[<span class="dv">10</span>] <span class="op">~</span><span class="st"> </span>cn <span class="op">+</span><span class="st"> </span>jitter[<span class="dv">10</span>]</a>
<a class="sourceLine" id="cb339-104" data-line-number="104">  ))</a>
<a class="sourceLine" id="cb339-105" data-line-number="105"></a>
<a class="sourceLine" id="cb339-106" data-line-number="106"><span class="kw">plot_moving_average</span>(<span class="dt">chr =</span> <span class="st">&quot;chr19&quot;</span>,</a>
<a class="sourceLine" id="cb339-107" data-line-number="107">                    <span class="dt">genes =</span> <span class="kw">c</span>(<span class="st">&quot;SH3GL1&quot;</span>,<span class="st">&quot;PIK3R2&quot;</span>))</a></code></pre></div>
<p><img src="_main_files/figure-html/moving_averages-1.png" width="672" /></p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb340-1" data-line-number="1"><span class="kw">plot_moving_average</span>(<span class="dt">chr =</span> <span class="st">&quot;chr11&quot;</span>,</a>
<a class="sourceLine" id="cb340-2" data-line-number="2">                    <span class="dt">genes =</span> <span class="kw">c</span>(<span class="st">&quot;ATM&quot;</span>,<span class="st">&quot;CBL&quot;</span>))</a></code></pre></div>
<p><img src="_main_files/figure-html/moving_averages-2.png" width="672" /></p>
</div>
<div id="de" class="section level2">
<h2><span class="header-section-number">8.6</span> DE</h2>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb341-1" data-line-number="1">samples &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(dds)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb341-2" data-line-number="2"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample_name =</span> <span class="kw">str_extract</span>(sample, <span class="st">&quot;MDAMB231C[0-9]+&quot;</span>))</a>
<a class="sourceLine" id="cb341-3" data-line-number="3"></a>
<a class="sourceLine" id="cb341-4" data-line-number="4">samples_cl &lt;-<span class="st"> </span><span class="kw">left_join</span>(samples, cl_info, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;sample_name&quot;</span> =<span class="st"> &quot;sample&quot;</span>))</a>
<a class="sourceLine" id="cb341-5" data-line-number="5"><span class="co">#sanity check</span></a>
<a class="sourceLine" id="cb341-6" data-line-number="6">samples_cl &lt;-<span class="st"> </span>samples_cl[<span class="kw">match</span>(samples_cl<span class="op">$</span>sample, <span class="kw">colData</span>(dds)<span class="op">$</span>sample),]</a>
<a class="sourceLine" id="cb341-7" data-line-number="7"></a>
<a class="sourceLine" id="cb341-8" data-line-number="8"><span class="co"># removing unused levels</span></a>
<a class="sourceLine" id="cb341-9" data-line-number="9"><span class="kw">colData</span>(dds)<span class="op">$</span>cluster &lt;-<span class="st"> </span><span class="kw">droplevels</span>(<span class="kw">as.factor</span>(samples_cl<span class="op">$</span>subclones))</a>
<a class="sourceLine" id="cb341-10" data-line-number="10"></a>
<a class="sourceLine" id="cb341-11" data-line-number="11"><span class="kw">design</span>(dds) &lt;-<span class="st"> </span><span class="kw">formula</span>(<span class="op">~</span><span class="st"> </span>cluster)</a>
<a class="sourceLine" id="cb341-12" data-line-number="12"><span class="kw">register</span>(<span class="kw">BatchtoolsParam</span>(<span class="dt">workers =</span> <span class="dv">60</span>), <span class="dt">default =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb341-13" data-line-number="13">dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds, <span class="dt">parallel =</span> <span class="ot">TRUE</span>, <span class="dt">betaPrior =</span> T)</a></code></pre></div>
<pre><code>## using pre-existing normalization factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 60 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, MLE betas: 60 workers</code></pre>
<pre><code>## fitting model and testing: 60 workers</code></pre>
<pre><code>## -- replacing outliers and refitting for 7 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb351-1" data-line-number="1"><span class="kw">resultsNames</span>(dds)</a></code></pre></div>
<pre><code>##  [1] &quot;Intercept&quot;  &quot;clusterc11&quot; &quot;clusterc12&quot; &quot;clusterc13&quot; &quot;clusterc5&quot;  &quot;clusterc6&quot;  &quot;clusterc8&quot;  &quot;clusterc3&quot;
##  [9] &quot;clusterc4&quot;  &quot;clusterc1&quot;  &quot;clusterc2&quot;</code></pre>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb353-1" data-line-number="1">clusters &lt;-<span class="st"> </span>samples_cl <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb353-2" data-line-number="2"><span class="st">  </span><span class="kw">droplevels</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb353-3" data-line-number="3"><span class="st">  </span><span class="kw">distinct</span>(subclones) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb353-4" data-line-number="4"><span class="st">  </span><span class="kw">pull</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb353-5" data-line-number="5"><span class="st">  </span><span class="kw">as.character</span>()</a></code></pre></div>
</div>
<div id="rna-pca" class="section level2">
<h2><span class="header-section-number">8.7</span> RNA PCA</h2>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb354-1" data-line-number="1">pcaData &lt;-<span class="st"> </span><span class="kw">plotPCA</span>(<span class="kw">vst</span>(dds), <span class="st">&quot;cluster&quot;</span>, <span class="dt">returnData =</span> T) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb354-2" data-line-number="2"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">triplicate =</span> <span class="kw">str_extract</span>(name, <span class="st">&quot;C[0-9]+&quot;</span>),</a>
<a class="sourceLine" id="cb354-3" data-line-number="3">         <span class="dt">triplicate =</span> <span class="kw">tolower</span>(triplicate)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb354-4" data-line-number="4"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">str_extract</span>(name, <span class="st">&quot;MDAMB231C[0-9]+&quot;</span>))</a>
<a class="sourceLine" id="cb354-5" data-line-number="5">percentVar &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="dv">100</span> <span class="op">*</span><span class="st"> </span><span class="kw">attr</span>(pcaData, <span class="st">&quot;percentVar&quot;</span>))</a>
<a class="sourceLine" id="cb354-6" data-line-number="6"></a>
<a class="sourceLine" id="cb354-7" data-line-number="7">cluster_anno &lt;-<span class="st"> </span>mdamb231_sc_bulk_clustering <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb354-8" data-line-number="8"><span class="st">  </span><span class="kw">filter</span>(<span class="kw">str_detect</span>(cells, <span class="st">&quot;gDNA&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb354-9" data-line-number="9"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">str_remove</span>(cells, <span class="st">&quot;gDNA&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb354-10" data-line-number="10"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">str_extract</span>(sample, <span class="st">&quot;MDAMB231C[0-9]+&quot;</span>))</a>
<a class="sourceLine" id="cb354-11" data-line-number="11"></a>
<a class="sourceLine" id="cb354-12" data-line-number="12">pcaData &lt;-<span class="st"> </span>pcaData <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb354-13" data-line-number="13"><span class="st">  </span><span class="kw">left_join</span>(cluster_anno, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;group&quot;</span> =<span class="st"> &quot;subclones&quot;</span>, </a>
<a class="sourceLine" id="cb354-14" data-line-number="14">                                 <span class="st">&quot;sample&quot;</span> =<span class="st"> &quot;sample&quot;</span>))</a>
<a class="sourceLine" id="cb354-15" data-line-number="15"></a>
<a class="sourceLine" id="cb354-16" data-line-number="16">p_pca &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pcaData, <span class="kw">aes</span>(<span class="dt">x =</span> PC1,</a>
<a class="sourceLine" id="cb354-17" data-line-number="17">                             <span class="dt">y =</span> PC2)) <span class="op">+</span></a>
<a class="sourceLine" id="cb354-18" data-line-number="18"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(colors_vector<span class="op">$</span>subclones,</a>
<a class="sourceLine" id="cb354-19" data-line-number="19">                                colors_vector<span class="op">$</span>superclones)) <span class="op">+</span></a>
<a class="sourceLine" id="cb354-20" data-line-number="20"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">color =</span> <span class="kw">fct_relevel</span>(superclones, gtools<span class="op">::</span><span class="kw">mixedsort</span>(<span class="kw">unique</span>(<span class="kw">as.character</span>(pcaData<span class="op">$</span>superclones))))), <span class="dt">size =</span> <span class="dv">8</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb354-21" data-line-number="21"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">color =</span> <span class="kw">fct_relevel</span>(group, gtools<span class="op">::</span><span class="kw">mixedsort</span>(<span class="kw">unique</span>(<span class="kw">as.character</span>(pcaData<span class="op">$</span>group))))), <span class="dt">size =</span> <span class="dv">3</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb354-22" data-line-number="22"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;PC1&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb354-23" data-line-number="23"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;PC2&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb354-24" data-line-number="24"><span class="st">  </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb354-25" data-line-number="25"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.title.x=</span><span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">16</span>),</a>
<a class="sourceLine" id="cb354-26" data-line-number="26">        <span class="dt">axis.text.x=</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb354-27" data-line-number="27">        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb354-28" data-line-number="28">        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">16</span>),</a>
<a class="sourceLine" id="cb354-29" data-line-number="29">        <span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb354-30" data-line-number="30">        <span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb354-31" data-line-number="31">        <span class="dt">axis.line =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb354-32" data-line-number="32">        <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>),</a>
<a class="sourceLine" id="cb354-33" data-line-number="33">        <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,</a>
<a class="sourceLine" id="cb354-34" data-line-number="34">        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">color =</span> <span class="st">&quot;black&quot;</span>,</a>
<a class="sourceLine" id="cb354-35" data-line-number="35">                                    <span class="dt">fill =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb354-36" data-line-number="36">                                    <span class="dt">size =</span> <span class="dv">2</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb354-37" data-line-number="37"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">color =</span> <span class="st">&quot;subclone&quot;</span>)</a></code></pre></div>
</div>
<div id="fgsea" class="section level2">
<h2><span class="header-section-number">8.8</span> FGSEA</h2>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb355-1" data-line-number="1"><span class="co"># Thanks to https://stephenturner.github.io/deseq-to-fgsea/</span></a>
<a class="sourceLine" id="cb355-2" data-line-number="2"></a>
<a class="sourceLine" id="cb355-3" data-line-number="3">fgseaResTidy &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb355-4" data-line-number="4"></a>
<a class="sourceLine" id="cb355-5" data-line-number="5"><span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(clusters)) {</a>
<a class="sourceLine" id="cb355-6" data-line-number="6">  res &lt;-</a>
<a class="sourceLine" id="cb355-7" data-line-number="7"><span class="st">    </span><span class="kw">results</span>(dds,</a>
<a class="sourceLine" id="cb355-8" data-line-number="8">            <span class="dt">contrast =</span> <span class="kw">list</span>(</a>
<a class="sourceLine" id="cb355-9" data-line-number="9">              <span class="kw">paste0</span>(<span class="st">&quot;cluster&quot;</span>, clusters[x]),</a>
<a class="sourceLine" id="cb355-10" data-line-number="10">              <span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb355-11" data-line-number="11">                <span class="st">&quot;cluster&quot;</span>,</a>
<a class="sourceLine" id="cb355-12" data-line-number="12">                clusters[clusters <span class="op">%!in%</span><span class="st"> </span>clusters[x]]</a>
<a class="sourceLine" id="cb355-13" data-line-number="13">              )</a>
<a class="sourceLine" id="cb355-14" data-line-number="14">            ),</a>
<a class="sourceLine" id="cb355-15" data-line-number="15">            <span class="dt">listValues =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">-1</span> <span class="op">/</span><span class="st"> </span><span class="dv">10</span>))</a>
<a class="sourceLine" id="cb355-16" data-line-number="16">  </a>
<a class="sourceLine" id="cb355-17" data-line-number="17">  <span class="kw">rownames</span>(res) &lt;-<span class="st"> </span><span class="kw">str_replace</span>(<span class="kw">rownames</span>(res),</a>
<a class="sourceLine" id="cb355-18" data-line-number="18">                               <span class="st">&quot;.[0-9]+$&quot;</span>, <span class="st">&quot;&quot;</span>)</a>
<a class="sourceLine" id="cb355-19" data-line-number="19">  </a>
<a class="sourceLine" id="cb355-20" data-line-number="20"></a>
<a class="sourceLine" id="cb355-21" data-line-number="21">  res<span class="op">$</span>symbol &lt;-<span class="st"> </span><span class="kw">mapIds</span>(</a>
<a class="sourceLine" id="cb355-22" data-line-number="22">    org.Hs.eg.db,</a>
<a class="sourceLine" id="cb355-23" data-line-number="23">    <span class="dt">keys =</span> <span class="kw">row.names</span>(res),</a>
<a class="sourceLine" id="cb355-24" data-line-number="24">    <span class="dt">column =</span> <span class="st">&quot;SYMBOL&quot;</span>,</a>
<a class="sourceLine" id="cb355-25" data-line-number="25">    <span class="dt">keytype =</span> <span class="st">&quot;ENSEMBL&quot;</span>,</a>
<a class="sourceLine" id="cb355-26" data-line-number="26">    <span class="dt">multiVals =</span> <span class="st">&quot;first&quot;</span></a>
<a class="sourceLine" id="cb355-27" data-line-number="27">  )</a>
<a class="sourceLine" id="cb355-28" data-line-number="28">  </a>
<a class="sourceLine" id="cb355-29" data-line-number="29">  res<span class="op">$</span>map &lt;-<span class="st"> </span><span class="kw">mapIds</span>(</a>
<a class="sourceLine" id="cb355-30" data-line-number="30">    org.Hs.eg.db,</a>
<a class="sourceLine" id="cb355-31" data-line-number="31">    <span class="dt">keys =</span> <span class="kw">row.names</span>(res),</a>
<a class="sourceLine" id="cb355-32" data-line-number="32">    <span class="dt">column =</span> <span class="st">&quot;MAP&quot;</span>,</a>
<a class="sourceLine" id="cb355-33" data-line-number="33">    <span class="dt">keytype =</span> <span class="st">&quot;ENSEMBL&quot;</span>,</a>
<a class="sourceLine" id="cb355-34" data-line-number="34">    <span class="dt">multiVals =</span> <span class="st">&quot;first&quot;</span></a>
<a class="sourceLine" id="cb355-35" data-line-number="35">  )</a>
<a class="sourceLine" id="cb355-36" data-line-number="36">  <span class="kw">head</span>(res[<span class="kw">order</span>(res<span class="op">$</span>padj), ], <span class="dv">20</span>)</a>
<a class="sourceLine" id="cb355-37" data-line-number="37">  </a>
<a class="sourceLine" id="cb355-38" data-line-number="38">  res<span class="op">$</span>gene_map &lt;-<span class="st"> </span><span class="kw">paste0</span>(res<span class="op">$</span>symbol, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, res<span class="op">$</span>map, <span class="st">&quot;)&quot;</span>)</a>
<a class="sourceLine" id="cb355-39" data-line-number="39">  </a>
<a class="sourceLine" id="cb355-40" data-line-number="40">  res &lt;-<span class="st"> </span><span class="kw">drop_na</span>(<span class="kw">as.data.frame</span>(res))</a>
<a class="sourceLine" id="cb355-41" data-line-number="41">  </a>
<a class="sourceLine" id="cb355-42" data-line-number="42">  res2_de &lt;-<span class="st"> </span>res <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-43" data-line-number="43"><span class="st">    </span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-44" data-line-number="44"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(symbol,</a>
<a class="sourceLine" id="cb355-45" data-line-number="45">                  stat) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-46" data-line-number="46"><span class="st">    </span><span class="kw">na.omit</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-47" data-line-number="47"><span class="st">    </span><span class="kw">distinct</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-48" data-line-number="48"><span class="st">    </span><span class="kw">group_by</span>(symbol) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-49" data-line-number="49"><span class="st">    </span><span class="kw">summarize</span>(<span class="dt">stat =</span> <span class="kw">mean</span>(stat))</a>
<a class="sourceLine" id="cb355-50" data-line-number="50">  </a>
<a class="sourceLine" id="cb355-51" data-line-number="51">  ranks_de &lt;-<span class="st"> </span><span class="kw">deframe</span>(res2_de)</a>
<a class="sourceLine" id="cb355-52" data-line-number="52"></a>
<a class="sourceLine" id="cb355-53" data-line-number="53">  pathways.hallmark &lt;-</a>
<a class="sourceLine" id="cb355-54" data-line-number="54"><span class="st">    </span><span class="kw">gmtPathways</span>(</a>
<a class="sourceLine" id="cb355-55" data-line-number="55">      <span class="kw">here</span>(<span class="st">&quot;extdata/lib/h.all.v6.2.symbols.gmt&quot;</span>)</a>
<a class="sourceLine" id="cb355-56" data-line-number="56">    )</a>
<a class="sourceLine" id="cb355-57" data-line-number="57">  </a>
<a class="sourceLine" id="cb355-58" data-line-number="58">  fgseaRes &lt;-</a>
<a class="sourceLine" id="cb355-59" data-line-number="59"><span class="st">    </span><span class="kw">fgsea</span>(<span class="dt">pathways =</span> pathways.hallmark,</a>
<a class="sourceLine" id="cb355-60" data-line-number="60">          <span class="dt">stats =</span> ranks_de,</a>
<a class="sourceLine" id="cb355-61" data-line-number="61">          <span class="dt">nperm =</span> <span class="dv">2000</span>)</a>
<a class="sourceLine" id="cb355-62" data-line-number="62">  </a>
<a class="sourceLine" id="cb355-63" data-line-number="63">  fgseaResTidy[[x]] &lt;-<span class="st"> </span>fgseaRes <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-64" data-line-number="64"><span class="st">    </span><span class="kw">as_tibble</span>()</a>
<a class="sourceLine" id="cb355-65" data-line-number="65">  </a>
<a class="sourceLine" id="cb355-66" data-line-number="66">  <span class="kw">names</span>(fgseaResTidy)[x] &lt;-<span class="st"> </span>clusters[x]</a>
<a class="sourceLine" id="cb355-67" data-line-number="67">  </a>
<a class="sourceLine" id="cb355-68" data-line-number="68">  </a>
<a class="sourceLine" id="cb355-69" data-line-number="69">}</a>
<a class="sourceLine" id="cb355-70" data-line-number="70"></a>
<a class="sourceLine" id="cb355-71" data-line-number="71">fgseaResTidy_res &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(fgseaResTidy, <span class="dt">.id =</span> <span class="st">&quot;comparison&quot;</span>)</a>
<a class="sourceLine" id="cb355-72" data-line-number="72"></a>
<a class="sourceLine" id="cb355-73" data-line-number="73"><span class="co"># keeping only the significant pathways in x or more hallmarks</span></a>
<a class="sourceLine" id="cb355-74" data-line-number="74">sig_path &lt;-<span class="st"> </span>fgseaResTidy_res <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-75" data-line-number="75"><span class="st">  </span><span class="kw">filter</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-76" data-line-number="76"><span class="st">  </span><span class="kw">group_by</span>(pathway) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-77" data-line-number="77"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">count</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-78" data-line-number="78"><span class="st">  </span><span class="kw">arrange</span>(<span class="kw">desc</span>(n)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-79" data-line-number="79"><span class="st">  </span><span class="kw">filter</span>(n <span class="op">&gt;</span><span class="st"> </span><span class="dv">3</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-80" data-line-number="80"><span class="st">  </span><span class="kw">pull</span>(pathway)</a>
<a class="sourceLine" id="cb355-81" data-line-number="81"></a>
<a class="sourceLine" id="cb355-82" data-line-number="82"></a>
<a class="sourceLine" id="cb355-83" data-line-number="83">fgseaResTidy_res_f &lt;-<span class="st"> </span>fgseaResTidy_res <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-84" data-line-number="84"><span class="st">  </span><span class="kw">filter</span>(pathway <span class="op">%in%</span><span class="st"> </span>sig_path) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-85" data-line-number="85"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">pathway =</span> <span class="kw">str_remove</span>(pathway, <span class="st">&quot;HALLMARK_&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-86" data-line-number="86"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">pathway =</span> <span class="kw">str_replace_all</span>(pathway, <span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-87" data-line-number="87"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">pathway =</span> <span class="kw">str_to_lower</span>(pathway))  <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-88" data-line-number="88"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">comparison =</span> <span class="kw">paste0</span>(<span class="st">&quot;e&quot;</span>, comparison))</a>
<a class="sourceLine" id="cb355-89" data-line-number="89"></a>
<a class="sourceLine" id="cb355-90" data-line-number="90">fg_cl &lt;-<span class="st"> </span><span class="kw">pivot_wider</span>(</a>
<a class="sourceLine" id="cb355-91" data-line-number="91">  fgseaResTidy_res_f,</a>
<a class="sourceLine" id="cb355-92" data-line-number="92">  <span class="dt">names_from =</span> pathway,</a>
<a class="sourceLine" id="cb355-93" data-line-number="93">  <span class="dt">values_from =</span> NES,</a>
<a class="sourceLine" id="cb355-94" data-line-number="94">  <span class="dt">id_cols =</span> comparison</a>
<a class="sourceLine" id="cb355-95" data-line-number="95">) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-96" data-line-number="96"><span class="st">  </span><span class="kw">as.data.frame</span>()</a>
<a class="sourceLine" id="cb355-97" data-line-number="97"></a>
<a class="sourceLine" id="cb355-98" data-line-number="98"><span class="kw">rownames</span>(fg_cl) &lt;-<span class="st"> </span>fg_cl[, <span class="dv">1</span>]</a>
<a class="sourceLine" id="cb355-99" data-line-number="99"></a>
<a class="sourceLine" id="cb355-100" data-line-number="100">fg_cl_hclust &lt;-<span class="st"> </span><span class="kw">hclust</span>(<span class="kw">dist</span>(fg_cl[, <span class="dv">-1</span>],</a>
<a class="sourceLine" id="cb355-101" data-line-number="101">                            <span class="dt">method =</span> <span class="st">&quot;euclidean&quot;</span>),</a>
<a class="sourceLine" id="cb355-102" data-line-number="102">                       <span class="dt">method =</span> <span class="st">&quot;ward.D&quot;</span>)</a>
<a class="sourceLine" id="cb355-103" data-line-number="103"></a>
<a class="sourceLine" id="cb355-104" data-line-number="104"></a>
<a class="sourceLine" id="cb355-105" data-line-number="105">fg_path &lt;-<span class="st"> </span><span class="kw">pivot_wider</span>(</a>
<a class="sourceLine" id="cb355-106" data-line-number="106">  fgseaResTidy_res_f,</a>
<a class="sourceLine" id="cb355-107" data-line-number="107">  <span class="dt">names_from =</span></a>
<a class="sourceLine" id="cb355-108" data-line-number="108">    comparison,</a>
<a class="sourceLine" id="cb355-109" data-line-number="109">  <span class="dt">values_from =</span> NES,</a>
<a class="sourceLine" id="cb355-110" data-line-number="110">  <span class="dt">id_cols =</span> pathway</a>
<a class="sourceLine" id="cb355-111" data-line-number="111">) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-112" data-line-number="112"><span class="st">  </span><span class="kw">as.data.frame</span>()</a>
<a class="sourceLine" id="cb355-113" data-line-number="113"></a>
<a class="sourceLine" id="cb355-114" data-line-number="114"><span class="kw">rownames</span>(fg_path) &lt;-<span class="st"> </span>fg_path[, <span class="dv">1</span>]</a>
<a class="sourceLine" id="cb355-115" data-line-number="115"></a>
<a class="sourceLine" id="cb355-116" data-line-number="116">fg_path_hclust &lt;-<span class="st"> </span><span class="kw">hclust</span>(<span class="kw">dist</span>(fg_path[, <span class="dv">-1</span>],</a>
<a class="sourceLine" id="cb355-117" data-line-number="117">                              <span class="dt">method =</span> <span class="st">&quot;euclidean&quot;</span>),</a>
<a class="sourceLine" id="cb355-118" data-line-number="118">                         <span class="dt">method =</span> <span class="st">&quot;ward.D&quot;</span>)</a>
<a class="sourceLine" id="cb355-119" data-line-number="119"></a>
<a class="sourceLine" id="cb355-120" data-line-number="120"></a>
<a class="sourceLine" id="cb355-121" data-line-number="121"><span class="co"># thanks to https://mgrcbioinfo.github.io/my_GSEA_plot/</span></a>
<a class="sourceLine" id="cb355-122" data-line-number="122"></a>
<a class="sourceLine" id="cb355-123" data-line-number="123">myGseaPlot &lt;-<span class="st"> </span><span class="cf">function</span>(gseaDataFrame, <span class="dt">adjPCutoff =</span> <span class="fl">0.05</span>) {</a>
<a class="sourceLine" id="cb355-124" data-line-number="124">  <span class="co"># subset data frame using p value cutoff and add status factor</span></a>
<a class="sourceLine" id="cb355-125" data-line-number="125">  toPlot &lt;-<span class="st"> </span><span class="kw">subset</span>(gseaDataFrame, padj <span class="op">&lt;=</span><span class="st"> </span>adjPCutoff) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-126" data-line-number="126"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="kw">case_when</span>(NES <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span> <span class="op">~</span><span class="st"> &quot;Up&quot;</span>, NES <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span> <span class="op">~</span><span class="st"> &quot;Down&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-127" data-line-number="127"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="kw">factor</span>(status, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;Up&quot;</span>, <span class="st">&quot;Down&quot;</span>)))</a>
<a class="sourceLine" id="cb355-128" data-line-number="128">  </a>
<a class="sourceLine" id="cb355-129" data-line-number="129">  <span class="co"># create plot</span></a>
<a class="sourceLine" id="cb355-130" data-line-number="130">  p &lt;-</a>
<a class="sourceLine" id="cb355-131" data-line-number="131"><span class="st">    </span><span class="kw">ggplot</span>(</a>
<a class="sourceLine" id="cb355-132" data-line-number="132">      <span class="dt">data =</span> toPlot,</a>
<a class="sourceLine" id="cb355-133" data-line-number="133">      <span class="dt">mapping =</span> <span class="kw">aes</span>(</a>
<a class="sourceLine" id="cb355-134" data-line-number="134">        <span class="dt">x =</span> <span class="kw">fct_relevel</span>(comparison,</a>
<a class="sourceLine" id="cb355-135" data-line-number="135">                        <span class="kw">rownames</span>(fg_cl)[fg_cl_hclust<span class="op">$</span>order]),</a>
<a class="sourceLine" id="cb355-136" data-line-number="136">        <span class="dt">y =</span> <span class="kw">fct_relevel</span>(pathway,</a>
<a class="sourceLine" id="cb355-137" data-line-number="137">                        <span class="kw">rownames</span>(fg_path)[fg_path_hclust<span class="op">$</span>order]),</a>
<a class="sourceLine" id="cb355-138" data-line-number="138">        <span class="dt">color =</span> status,</a>
<a class="sourceLine" id="cb355-139" data-line-number="139">        <span class="dt">shape =</span> status,</a>
<a class="sourceLine" id="cb355-140" data-line-number="140">        <span class="dt">fill =</span> NES,</a>
<a class="sourceLine" id="cb355-141" data-line-number="141">        <span class="dt">size =</span> <span class="op">-</span><span class="kw">log10</span>(padj)</a>
<a class="sourceLine" id="cb355-142" data-line-number="142">      )</a>
<a class="sourceLine" id="cb355-143" data-line-number="143">    ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-144" data-line-number="144"><span class="st">    </span><span class="kw">geom_point</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb355-145" data-line-number="145"><span class="st">    </span><span class="kw">scale_shape_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="dv">24</span>, <span class="dv">25</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-146" data-line-number="146"><span class="st">    </span><span class="kw">scale_fill_gradient2</span>(</a>
<a class="sourceLine" id="cb355-147" data-line-number="147">      <span class="dt">high =</span> <span class="st">&quot;firebrick3&quot;</span>,</a>
<a class="sourceLine" id="cb355-148" data-line-number="148">      <span class="dt">low =</span> <span class="st">&quot;deepskyblue3&quot;</span>,</a>
<a class="sourceLine" id="cb355-149" data-line-number="149">      <span class="dt">mid =</span> <span class="st">&quot;white&quot;</span>,</a>
<a class="sourceLine" id="cb355-150" data-line-number="150">      <span class="dt">midpoint =</span> <span class="dv">0</span></a>
<a class="sourceLine" id="cb355-151" data-line-number="151">    ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-152" data-line-number="152"><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="dt">Up =</span> <span class="st">&quot;coral3&quot;</span>, <span class="dt">Down =</span> <span class="st">&quot;deepskyblue3&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-153" data-line-number="153"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;GSEA&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-154" data-line-number="154"><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb355-155" data-line-number="155"><span class="st">    </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb355-156" data-line-number="156">      <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>, <span class="dt">size =</span> <span class="dv">18</span>),</a>
<a class="sourceLine" id="cb355-157" data-line-number="157">      <span class="dt">axis.title.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">16</span>),</a>
<a class="sourceLine" id="cb355-158" data-line-number="158">      <span class="dt">panel.grid =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb355-159" data-line-number="159">      <span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>,</a>
<a class="sourceLine" id="cb355-160" data-line-number="160">      <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(</a>
<a class="sourceLine" id="cb355-161" data-line-number="161">        <span class="dt">angle =</span> <span class="dv">90</span>,</a>
<a class="sourceLine" id="cb355-162" data-line-number="162">        <span class="dt">hjust =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb355-163" data-line-number="163">        <span class="dt">size =</span> <span class="dv">16</span>,</a>
<a class="sourceLine" id="cb355-164" data-line-number="164">        <span class="dt">vjust =</span> <span class="fl">.5</span></a>
<a class="sourceLine" id="cb355-165" data-line-number="165">      ),</a>
<a class="sourceLine" id="cb355-166" data-line-number="166">      <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">14</span>)</a>
<a class="sourceLine" id="cb355-167" data-line-number="167">    ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-168" data-line-number="168"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-169" data-line-number="169"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;expanded clusters&quot;</span>)</a>
<a class="sourceLine" id="cb355-170" data-line-number="170">  </a>
<a class="sourceLine" id="cb355-171" data-line-number="171">  <span class="kw">return</span>(p)</a>
<a class="sourceLine" id="cb355-172" data-line-number="172">}</a>
<a class="sourceLine" id="cb355-173" data-line-number="173"></a>
<a class="sourceLine" id="cb355-174" data-line-number="174">hallmarks_plot &lt;-</a>
<a class="sourceLine" id="cb355-175" data-line-number="175"><span class="st">  </span><span class="kw">myGseaPlot</span>(<span class="dt">gseaDataFrame =</span> fgseaResTidy_res_f, <span class="dt">adjPCutoff =</span> <span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb355-176" data-line-number="176"></a>
<a class="sourceLine" id="cb355-177" data-line-number="177">hallmarks_plot</a></code></pre></div>
<p><img src="_main_files/figure-html/fgsea-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cell-lines.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="session-info.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
