<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Session Info | ACT Paper</title>
  <meta name="description" content="9 Session Info | ACT Paper" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Session Info | ACT Paper" />
  <meta property="og:type" content="book" />





  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Session Info | ACT Paper" />




<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />


<link rel="prev" href="mda-231-dna-rna.html"/>
<link rel="next" href="instructions-1.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Instructions</a></li>
<li class="chapter" data-level="2" data-path="technical-properties.html"><a href="technical-properties.html"><i class="fa fa-check"></i><b>2</b> Technical properties</a><ul>
<li class="chapter" data-level="2.1" data-path="technical-properties.html"><a href="technical-properties.html#breadth-of-coverage"><i class="fa fa-check"></i><b>2.1</b> Breadth of coverage</a></li>
<li class="chapter" data-level="2.2" data-path="technical-properties.html"><a href="technical-properties.html#overdispersion"><i class="fa fa-check"></i><b>2.2</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="tumors.html"><a href="tumors.html"><i class="fa fa-check"></i><b>3</b> Tumors</a><ul>
<li class="chapter" data-level="3.1" data-path="tumors.html"><a href="tumors.html#tn1"><i class="fa fa-check"></i><b>3.1</b> TN1</a></li>
<li class="chapter" data-level="3.2" data-path="tumors.html"><a href="tumors.html#tn2"><i class="fa fa-check"></i><b>3.2</b> TN2</a></li>
<li class="chapter" data-level="3.3" data-path="tumors.html"><a href="tumors.html#tn3"><i class="fa fa-check"></i><b>3.3</b> TN3</a></li>
<li class="chapter" data-level="3.4" data-path="tumors.html"><a href="tumors.html#tn4"><i class="fa fa-check"></i><b>3.4</b> TN4</a></li>
<li class="chapter" data-level="3.5" data-path="tumors.html"><a href="tumors.html#tn5"><i class="fa fa-check"></i><b>3.5</b> TN5</a></li>
<li class="chapter" data-level="3.6" data-path="tumors.html"><a href="tumors.html#tn6"><i class="fa fa-check"></i><b>3.6</b> TN6</a></li>
<li class="chapter" data-level="3.7" data-path="tumors.html"><a href="tumors.html#tn7"><i class="fa fa-check"></i><b>3.7</b> TN7</a></li>
<li class="chapter" data-level="3.8" data-path="tumors.html"><a href="tumors.html#tn8"><i class="fa fa-check"></i><b>3.8</b> TN8</a></li>
<li class="chapter" data-level="3.9" data-path="tumors.html"><a href="tumors.html#clones-barplot"><i class="fa fa-check"></i><b>3.9</b> Clones barplot</a></li>
<li class="chapter" data-level="3.10" data-path="tumors.html"><a href="tumors.html#shannon-diversity"><i class="fa fa-check"></i><b>3.10</b> Shannon diversity</a></li>
<li class="chapter" data-level="3.11" data-path="tumors.html"><a href="tumors.html#gainslosses-plot"><i class="fa fa-check"></i><b>3.11</b> Gains/Losses plot</a></li>
<li class="chapter" data-level="3.12" data-path="tumors.html"><a href="tumors.html#genomic-classes-plots"><i class="fa fa-check"></i><b>3.12</b> Genomic classes plots</a><ul>
<li class="chapter" data-level="3.12.1" data-path="tumors.html"><a href="tumors.html#counts"><i class="fa fa-check"></i><b>3.12.1</b> Counts</a></li>
<li class="chapter" data-level="3.12.2" data-path="tumors.html"><a href="tumors.html#segment-length"><i class="fa fa-check"></i><b>3.12.2</b> Segment length</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="single-cell-trees.html"><a href="single-cell-trees.html"><i class="fa fa-check"></i><b>4</b> Single-Cell Trees</a></li>
<li class="chapter" data-level="5" data-path="act-10x-cna.html"><a href="act-10x-cna.html"><i class="fa fa-check"></i><b>5</b> ACT &amp; 10X CNA</a><ul>
<li class="chapter" data-level="5.1" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn1-co-clustering-umaps"><i class="fa fa-check"></i><b>5.1</b> TN1 Co-clustering umaps</a></li>
<li class="chapter" data-level="5.2" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn1-gene-frequency-barplot"><i class="fa fa-check"></i><b>5.2</b> TN1 Gene Frequency Barplot</a></li>
<li class="chapter" data-level="5.3" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn3-co-clustering-umaps"><i class="fa fa-check"></i><b>5.3</b> TN3 Co-clustering umaps</a></li>
<li class="chapter" data-level="5.4" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn3-gene-frequency-barplot"><i class="fa fa-check"></i><b>5.4</b> TN3 Gene Frequency Barplot</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html"><i class="fa fa-check"></i><b>6</b> Exome and WGD</a><ul>
<li class="chapter" data-level="6.1" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#data"><i class="fa fa-check"></i><b>6.1</b> Data</a></li>
<li class="chapter" data-level="6.2" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#filtering"><i class="fa fa-check"></i><b>6.2</b> Filtering</a></li>
<li class="chapter" data-level="6.3" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#mut-timer-information"><i class="fa fa-check"></i><b>6.3</b> Mut timer information</a></li>
<li class="chapter" data-level="6.4" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#mutation-burden"><i class="fa fa-check"></i><b>6.4</b> Mutation burden</a></li>
<li class="chapter" data-level="6.5" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#oncomap"><i class="fa fa-check"></i><b>6.5</b> Oncomap</a></li>
<li class="chapter" data-level="6.6" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#wgd-estimation-in-mutational-timing"><i class="fa fa-check"></i><b>6.6</b> WGD estimation in mutational timing</a></li>
<li class="chapter" data-level="6.7" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#wgd-estimations-in-copy-number-time"><i class="fa fa-check"></i><b>6.7</b> WGD estimations in copy number time</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cell-lines.html"><a href="cell-lines.html"><i class="fa fa-check"></i><b>7</b> Cell lines</a><ul>
<li class="chapter" data-level="7.1" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231"><i class="fa fa-check"></i><b>7.1</b> MDA-MB-231</a></li>
<li class="chapter" data-level="7.2" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231-ex1"><i class="fa fa-check"></i><b>7.2</b> MDA-MB-231-EX1</a></li>
<li class="chapter" data-level="7.3" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231-ex2"><i class="fa fa-check"></i><b>7.3</b> MDA-MB-231-EX2</a></li>
<li class="chapter" data-level="7.4" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-453"><i class="fa fa-check"></i><b>7.4</b> MDA-MB-453</a></li>
<li class="chapter" data-level="7.5" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-157"><i class="fa fa-check"></i><b>7.5</b> MDA-MB-157</a></li>
<li class="chapter" data-level="7.6" data-path="cell-lines.html"><a href="cell-lines.html#bt20"><i class="fa fa-check"></i><b>7.6</b> BT20</a></li>
<li class="chapter" data-level="7.7" data-path="cell-lines.html"><a href="cell-lines.html#clones-barplot-1"><i class="fa fa-check"></i><b>7.7</b> Clones barplot</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html"><i class="fa fa-check"></i><b>8</b> MDA-231 DNA &amp; RNA</a><ul>
<li class="chapter" data-level="8.1" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#co-clustering"><i class="fa fa-check"></i><b>8.1</b> Co-clustering</a></li>
<li class="chapter" data-level="8.2" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#single-cell-bulk-heatmap"><i class="fa fa-check"></i><b>8.2</b> Single-cell + Bulk Heatmap</a></li>
<li class="chapter" data-level="8.3" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#bulk-heatmap"><i class="fa fa-check"></i><b>8.3</b> Bulk Heatmap</a></li>
<li class="chapter" data-level="8.4" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#dna-rna"><i class="fa fa-check"></i><b>8.4</b> DNA &amp; RNA</a></li>
<li class="chapter" data-level="8.5" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#moving-averages"><i class="fa fa-check"></i><b>8.5</b> Moving averages</a></li>
<li class="chapter" data-level="8.6" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#de"><i class="fa fa-check"></i><b>8.6</b> DE</a></li>
<li class="chapter" data-level="8.7" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#rna-pca"><i class="fa fa-check"></i><b>8.7</b> RNA PCA</a></li>
<li class="chapter" data-level="8.8" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#fgsea"><i class="fa fa-check"></i><b>8.8</b> FGSEA</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>9</b> Session Info</a><ul>
<li class="chapter" data-level="9.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>9.1</b> Dockerfile</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="instructions-1.html"><a href="instructions-1.html"><i class="fa fa-check"></i><b>10</b> Instructions</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ACT Paper</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="session-info" class="section level1">
<h1><span class="header-section-number">9</span> Session Info</h1>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb356-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>## R version 3.6.2 (2019-12-12)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Red Hat Enterprise Linux
##
## Matrix products: default
## BLAS:   /usr/lib64/R/lib/libRblas.so
## LAPACK: /usr/lib64/R/lib/libRlapack.so
##
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C
##
## attached base packages:
##  [1] stats4    grid      parallel  stats     graphics  grDevices utils     datasets  methods   base
##
## other attached packages:
##  [1] containerit_0.6.0.9003                  scran_1.14.6
##  [3] SingleCellExperiment_1.8.0              fgsea_1.12.0
##  [5] Rcpp_1.0.4.6                            biomaRt_2.42.1
##  [7] boot_1.3-25                             janitor_2.0.1
##  [9] jcolors_0.0.4                           vcfR_1.12.0
## [11] conflicted_1.0.4                        Homo.sapiens_1.3.1
## [13] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 org.Hs.eg.db_3.10.0
## [15] GO.db_3.10.0                            OrganismDbi_1.28.0
## [17] DESeq2_1.26.0                           SummarizedExperiment_1.16.1
## [19] DelayedArray_0.12.3                     BiocParallel_1.20.1
## [21] matrixStats_0.56.0                      jsonlite_1.7.0
## [23] AnnotationHub_2.18.0                    BiocFileCache_1.10.2
## [25] dbplyr_1.4.4                            GenomicFeatures_1.38.2
## [27] AnnotationDbi_1.48.0                    Biobase_2.46.0
## [29] GenomicRanges_1.38.0                    GenomeInfoDb_1.22.1
## [31] IRanges_2.20.2                          S4Vectors_0.24.4
## [33] BiocGenerics_0.32.0                     tximport_1.14.2
## [35] future.apply_1.7.0                      future_1.21.0
## [37] ggtree_2.0.4                            cowplot_1.0.0
## [39] amap_0.8-18                             ComplexHeatmap_2.2.0
## [41] forcats_0.5.0                           stringr_1.4.0
## [43] dplyr_1.0.0                             purrr_0.3.4
## [45] readr_1.3.1                             tidyr_1.0.2
## [47] tibble_3.0.3                            tidyverse_1.3.0
## [49] uwot_0.1.8                              Matrix_1.2-18
## [51] phangorn_2.5.5                          ape_5.4
## [53] ggbeeswarm_0.6.0                        ggplot2_3.2.1
## [55] umap_0.2.5.0                            patchwork_1.0.0
## [57] rstatix_0.6.0                           here_0.1
## [59] paletteer_1.1.0                         bookdown_0.18
##
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                    Hmisc_4.4-1                   Rsamtools_2.2.3
##   [4] rprojroot_1.3-2               crayon_1.3.4                  MASS_7.3-51.6
##   [7] nlme_3.1-148                  backports_1.1.6               reprex_0.3.0
##  [10] rlang_0.4.7                   XVector_0.26.0                readxl_1.3.1
##  [13] irlba_2.3.3                   limma_3.42.2                  scater_1.14.6
##  [16] rjson_0.2.20                  bit64_4.0.2                   glue_1.4.0
##  [19] geiger_2.0.6.4                vipor_0.4.5                   PNWColors_0.1.0
##  [22] base64url_1.4                 haven_2.3.1                   tidyselect_1.1.0
##  [25] rio_0.5.16                    XML_3.99-0.3                  packrat_0.5.0
##  [28] GenomicAlignments_1.22.1      xtable_1.8-4                  magrittr_1.5
##  [31] evaluate_0.14                 cli_2.0.2                     zlibbioc_1.32.0
##  [34] dbscan_1.1-5                  miniUI_0.1.1.1                rstudioapi_0.11
##  [37] rpart_4.1-15                  fastmatch_1.1-0               lambda.r_1.2.4
##  [40] treeio_1.10.0                 maps_3.3.0                    shiny_1.5.0
##  [43] BiocSingular_1.2.2            xfun_0.16                     askpass_1.1
##  [46] clue_0.3-57                   cluster_2.1.0                 clusterGeneration_1.3.4
##  [49] expm_0.999-5                  interactiveDisplayBase_1.24.0 subplex_1.6
##  [52] listenv_0.8.0                 Biostrings_2.54.0             png_0.1-7
##  [55] permute_0.9-5                 withr_2.2.0                   bitops_1.0-6
##  [58] RBGL_1.62.1                   cellranger_1.1.0              coda_0.19-3
##  [61] dqrng_0.2.1                   pillar_1.4.6                  GlobalOptions_0.1.1
##  [64] fs_1.5.0                      scatterplot3d_0.3-41          GetoptLong_0.1.8
##  [67] DelayedMatrixStats_1.8.0      vctrs_0.3.2                   ellipsis_0.3.1
##  [70] generics_0.0.2                tools_3.6.2                   foreign_0.8-76
##  [73] beeswarm_0.2.3                munsell_0.5.0                 fastmap_1.0.1
##  [76] compiler_3.6.2                abind_1.4-5                   httpuv_1.5.4
##  [79] rtracklayer_1.46.0            stevedore_0.9.3               rcartocolor_2.0.0
##  [82] scico_1.2.0                   animation_2.6                 GenomeInfoDbData_1.2.2
##  [85] gridExtra_2.3                 edgeR_3.28.1                  lattice_0.20-41
##  [88] utf8_1.1.4                    later_1.1.0.1                 prismatic_0.2.0
##  [91] semver_0.2.0                  scales_1.1.1                  graph_1.64.0
##  [94] tidytree_0.3.3                carData_3.0-4                 genefilter_1.68.0
##  [97] lazyeval_0.2.2                promises_1.1.1                car_3.0-10
## [100] latticeExtra_0.6-29           brew_1.0-6                    reticulate_1.16
## [103] checkmate_2.0.0               rmarkdown_2.3                 openxlsx_4.1.5
## [106] statmod_1.4.34                dichromat_2.0-0               igraph_1.2.5
## [109] survival_3.2-3                numDeriv_2016.8-1.1           yaml_2.2.1
## [112] plotrix_3.7-8                 versions_0.3                  htmltools_0.5.1.1
## [115] memoise_1.1.0                 locfit_1.5-9.4                batchtools_0.9.13
## [118] quadprog_1.5-8                memuse_4.1-0                  viridisLite_0.3.0
## [121] digest_0.6.25                 assertthat_0.2.1              mime_0.9
## [124] rappdirs_0.3.1                futile.options_1.0.1          palr_0.2.0
## [127] pals_1.6                      RSQLite_2.2.0                 debugme_1.1.0
## [130] mapproj_1.2.7                 data.table_1.13.0             blob_1.2.1
## [133] vegan_2.5-6                   futile.logger_1.4.3           deSolve_1.28
## [136] splines_3.6.2                 Formula_1.2-3                 labeling_0.3
## [139] rematch2_2.1.1                RCurl_1.98-1.2                broom_0.7.0
## [142] hms_0.5.3                     modelr_0.1.8                  colorspace_1.4-1
## [145] base64enc_0.1-3               BiocManager_1.30.10           mnormt_1.5-6
## [148] shape_1.4.4                   evobiR_1.1                    nnet_7.3-14
## [151] mvtnorm_1.1-0                 circlize_0.4.8                fansi_0.4.1
## [154] parallelly_1.23.0             R6_2.4.1                      lifecycle_0.2.0
## [157] phytools_0.7-20               formatR_1.7                   oompaBase_3.2.9
## [160] zip_2.1.0                     curl_4.3                      snakecase_0.11.0
## [163] fastcluster_1.1.25            desc_1.2.0                    RcppAnnoy_0.0.16
## [166] pinfsc50_1.2.0                RColorBrewer_1.1-2            htmlwidgets_1.5.1
## [169] rvest_0.3.6                   mgcv_1.8-31                   globals_0.14.0
## [172] openssl_1.4.2                 htmlTable_2.0.1               codetools_0.2-16
## [175] lubridate_1.7.8               gtools_3.8.2                  prettyunits_1.1.1
## [178] RSpectra_0.16-0               gtable_0.3.0                  DBI_1.1.0
## [181] httr_1.4.2                    stringi_1.4.6                 progress_1.2.2
## [184] farver_2.0.3                  annotate_1.64.0               viridis_0.5.1
## [187] seqinr_3.6-1                  xml2_1.3.2                    combinat_0.0-8
## [190] shinyFiles_0.9.0              rvcheck_0.1.8                 BiocNeighbors_1.4.2
## [193] ade4_1.7-15                   geneplotter_1.64.0            BiocVersion_3.10.1
## [196] bit_1.1-15.2                  jpeg_0.1-8.1                  pkgconfig_2.0.3
## [199] knitr_1.29</code></pre>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb358-1" data-line-number="1"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(<span class="st">&quot;containerit&quot;</span>))</a>
<a class="sourceLine" id="cb358-2" data-line-number="2">my_dockerfile &lt;-<span class="st"> </span>containerit<span class="op">::</span><span class="kw">dockerfile</span>(<span class="dt">from =</span> utils<span class="op">::</span><span class="kw">sessionInfo</span>(),</a>
<a class="sourceLine" id="cb358-3" data-line-number="3">                                         <span class="dt">versioned_libs =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb358-4" data-line-number="4">                                         <span class="dt">versioned_packages =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## INFO [2021-03-18 00:43:10] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:10] Trying to determine system requirements for the package(s) &#39;amap&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:10] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:10] Trying to determine system requirements for the package(s) &#39;AnnotationDbi&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:11] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:11] Trying to determine system requirements for the package(s) &#39;AnnotationHub&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:11] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:11] Trying to determine system requirements for the package(s) &#39;ape&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:12] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:12] Trying to determine system requirements for the package(s) &#39;Biobase&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:12] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:12] Trying to determine system requirements for the package(s) &#39;BiocFileCache&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:13] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:13] Trying to determine system requirements for the package(s) &#39;BiocGenerics&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:14] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:14] Trying to determine system requirements for the package(s) &#39;BiocParallel&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:14] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:14] Trying to determine system requirements for the package(s) &#39;biomaRt&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:15] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:15] Trying to determine system requirements for the package(s) &#39;bookdown&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:15] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:15] Trying to determine system requirements for the package(s) &#39;boot&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:16] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:16] Trying to determine system requirements for the package(s) &#39;ComplexHeatmap&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:16] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:16] Trying to determine system requirements for the package(s) &#39;conflicted&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:17] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:17] Trying to determine system requirements for the package(s) &#39;cowplot&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:17] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:17] Trying to determine system requirements for the package(s) &#39;dbplyr&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:18] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:18] Trying to determine system requirements for the package(s) &#39;DelayedArray&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:18] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:18] Trying to determine system requirements for the package(s) &#39;DESeq2&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:19] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:19] Trying to determine system requirements for the package(s) &#39;dplyr&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:19] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:19] Trying to determine system requirements for the package(s) &#39;fgsea&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:20] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:20] Trying to determine system requirements for the package(s) &#39;forcats&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:20] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:20] Trying to determine system requirements for the package(s) &#39;future&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:21] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:21] Trying to determine system requirements for the package(s) &#39;future.apply&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:22] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:22] Trying to determine system requirements for the package(s) &#39;GenomeInfoDb&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:22] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:22] Trying to determine system requirements for the package(s) &#39;GenomicFeatures&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:23] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:23] Trying to determine system requirements for the package(s) &#39;GenomicRanges&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:23] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:23] Trying to determine system requirements for the package(s) &#39;ggbeeswarm&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:24] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:24] Trying to determine system requirements for the package(s) &#39;ggplot2&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:24] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:24] Trying to determine system requirements for the package(s) &#39;ggtree&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:25] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:25] Trying to determine system requirements for the package(s) &#39;GO.db&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:25] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:26] Trying to determine system requirements for the package(s) &#39;here&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:26] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:26] Trying to determine system requirements for the package(s) &#39;Homo.sapiens&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:26] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:26] Trying to determine system requirements for the package(s) &#39;IRanges&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:27] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:27] Trying to determine system requirements for the package(s) &#39;janitor&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:28] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:28] Trying to determine system requirements for the package(s) &#39;jcolors&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:28] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:28] Trying to determine system requirements for the package(s) &#39;jsonlite&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:29] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:29] Trying to determine system requirements for the package(s) &#39;Matrix&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:29] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:29] Trying to determine system requirements for the package(s) &#39;matrixStats&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:29] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:30] Trying to determine system requirements for the package(s) &#39;org.Hs.eg.db&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:30] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:30] Trying to determine system requirements for the package(s) &#39;OrganismDbi&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:31] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:31] Trying to determine system requirements for the package(s) &#39;paletteer&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:31] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:31] Trying to determine system requirements for the package(s) &#39;patchwork&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:32] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:32] Trying to determine system requirements for the package(s) &#39;phangorn&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:32] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:32] Trying to determine system requirements for the package(s) &#39;purrr&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:33] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:33] Trying to determine system requirements for the package(s) &#39;Rcpp&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:33] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:33] Trying to determine system requirements for the package(s) &#39;readr&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:34] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:34] Trying to determine system requirements for the package(s) &#39;rstatix&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:34] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:34] Trying to determine system requirements for the package(s) &#39;S4Vectors&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:35] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:35] Trying to determine system requirements for the package(s) &#39;scran&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:35] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:35] Trying to determine system requirements for the package(s) &#39;SingleCellExperiment&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:36] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:36] Trying to determine system requirements for the package(s) &#39;stringr&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:36] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:36] Trying to determine system requirements for the package(s) &#39;SummarizedExperiment&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:37] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:37] Trying to determine system requirements for the package(s) &#39;tibble&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:37] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:37] Trying to determine system requirements for the package(s) &#39;tidyr&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:38] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:38] Trying to determine system requirements for the package(s) &#39;tidyverse&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:38] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:38] Trying to determine system requirements for the package(s) &#39;TxDb.Hsapiens.UCSC.hg19.knownGene&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:39] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:39] Trying to determine system requirements for the package(s) &#39;tximport&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:39] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:39] Trying to determine system requirements for the package(s) &#39;umap&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:40] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:40] Trying to determine system requirements for the package(s) &#39;uwot&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:40] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2021-03-18 00:43:40] Trying to determine system requirements for the package(s) &#39;vcfR&#39; from sysreqs online DB
## INFO [2021-03-18 00:43:41] Versioned packages enabled, installing &#39;versions&#39;
## INFO [2021-03-18 00:43:41] Adding versioned CRAN packages: Rcpp, boot, janitor, jcolors, vcfR, conflicted, matrixStats, jsonlite, dbplyr, future.apply, future, cowplot, amap, forcats, stringr, dplyr, purrr, readr, tidyr, tibble, tidyverse, uwot, Matrix, phangorn, ape, ggbeeswarm, ggplot2, umap, patchwork, rstatix, here, paletteer, bookdown
## WARN [2021-03-18 00:43:41] Adding versioned Bioconductor packages not supported: scran, SingleCellExperiment, fgsea, biomaRt, Homo.sapiens, TxDb.Hsapiens.UCSC.hg19.knownGene, org.Hs.eg.db, GO.db, OrganismDbi, DESeq2, SummarizedExperiment, DelayedArray, BiocParallel, AnnotationHub, BiocFileCache, GenomicFeatures, AnnotationDbi, Biobase, GenomicRanges, GenomeInfoDb, IRanges, S4Vectors, BiocGenerics, tximport, ggtree, ComplexHeatmap
## INFO [2021-03-18 00:43:41] Adding Bioconductor packages: AnnotationDbi, AnnotationHub, Biobase, BiocFileCache, BiocGenerics, BiocParallel, biomaRt, ComplexHeatmap, DelayedArray, DESeq2, fgsea, GenomeInfoDb, GenomicFeatures, GenomicRanges, ggtree, GO.db, Homo.sapiens, IRanges, org.Hs.eg.db, OrganismDbi, S4Vectors, scran, SingleCellExperiment, SummarizedExperiment, TxDb.Hsapiens.UCSC.hg19.knownGene, tximport
## INFO [2021-03-18 00:43:41] Created Dockerfile-Object based on sessionInfo</code></pre>
<div id="dockerfile" class="section level2">
<h2><span class="header-section-number">9.1</span> Dockerfile</h2>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb360-1" data-line-number="1"><span class="kw">print</span>(my_dockerfile)</a></code></pre></div>
<pre><code>## FROM rocker/r-ver:3.6.2
## LABEL maintainer=&quot;dminussi&quot;
## RUN export DEBIAN_FRONTEND=noninteractive; apt-get -y update \
##   &amp;&amp; apt-get install -y pandoc \
##  pandoc-citeproc
## RUN [&quot;install2.r&quot;, &quot;versions&quot;]
## RUN [&quot;Rscript&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;Rcpp&#39;, &#39;1.0.4.6&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;boot&#39;, &#39;1.3-25&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;janitor&#39;, &#39;2.0.1&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;jcolors&#39;, &#39;0.0.4&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;vcfR&#39;, &#39;1.12.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;conflicted&#39;, &#39;1.0.4&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;matrixStats&#39;, &#39;0.56.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;jsonlite&#39;, &#39;1.7.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;dbplyr&#39;, &#39;1.4.4&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;future.apply&#39;, &#39;1.7.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;future&#39;, &#39;1.21.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;cowplot&#39;, &#39;1.0.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;amap&#39;, &#39;0.8-18&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;forcats&#39;, &#39;0.5.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;stringr&#39;, &#39;1.4.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;dplyr&#39;, &#39;1.0.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;purrr&#39;, &#39;0.3.4&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;readr&#39;, &#39;1.3.1&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;tidyr&#39;, &#39;1.0.2&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;tibble&#39;, &#39;3.0.3&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;tidyverse&#39;, &#39;1.3.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;uwot&#39;, &#39;0.1.8&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;Matrix&#39;, &#39;1.2-18&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;phangorn&#39;, &#39;2.5.5&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;ape&#39;, &#39;5.4&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;ggbeeswarm&#39;, &#39;0.6.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;ggplot2&#39;, &#39;3.2.1&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;umap&#39;, &#39;0.2.5.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;patchwork&#39;, &#39;1.0.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;rstatix&#39;, &#39;0.6.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;here&#39;, &#39;0.1&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;paletteer&#39;, &#39;1.1.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;bookdown&#39;, &#39;0.18&#39;)&quot;]
## RUN [&quot;install2.r&quot;, &quot;-r https://bioconductor.org/packages/3.10/bioc -r https://bioconductor.org/packages/3.10/data/annotation -r https://bioconductor.org/packages/3.10/data/experiment -r https://bioconductor.org/packages/3.10/workflows&quot;, &quot;AnnotationDbi&quot;, &quot;AnnotationHub&quot;, &quot;Biobase&quot;, &quot;BiocFileCache&quot;, &quot;BiocGenerics&quot;, &quot;BiocParallel&quot;, &quot;biomaRt&quot;, &quot;ComplexHeatmap&quot;, &quot;DelayedArray&quot;, &quot;DESeq2&quot;, &quot;fgsea&quot;, &quot;GenomeInfoDb&quot;, &quot;GenomicFeatures&quot;, &quot;GenomicRanges&quot;, &quot;ggtree&quot;, &quot;GO.db&quot;, &quot;Homo.sapiens&quot;, &quot;IRanges&quot;, &quot;org.Hs.eg.db&quot;, &quot;OrganismDbi&quot;, &quot;S4Vectors&quot;, &quot;scran&quot;, &quot;SingleCellExperiment&quot;, &quot;SummarizedExperiment&quot;, &quot;TxDb.Hsapiens.UCSC.hg19.knownGene&quot;, &quot;tximport&quot;]
## WORKDIR /payload/
## CMD [&quot;R&quot;]</code></pre>

<p>This repository contains the scripts used for in the manuscript:
<em>Breast Tumors Maintain a Reservoir of Subclonal Diversity During Primary Expansion</em></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mda-231-dna-rna.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="instructions-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
