<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Technical properties | ACT Paper</title>
  <meta name="description" content="2 Technical properties | ACT Paper" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Technical properties | ACT Paper" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Technical properties | ACT Paper" />
  
  
  

<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="instructions.html"/>
<link rel="next" href="tumors.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="instructions.html"><a href="instructions.html"><i class="fa fa-check"></i><b>1</b> Instructions</a></li>
<li class="chapter" data-level="2" data-path="technical-properties.html"><a href="technical-properties.html"><i class="fa fa-check"></i><b>2</b> Technical properties</a><ul>
<li class="chapter" data-level="2.1" data-path="technical-properties.html"><a href="technical-properties.html#breadth-of-coverage"><i class="fa fa-check"></i><b>2.1</b> Breadth of coverage</a></li>
<li class="chapter" data-level="2.2" data-path="technical-properties.html"><a href="technical-properties.html#overdispersion"><i class="fa fa-check"></i><b>2.2</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="tumors.html"><a href="tumors.html"><i class="fa fa-check"></i><b>3</b> Tumors</a><ul>
<li class="chapter" data-level="3.1" data-path="tumors.html"><a href="tumors.html#tn1"><i class="fa fa-check"></i><b>3.1</b> TN1</a></li>
<li class="chapter" data-level="3.2" data-path="tumors.html"><a href="tumors.html#tn2"><i class="fa fa-check"></i><b>3.2</b> TN2</a></li>
<li class="chapter" data-level="3.3" data-path="tumors.html"><a href="tumors.html#tn3"><i class="fa fa-check"></i><b>3.3</b> TN3</a></li>
<li class="chapter" data-level="3.4" data-path="tumors.html"><a href="tumors.html#tn4"><i class="fa fa-check"></i><b>3.4</b> TN4</a></li>
<li class="chapter" data-level="3.5" data-path="tumors.html"><a href="tumors.html#tn5"><i class="fa fa-check"></i><b>3.5</b> TN5</a></li>
<li class="chapter" data-level="3.6" data-path="tumors.html"><a href="tumors.html#tn6"><i class="fa fa-check"></i><b>3.6</b> TN6</a></li>
<li class="chapter" data-level="3.7" data-path="tumors.html"><a href="tumors.html#tn7"><i class="fa fa-check"></i><b>3.7</b> TN7</a></li>
<li class="chapter" data-level="3.8" data-path="tumors.html"><a href="tumors.html#tn8"><i class="fa fa-check"></i><b>3.8</b> TN8</a></li>
<li class="chapter" data-level="3.9" data-path="tumors.html"><a href="tumors.html#clones-barplot"><i class="fa fa-check"></i><b>3.9</b> Clones barplot</a></li>
<li class="chapter" data-level="3.10" data-path="tumors.html"><a href="tumors.html#shannon-diversity"><i class="fa fa-check"></i><b>3.10</b> Shannon diversity</a></li>
<li class="chapter" data-level="3.11" data-path="tumors.html"><a href="tumors.html#gainslosses-plot"><i class="fa fa-check"></i><b>3.11</b> Gains/Losses plot</a></li>
<li class="chapter" data-level="3.12" data-path="tumors.html"><a href="tumors.html#genomic-classes-plots"><i class="fa fa-check"></i><b>3.12</b> Genomic classes plots</a><ul>
<li class="chapter" data-level="3.12.1" data-path="tumors.html"><a href="tumors.html#counts"><i class="fa fa-check"></i><b>3.12.1</b> Counts</a></li>
<li class="chapter" data-level="3.12.2" data-path="tumors.html"><a href="tumors.html#segment-length"><i class="fa fa-check"></i><b>3.12.2</b> Segment length</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="single-cell-trees.html"><a href="single-cell-trees.html"><i class="fa fa-check"></i><b>4</b> Single-Cell Trees</a></li>
<li class="chapter" data-level="5" data-path="act-10x-cna.html"><a href="act-10x-cna.html"><i class="fa fa-check"></i><b>5</b> ACT &amp; 10X CNA</a><ul>
<li class="chapter" data-level="5.1" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn1-co-clustering-umaps"><i class="fa fa-check"></i><b>5.1</b> TN1 Co-clustering umaps</a></li>
<li class="chapter" data-level="5.2" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn1-gene-frequency-barplot"><i class="fa fa-check"></i><b>5.2</b> TN1 Gene Frequency Barplot</a></li>
<li class="chapter" data-level="5.3" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn3-co-clustering-umaps"><i class="fa fa-check"></i><b>5.3</b> TN3 Co-clustering umaps</a></li>
<li class="chapter" data-level="5.4" data-path="act-10x-cna.html"><a href="act-10x-cna.html#tn3-gene-frequency-barplot"><i class="fa fa-check"></i><b>5.4</b> TN3 Gene Frequency Barplot</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html"><i class="fa fa-check"></i><b>6</b> Exome and WGD</a><ul>
<li class="chapter" data-level="6.1" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#data"><i class="fa fa-check"></i><b>6.1</b> Data</a></li>
<li class="chapter" data-level="6.2" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#filtering"><i class="fa fa-check"></i><b>6.2</b> Filtering</a></li>
<li class="chapter" data-level="6.3" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#mut-timer-information"><i class="fa fa-check"></i><b>6.3</b> Mut timer information</a></li>
<li class="chapter" data-level="6.4" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#mutation-burden"><i class="fa fa-check"></i><b>6.4</b> Mutation burden</a></li>
<li class="chapter" data-level="6.5" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#oncomap"><i class="fa fa-check"></i><b>6.5</b> Oncomap</a></li>
<li class="chapter" data-level="6.6" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#wgd-estimation-in-mutational-timing"><i class="fa fa-check"></i><b>6.6</b> WGD estimation in mutational timing</a></li>
<li class="chapter" data-level="6.7" data-path="exome-and-wgd.html"><a href="exome-and-wgd.html#wgd-estimations-in-copy-number-time"><i class="fa fa-check"></i><b>6.7</b> WGD estimations in copy number time</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cell-lines.html"><a href="cell-lines.html"><i class="fa fa-check"></i><b>7</b> Cell lines</a><ul>
<li class="chapter" data-level="7.1" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231"><i class="fa fa-check"></i><b>7.1</b> MDA-MB-231</a></li>
<li class="chapter" data-level="7.2" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231-ex1"><i class="fa fa-check"></i><b>7.2</b> MDA-MB-231-EX1</a></li>
<li class="chapter" data-level="7.3" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-231-ex2"><i class="fa fa-check"></i><b>7.3</b> MDA-MB-231-EX2</a></li>
<li class="chapter" data-level="7.4" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-453"><i class="fa fa-check"></i><b>7.4</b> MDA-MB-453</a></li>
<li class="chapter" data-level="7.5" data-path="cell-lines.html"><a href="cell-lines.html#mda-mb-157"><i class="fa fa-check"></i><b>7.5</b> MDA-MB-157</a></li>
<li class="chapter" data-level="7.6" data-path="cell-lines.html"><a href="cell-lines.html#bt20"><i class="fa fa-check"></i><b>7.6</b> BT20</a></li>
<li class="chapter" data-level="7.7" data-path="cell-lines.html"><a href="cell-lines.html#clones-barplot-1"><i class="fa fa-check"></i><b>7.7</b> Clones barplot</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html"><i class="fa fa-check"></i><b>8</b> MDA-231 DNA &amp; RNA</a><ul>
<li class="chapter" data-level="8.1" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#co-clustering"><i class="fa fa-check"></i><b>8.1</b> Co-clustering</a></li>
<li class="chapter" data-level="8.2" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#single-cell-bulk-heatmap"><i class="fa fa-check"></i><b>8.2</b> Single-cell + Bulk Heatmap</a></li>
<li class="chapter" data-level="8.3" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#bulk-heatmap"><i class="fa fa-check"></i><b>8.3</b> Bulk Heatmap</a></li>
<li class="chapter" data-level="8.4" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#dna-rna"><i class="fa fa-check"></i><b>8.4</b> DNA &amp; RNA</a></li>
<li class="chapter" data-level="8.5" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#moving-averages"><i class="fa fa-check"></i><b>8.5</b> Moving averages</a></li>
<li class="chapter" data-level="8.6" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#de"><i class="fa fa-check"></i><b>8.6</b> DE</a></li>
<li class="chapter" data-level="8.7" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#rna-pca"><i class="fa fa-check"></i><b>8.7</b> RNA PCA</a></li>
<li class="chapter" data-level="8.8" data-path="mda-231-dna-rna.html"><a href="mda-231-dna-rna.html#fgsea"><i class="fa fa-check"></i><b>8.8</b> FGSEA</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>9</b> Session Info</a><ul>
<li class="chapter" data-level="9.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>9.1</b> Dockerfile</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="instructions-1.html"><a href="instructions-1.html"><i class="fa fa-check"></i><b>10</b> Instructions</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ACT Paper</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="technical-properties" class="section level1">
<h1><span class="header-section-number">2</span> Technical properties</h1>
<div id="breadth-of-coverage" class="section level2">
<h2><span class="header-section-number">2.1</span> Breadth of coverage</h2>
<p>Code to generate *.covhist files. <code>covhist</code> files are included in the <code>extdata/covhist/</code> directory.</p>
<p>For ACT, 100 cells were sampled from each dataset.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">source</span>(<span class="st">&quot;R/setup.R&quot;</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">source</span>(<span class="st">&quot;R/calc_coverage.R&quot;</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">source</span>(<span class="st">&quot;R/overdispersion.R&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="op">&gt;</span><span class="st"> </span>headers_commands.sh</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="op">&gt;</span><span class="st"> </span>reads_commands.sh</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="op">&gt;</span><span class="st"> </span>delete_commands.sh</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="op">&gt;</span><span class="st"> </span>sort_commands.sh</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="op">&gt;</span><span class="st"> </span>genomecoveragebed_commands.sh</a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="op">&gt;</span><span class="st"> </span>mark_duplicates_commands.sh</a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="cf">for</span> INBAM <span class="cf">in</span> <span class="op">/</span>PATH<span class="op">/</span>TO<span class="op">/</span>BAM<span class="op">/</span>FILES<span class="op">/</span><span class="er">*</span>.sort.bam</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">do</a>
<a class="sourceLine" id="cb2-9" data-line-number="9">    INFILEBASE=<span class="st">`</span><span class="dt">basename $INBAM</span><span class="st">`</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10">    INSAMPLEBASE=<span class="er">$</span>{INFILEBASE%.sort.bam}</a>
<a class="sourceLine" id="cb2-11" data-line-number="11">    COVHISTFILE=<span class="er">$</span>{INSAMPLEBASE}.covhist.txt</a>
<a class="sourceLine" id="cb2-12" data-line-number="12">    OUTBAM=<span class="er">$</span>INSAMPLEBASE.sorted.bam</a>
<a class="sourceLine" id="cb2-13" data-line-number="13">    INSAMPLEBASE_MD=<span class="st">`</span><span class="dt">basename -s .sort.bam $INFILEBASE</span><span class="st">`</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14">    OUTBAM_MD=<span class="er">$</span>INSAMPLEBASE</a>
<a class="sourceLine" id="cb2-15" data-line-number="15"></a>
<a class="sourceLine" id="cb2-16" data-line-number="16">    <span class="co"># marking duplicates</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17">    touch <span class="op">$</span>OUTBAM.txt</a>
<a class="sourceLine" id="cb2-18" data-line-number="18">    </a>
<a class="sourceLine" id="cb2-19" data-line-number="19">    printf <span class="st">&quot;%s</span><span class="ch">\n</span><span class="st">&quot;</span> <span class="st">&quot;java -XX:ParallelGCThreads=4 -Xmx150g -jar picard.jar MarkDuplicates I=$INFILEBASE O=$OUTBAM_MD.marked.sort.bam M=$OUTBAM_MD.marked.sort.bam.txt  VALIDATION_STRINGENCY=SILENT AS=true REMOVE_DUPLICATES=true  MAX_RECORDS_IN_RAM=10000000 TMP_DIR=tmp&quot;</span> <span class="op">&gt;</span><span class="er">&gt;</span><span class="st"> </span>mark_duplicates_commands.sh</a>
<a class="sourceLine" id="cb2-20" data-line-number="20"></a>
<a class="sourceLine" id="cb2-21" data-line-number="21">    <span class="co"># Downsampling</span></a>
<a class="sourceLine" id="cb2-22" data-line-number="22">    <span class="co"># Save headers, which will not be included when downsampling</span></a>
<a class="sourceLine" id="cb2-23" data-line-number="23">    printf <span class="st">&quot;%s</span><span class="ch">\n</span><span class="st">&quot;</span> <span class="st">&quot;samtools view $INSAMPLEBASE.marked.sort.bam -H &gt; $INSAMPLEBASE.sam&quot;</span> <span class="op">&gt;</span><span class="er">&gt;</span><span class="st"> </span>headers_commands.sh</a>
<a class="sourceLine" id="cb2-24" data-line-number="24">    <span class="co"># Downsample by randomly selecting a subset of rows besides the header, and</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25">    <span class="co"># appending them to the header</span></a>
<a class="sourceLine" id="cb2-26" data-line-number="26">    <span class="co"># Use the file itself as the seed for the random number generator</span></a>
<a class="sourceLine" id="cb2-27" data-line-number="27">    printf <span class="st">&quot;%s</span><span class="ch">\n</span><span class="st">&quot;</span> <span class="st">&quot;samtools view $INSAMPLEBASE.marked.sort.bam | shuf -n 800000 --random-source=$INSAMPLEBASE.marked.sort.bam &gt;&gt; $INSAMPLEBASE.sam&quot;</span> <span class="op">&gt;</span><span class="er">&gt;</span><span class="st"> </span>reads_commands.sh</a>
<a class="sourceLine" id="cb2-28" data-line-number="28">    <span class="co"># Sort the downsampled SAM file</span></a>
<a class="sourceLine" id="cb2-29" data-line-number="29">    printf <span class="st">&quot;%s</span><span class="ch">\n</span><span class="st">&quot;</span> <span class="st">&quot;samtools sort -o $OUTBAM $INSAMPLEBASE.sam&quot;</span> <span class="op">&gt;</span><span class="er">&gt;</span><span class="st"> </span>sort_commands.sh</a>
<a class="sourceLine" id="cb2-30" data-line-number="30">    <span class="co"># Delete the unsorted SAM files</span></a>
<a class="sourceLine" id="cb2-31" data-line-number="31">    printf <span class="st">&quot;%s</span><span class="ch">\n</span><span class="st">&quot;</span> <span class="st">&quot;rm $INSAMPLEBASE.sam&quot;</span> <span class="op">&gt;</span><span class="er">&gt;</span><span class="st"> </span>delete_commands.sh</a>
<a class="sourceLine" id="cb2-32" data-line-number="32">    <span class="co"># Calculate coverage maximum read size of 50 to all reads</span></a>
<a class="sourceLine" id="cb2-33" data-line-number="33">    printf <span class="st">&quot;genomeCoverageBed -ibam $OUTBAM -fs 50 -g genomes/human.hg19.genome &gt; $COVHISTFILE</span><span class="ch">\n</span><span class="st">&quot;</span> <span class="op">&gt;</span><span class="er">&gt;</span><span class="st"> </span>genomecoveragebed_commands.sh</a>
<a class="sourceLine" id="cb2-34" data-line-number="34">done</a>
<a class="sourceLine" id="cb2-35" data-line-number="35">parallel <span class="op">--</span>jobs <span class="dv">30</span> <span class="op">&lt;</span><span class="st"> </span>mark_duplicates_commands.sh</a>
<a class="sourceLine" id="cb2-36" data-line-number="36">parallel <span class="op">--</span>jobs <span class="dv">70</span> <span class="op">&lt;</span><span class="st"> </span>headers_commands.sh</a>
<a class="sourceLine" id="cb2-37" data-line-number="37">parallel <span class="op">--</span>jobs <span class="dv">70</span> <span class="op">&lt;</span><span class="st"> </span>reads_commands.sh</a>
<a class="sourceLine" id="cb2-38" data-line-number="38">parallel <span class="op">--</span>jobs <span class="dv">70</span> <span class="op">&lt;</span><span class="st"> </span>sort_commands.sh</a>
<a class="sourceLine" id="cb2-39" data-line-number="39">parallel <span class="op">--</span>jobs <span class="dv">70</span> <span class="op">&lt;</span><span class="st"> </span>delete_commands.sh</a>
<a class="sourceLine" id="cb2-40" data-line-number="40">parallel <span class="op">--</span>jobs <span class="dv">70</span> <span class="op">&lt;</span><span class="st"> </span>genomecoveragebed_commands.sh</a></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">tn1_cov &lt;-<span class="st"> </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/TN1/&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN1&quot;</span>,</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">         <span class="dt">tech =</span> <span class="st">&quot;ACT&quot;</span>,</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">         <span class="dt">cellname =</span> <span class="kw">str_replace</span>(cellname, <span class="st">&quot;TN28&quot;</span>, <span class="st">&quot;TN1&quot;</span>))</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"></a>
<a class="sourceLine" id="cb3-6" data-line-number="6">tn2_cov &lt;-<span class="st"> </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/TN2/&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN2&quot;</span>,</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">         <span class="dt">tech =</span> <span class="st">&quot;ACT&quot;</span>,</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">         <span class="dt">cellname =</span> <span class="kw">str_replace</span>(cellname, <span class="st">&quot;TN20&quot;</span>, <span class="st">&quot;TN2&quot;</span>)) </a>
<a class="sourceLine" id="cb3-10" data-line-number="10"></a>
<a class="sourceLine" id="cb3-11" data-line-number="11">tn3_cov &lt;-<span class="st"> </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/TN3/&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN3&quot;</span>,</a>
<a class="sourceLine" id="cb3-13" data-line-number="13">         <span class="dt">tech =</span> <span class="st">&quot;ACT&quot;</span>,</a>
<a class="sourceLine" id="cb3-14" data-line-number="14">         <span class="dt">cellname =</span> <span class="kw">str_replace</span>(cellname, <span class="st">&quot;TN17&quot;</span>, <span class="st">&quot;TN3&quot;</span>))</a>
<a class="sourceLine" id="cb3-15" data-line-number="15"></a>
<a class="sourceLine" id="cb3-16" data-line-number="16">tn4_cov &lt;-<span class="st"> </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/TN4/&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN4&quot;</span>,</a>
<a class="sourceLine" id="cb3-18" data-line-number="18">         <span class="dt">tech =</span> <span class="st">&quot;ACT&quot;</span>,</a>
<a class="sourceLine" id="cb3-19" data-line-number="19">         <span class="dt">cellname =</span> <span class="kw">str_replace</span>(cellname, <span class="st">&quot;TN26&quot;</span>, <span class="st">&quot;TN4&quot;</span>))</a>
<a class="sourceLine" id="cb3-20" data-line-number="20"></a>
<a class="sourceLine" id="cb3-21" data-line-number="21">tn1_10xcnv_cov &lt;-</a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="st">  </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/TN1_10XCNA/&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN1&quot;</span>,</a>
<a class="sourceLine" id="cb3-24" data-line-number="24">         <span class="dt">tech =</span> <span class="st">&quot;10X CNA&quot;</span>)</a>
<a class="sourceLine" id="cb3-25" data-line-number="25"></a>
<a class="sourceLine" id="cb3-26" data-line-number="26">tn3_10xcnv_cov &lt;-<span class="st"> </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/TN3_10XCNA/&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN3&quot;</span>,</a>
<a class="sourceLine" id="cb3-28" data-line-number="28">         <span class="dt">tech =</span> <span class="st">&quot;10X CNA&quot;</span>)</a>
<a class="sourceLine" id="cb3-29" data-line-number="29"></a>
<a class="sourceLine" id="cb3-30" data-line-number="30">t2_doppcr_cov &lt;-<span class="st"> </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/T2_DOPPCR//&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;T2&quot;</span>,</a>
<a class="sourceLine" id="cb3-32" data-line-number="32">         <span class="dt">tech =</span> <span class="st">&quot;DOP-PCR&quot;</span>)</a>
<a class="sourceLine" id="cb3-33" data-line-number="33"></a>
<a class="sourceLine" id="cb3-34" data-line-number="34">t4_doppcr_cov &lt;-<span class="st"> </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/T4_DOPPCR/&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-35" data-line-number="35"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;T4&quot;</span>,</a>
<a class="sourceLine" id="cb3-36" data-line-number="36">         <span class="dt">tech =</span> <span class="st">&quot;DOP-PCR&quot;</span>)</a>
<a class="sourceLine" id="cb3-37" data-line-number="37"></a>
<a class="sourceLine" id="cb3-38" data-line-number="38">t8_doppcr_cov &lt;-<span class="st"> </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/T8_DOPPCR/&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-39" data-line-number="39"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;T8&quot;</span>,</a>
<a class="sourceLine" id="cb3-40" data-line-number="40">         <span class="dt">tech =</span> <span class="st">&quot;DOP-PCR&quot;</span>)</a>
<a class="sourceLine" id="cb3-41" data-line-number="41"></a>
<a class="sourceLine" id="cb3-42" data-line-number="42">t10_doppcr_cov &lt;-<span class="st"> </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/T10_DOPPCR/&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-43" data-line-number="43"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;T10&quot;</span>,</a>
<a class="sourceLine" id="cb3-44" data-line-number="44">         <span class="dt">tech =</span> <span class="st">&quot;DOP-PCR&quot;</span>)</a>
<a class="sourceLine" id="cb3-45" data-line-number="45"></a>
<a class="sourceLine" id="cb3-46" data-line-number="46">dlp_htert_cov &lt;-<span class="st"> </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/DLP_htert/&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-47" data-line-number="47"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;hTERT&quot;</span>,</a>
<a class="sourceLine" id="cb3-48" data-line-number="48">         <span class="dt">tech =</span> <span class="st">&quot;DLP&quot;</span>)</a>
<a class="sourceLine" id="cb3-49" data-line-number="49"></a>
<a class="sourceLine" id="cb3-50" data-line-number="50">dlp_xeno_cov &lt;-<span class="st"> </span><span class="kw">calc_coverage</span>(<span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;extdata/covhist/DLP_xeno/&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-51" data-line-number="51"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;Xeno&quot;</span>,</a>
<a class="sourceLine" id="cb3-52" data-line-number="52">         <span class="dt">tech =</span> <span class="st">&quot;DLP&quot;</span>)</a>
<a class="sourceLine" id="cb3-53" data-line-number="53"></a>
<a class="sourceLine" id="cb3-54" data-line-number="54"><span class="co"># bincounts</span></a>
<a class="sourceLine" id="cb3-55" data-line-number="55">TN1_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/TN1_bincounts.rds&quot;</span>))</a>
<a class="sourceLine" id="cb3-56" data-line-number="56"></a>
<a class="sourceLine" id="cb3-57" data-line-number="57">TN2_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/TN2_bincounts.rds&quot;</span>))</a>
<a class="sourceLine" id="cb3-58" data-line-number="58"></a>
<a class="sourceLine" id="cb3-59" data-line-number="59">TN3_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/TN3_bincounts.rds&quot;</span>))</a>
<a class="sourceLine" id="cb3-60" data-line-number="60"></a>
<a class="sourceLine" id="cb3-61" data-line-number="61">TN4_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/TN4_bincounts.rds&quot;</span>))</a>
<a class="sourceLine" id="cb3-62" data-line-number="62"></a>
<a class="sourceLine" id="cb3-63" data-line-number="63">TN1_10XCNA_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/TN1_10XCNA_bincounts.rds&quot;</span>))</a>
<a class="sourceLine" id="cb3-64" data-line-number="64"></a>
<a class="sourceLine" id="cb3-65" data-line-number="65">TN3_10XCNA_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/TN3_10XCNA_bincounts.rds&quot;</span>))</a>
<a class="sourceLine" id="cb3-66" data-line-number="66"></a>
<a class="sourceLine" id="cb3-67" data-line-number="67">DLP_xeno_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/DLP_xeno_bincounts.rds&quot;</span>))</a>
<a class="sourceLine" id="cb3-68" data-line-number="68"></a>
<a class="sourceLine" id="cb3-69" data-line-number="69">DLP_htert_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/DLP_htert_bincounts.rds&quot;</span>))</a>
<a class="sourceLine" id="cb3-70" data-line-number="70"></a>
<a class="sourceLine" id="cb3-71" data-line-number="71">T2_DOPPCR_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/T2_DOPPCR_bincounts.rds&quot;</span>))</a>
<a class="sourceLine" id="cb3-72" data-line-number="72"></a>
<a class="sourceLine" id="cb3-73" data-line-number="73">T4_DOPPCR_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/T4_DOPPCR_bincounts.rds&quot;</span>))</a>
<a class="sourceLine" id="cb3-74" data-line-number="74"></a>
<a class="sourceLine" id="cb3-75" data-line-number="75">T8_DOPPCR_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/T8_DOPPCR_bincounts.rds&quot;</span>))</a>
<a class="sourceLine" id="cb3-76" data-line-number="76"></a>
<a class="sourceLine" id="cb3-77" data-line-number="77">T10_DOPPCR_bincounts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">here</span>(<span class="st">&quot;extdata/bincounts/T10_DOPPCR_bincounts.rds&quot;</span>))</a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">all_cov &lt;-<span class="st"> </span><span class="kw">rbind</span>(tn2_cov,</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">                 tn1_cov,</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">                 tn1_10xcnv_cov,</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">                 tn4_cov,</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">                 t2_doppcr_cov,</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">                 t4_doppcr_cov,</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">                 tn3_cov,</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">                 tn3_10xcnv_cov,</a>
<a class="sourceLine" id="cb4-9" data-line-number="9">                 t8_doppcr_cov,</a>
<a class="sourceLine" id="cb4-10" data-line-number="10">                 t10_doppcr_cov,</a>
<a class="sourceLine" id="cb4-11" data-line-number="11">                 dlp_htert_cov,</a>
<a class="sourceLine" id="cb4-12" data-line-number="12">                 dlp_xeno_cov)</a>
<a class="sourceLine" id="cb4-13" data-line-number="13"></a>
<a class="sourceLine" id="cb4-14" data-line-number="14">p_breadth &lt;-<span class="st"> </span>all_cov <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tech =</span> <span class="kw">as.factor</span>(tech)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">as.factor</span>(sample)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">fct_relevel</span>(sample, gtools<span class="op">::</span><span class="kw">mixedsort</span>(<span class="kw">unique</span>(all_cov<span class="op">$</span>sample)))) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tech =</span> <span class="kw">fct_relevel</span>(tech, <span class="kw">c</span>(<span class="st">&quot;ACT&quot;</span>, <span class="st">&quot;10X CNA&quot;</span>,<span class="st">&quot;DLP&quot;</span>, <span class="st">&quot;DOP-PCR&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="st">  </span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">x =</span> sample, <span class="dt">y =</span> breadth, <span class="dt">color =</span> tech)) <span class="op">+</span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="st">    </span><span class="kw">scale_color_paletteer_d</span>(<span class="st">&quot;nord::lumina&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="st">  </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="kw">vars</span>(tech), <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>,</a>
<a class="sourceLine" id="cb4-25" data-line-number="25">                                   <span class="dt">vjust =</span> <span class="fl">.5</span>,</a>
<a class="sourceLine" id="cb4-26" data-line-number="26">                                   <span class="dt">hjust =</span> <span class="dv">1</span>),</a>
<a class="sourceLine" id="cb4-27" data-line-number="27">        <span class="dt">strip.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;white&quot;</span>),</a>
<a class="sourceLine" id="cb4-28" data-line-number="28">        <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>)</a>
<a class="sourceLine" id="cb4-29" data-line-number="29">p_breadth</a></code></pre></div>
<p><img src="_main_files/figure-html/plots_coverage-1.png" width="672" /></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">p_breadth_tech &lt;-</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="st">  </span><span class="kw">ggplot</span>(all_cov) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>( <span class="kw">fct_relevel</span>(tech, <span class="kw">c</span>(<span class="st">&quot;ACT&quot;</span>, <span class="st">&quot;10X CNA&quot;</span>, <span class="st">&quot;DLP&quot;</span>,<span class="st">&quot;DOP-PCR&quot;</span>)), <span class="dt">y =</span> breadth, <span class="dt">color =</span> tech)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="st">  </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="st">    </span><span class="kw">scale_color_paletteer_d</span>(<span class="st">&quot;nord::lumina&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits =</span> <span class="kw">c</span>(<span class="fl">0.0108</span>, <span class="fl">0.0127</span>), <span class="dt">labels =</span> scales<span class="op">::</span><span class="kw">label_number</span>(<span class="dt">digits =</span> <span class="dv">3</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="st">    </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>)</a>
<a class="sourceLine" id="cb5-11" data-line-number="11">p_breadth_tech</a></code></pre></div>
<p><img src="_main_files/figure-html/plots_coverage-2.png" width="672" /></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">all_cov <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="st">  </span><span class="kw">kruskal_test</span>(breadth <span class="op">~</span><span class="st"> </span>tech) </a></code></pre></div>
<pre><code>## # A tibble: 1 x 6
##   .y.         n statistic    df         p method        
## * &lt;chr&gt;   &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;     &lt;dbl&gt; &lt;chr&gt;         
## 1 breadth   662      537.     3 4.06e-116 Kruskal-Wallis</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">all_cov <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="st">  </span><span class="kw">dunn_test</span>(breadth <span class="op">~</span><span class="st"> </span>tech)</a></code></pre></div>
<pre><code>## # A tibble: 6 x 9
##   .y.     group1  group2     n1    n2 statistic        p    p.adj p.adj.signif
## * &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;       
## 1 breadth 10X CNA ACT       200   100      6.98 2.97e-12 5.95e-12 ****        
## 2 breadth 10X CNA DLP       200   228    -11.3  2.10e-29 6.29e-29 ****        
## 3 breadth 10X CNA DOP-PCR   200   134    -16.3  1.24e-59 6.18e-59 ****        
## 4 breadth ACT     DLP       100   228    -16.2  3.62e-59 1.45e-58 ****        
## 5 breadth ACT     DOP-PCR   100   134    -20.2  5.71e-91 3.43e-90 ****        
## 6 breadth DLP     DOP-PCR   228   134     -6.68 2.35e-11 2.35e-11 ****</code></pre>
</div>
<div id="overdispersion" class="section level2">
<h2><span class="header-section-number">2.2</span> Overdispersion</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">TN1_overdispersion &lt;-<span class="st">  </span><span class="kw">map_dfr</span>(TN1_bincounts, </a>
<a class="sourceLine" id="cb10-2" data-line-number="2">                    overdispersion) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN1&quot;</span>,</a>
<a class="sourceLine" id="cb10-6" data-line-number="6">         <span class="dt">tech =</span> <span class="st">&quot;ACT&quot;</span>,</a>
<a class="sourceLine" id="cb10-7" data-line-number="7">         <span class="dt">cells =</span> <span class="kw">names</span>(TN1_bincounts)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a>
<a class="sourceLine" id="cb10-9" data-line-number="9"></a>
<a class="sourceLine" id="cb10-10" data-line-number="10">TN2_overdispersion &lt;-<span class="st">  </span><span class="kw">map_dfr</span>(TN2_bincounts, </a>
<a class="sourceLine" id="cb10-11" data-line-number="11">                    overdispersion) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-14" data-line-number="14"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN2&quot;</span>,</a>
<a class="sourceLine" id="cb10-15" data-line-number="15">         <span class="dt">tech =</span> <span class="st">&quot;ACT&quot;</span>,</a>
<a class="sourceLine" id="cb10-16" data-line-number="16">         <span class="dt">cells =</span> <span class="kw">names</span>(TN2_bincounts)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-17" data-line-number="17"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a>
<a class="sourceLine" id="cb10-18" data-line-number="18"></a>
<a class="sourceLine" id="cb10-19" data-line-number="19">TN3_overdispersion &lt;-<span class="st">  </span><span class="kw">map_dfr</span>(TN3_bincounts, </a>
<a class="sourceLine" id="cb10-20" data-line-number="20">                    overdispersion) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-21" data-line-number="21"><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-22" data-line-number="22"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-23" data-line-number="23"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN3&quot;</span>,</a>
<a class="sourceLine" id="cb10-24" data-line-number="24">         <span class="dt">tech =</span> <span class="st">&quot;ACT&quot;</span>,</a>
<a class="sourceLine" id="cb10-25" data-line-number="25">         <span class="dt">cells =</span> <span class="kw">names</span>(TN3_bincounts)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-26" data-line-number="26"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a>
<a class="sourceLine" id="cb10-27" data-line-number="27"></a>
<a class="sourceLine" id="cb10-28" data-line-number="28">TN4_overdispersion &lt;-<span class="st">  </span><span class="kw">map_dfr</span>(TN4_bincounts, </a>
<a class="sourceLine" id="cb10-29" data-line-number="29">                    overdispersion) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-30" data-line-number="30"><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-31" data-line-number="31"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-32" data-line-number="32"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN4&quot;</span>,</a>
<a class="sourceLine" id="cb10-33" data-line-number="33">         <span class="dt">tech =</span> <span class="st">&quot;ACT&quot;</span>,</a>
<a class="sourceLine" id="cb10-34" data-line-number="34">         <span class="dt">cells =</span> <span class="kw">names</span>(TN4_bincounts)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-35" data-line-number="35"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a>
<a class="sourceLine" id="cb10-36" data-line-number="36"></a>
<a class="sourceLine" id="cb10-37" data-line-number="37"><span class="co">#10X CNA</span></a>
<a class="sourceLine" id="cb10-38" data-line-number="38"><span class="kw">names</span>(TN1_10XCNA_bincounts) &lt;-<span class="st"> </span><span class="kw">str_replace</span>(<span class="kw">names</span>(TN1_10XCNA_bincounts), <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;-&quot;</span>)</a>
<a class="sourceLine" id="cb10-39" data-line-number="39"></a>
<a class="sourceLine" id="cb10-40" data-line-number="40">TN1_10XCNA_overdispersion &lt;-<span class="st"> </span><span class="kw">map_dfr</span>(TN1_10XCNA_bincounts, </a>
<a class="sourceLine" id="cb10-41" data-line-number="41">                          overdispersion) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-42" data-line-number="42"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN1&quot;</span>,</a>
<a class="sourceLine" id="cb10-43" data-line-number="43">         <span class="dt">tech =</span> <span class="st">&quot;10X CNA&quot;</span>,</a>
<a class="sourceLine" id="cb10-44" data-line-number="44">         <span class="dt">cells =</span> <span class="kw">names</span>(TN1_10XCNA_bincounts)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-45" data-line-number="45"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a>
<a class="sourceLine" id="cb10-46" data-line-number="46"></a>
<a class="sourceLine" id="cb10-47" data-line-number="47"><span class="kw">names</span>(TN3_10XCNA_bincounts) &lt;-<span class="st"> </span><span class="kw">str_replace</span>(<span class="kw">names</span>(TN3_10XCNA_bincounts), <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;-&quot;</span>)</a>
<a class="sourceLine" id="cb10-48" data-line-number="48"></a>
<a class="sourceLine" id="cb10-49" data-line-number="49">TN3_10XCNA_overdispersion &lt;-<span class="st"> </span><span class="kw">map_dfr</span>(TN3_10XCNA_bincounts, </a>
<a class="sourceLine" id="cb10-50" data-line-number="50">                          overdispersion) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-51" data-line-number="51"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;TN3&quot;</span>,</a>
<a class="sourceLine" id="cb10-52" data-line-number="52">         <span class="dt">tech =</span> <span class="st">&quot;10X CNA&quot;</span>,</a>
<a class="sourceLine" id="cb10-53" data-line-number="53">         <span class="dt">cells =</span> <span class="kw">names</span>(TN3_10XCNA_bincounts)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-54" data-line-number="54"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a>
<a class="sourceLine" id="cb10-55" data-line-number="55"></a>
<a class="sourceLine" id="cb10-56" data-line-number="56"><span class="co"># DLP</span></a>
<a class="sourceLine" id="cb10-57" data-line-number="57"></a>
<a class="sourceLine" id="cb10-58" data-line-number="58">DLP_xeno_overdispersion &lt;-<span class="st">  </span><span class="kw">map_dfr</span>(DLP_xeno_bincounts[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)], </a>
<a class="sourceLine" id="cb10-59" data-line-number="59">                    overdispersion) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-60" data-line-number="60"><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-61" data-line-number="61"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-62" data-line-number="62"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;Xeno&quot;</span>,</a>
<a class="sourceLine" id="cb10-63" data-line-number="63">         <span class="dt">tech =</span> <span class="st">&quot;DLP&quot;</span>,</a>
<a class="sourceLine" id="cb10-64" data-line-number="64">         <span class="dt">cells =</span> <span class="kw">names</span>(DLP_xeno_bincounts)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)]) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-65" data-line-number="65"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a>
<a class="sourceLine" id="cb10-66" data-line-number="66"></a>
<a class="sourceLine" id="cb10-67" data-line-number="67">DLP_htert_overdispersion &lt;-<span class="st">  </span><span class="kw">map_dfr</span>(DLP_htert_bincounts[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)], </a>
<a class="sourceLine" id="cb10-68" data-line-number="68">                    overdispersion) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-69" data-line-number="69"><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-70" data-line-number="70"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-71" data-line-number="71"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;hTERT&quot;</span>,</a>
<a class="sourceLine" id="cb10-72" data-line-number="72">         <span class="dt">tech =</span> <span class="st">&quot;DLP&quot;</span>,</a>
<a class="sourceLine" id="cb10-73" data-line-number="73">         <span class="dt">cells =</span> <span class="kw">names</span>(DLP_htert_bincounts)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)]) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-74" data-line-number="74"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a>
<a class="sourceLine" id="cb10-75" data-line-number="75"></a>
<a class="sourceLine" id="cb10-76" data-line-number="76"><span class="co">#DOP-PCR</span></a>
<a class="sourceLine" id="cb10-77" data-line-number="77">T2_DOPPCR_overdispersion &lt;-<span class="st">  </span><span class="kw">map_dfr</span>(T2_DOPPCR_bincounts[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)], </a>
<a class="sourceLine" id="cb10-78" data-line-number="78">                    overdispersion) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-79" data-line-number="79"><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-80" data-line-number="80"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-81" data-line-number="81"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;T2&quot;</span>,</a>
<a class="sourceLine" id="cb10-82" data-line-number="82">         <span class="dt">tech =</span> <span class="st">&quot;DOP-PCR&quot;</span>,</a>
<a class="sourceLine" id="cb10-83" data-line-number="83">         <span class="dt">cells =</span> <span class="kw">toupper</span>(<span class="kw">names</span>(T2_DOPPCR_bincounts)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)])) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-84" data-line-number="84"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a>
<a class="sourceLine" id="cb10-85" data-line-number="85"></a>
<a class="sourceLine" id="cb10-86" data-line-number="86">T4_DOPPCR_overdispersion &lt;-<span class="st">  </span><span class="kw">map_dfr</span>(T4_DOPPCR_bincounts[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)], </a>
<a class="sourceLine" id="cb10-87" data-line-number="87">                    overdispersion) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-88" data-line-number="88"><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-89" data-line-number="89"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-90" data-line-number="90"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;T4&quot;</span>,</a>
<a class="sourceLine" id="cb10-91" data-line-number="91">         <span class="dt">tech =</span> <span class="st">&quot;DOP-PCR&quot;</span>,</a>
<a class="sourceLine" id="cb10-92" data-line-number="92">         <span class="dt">cells =</span> <span class="kw">toupper</span>(<span class="kw">names</span>(T4_DOPPCR_bincounts)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)])) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-93" data-line-number="93"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a>
<a class="sourceLine" id="cb10-94" data-line-number="94"></a>
<a class="sourceLine" id="cb10-95" data-line-number="95">T8_DOPPCR_overdispersion &lt;-<span class="st">  </span><span class="kw">map_dfr</span>(T8_DOPPCR_bincounts[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)], </a>
<a class="sourceLine" id="cb10-96" data-line-number="96">                    overdispersion) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-97" data-line-number="97"><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-98" data-line-number="98"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-99" data-line-number="99"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;T8&quot;</span>,</a>
<a class="sourceLine" id="cb10-100" data-line-number="100">         <span class="dt">tech =</span> <span class="st">&quot;DOP-PCR&quot;</span>,</a>
<a class="sourceLine" id="cb10-101" data-line-number="101">         <span class="dt">cells =</span> <span class="kw">toupper</span>(<span class="kw">names</span>(T8_DOPPCR_bincounts)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)])) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-102" data-line-number="102"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a>
<a class="sourceLine" id="cb10-103" data-line-number="103"></a>
<a class="sourceLine" id="cb10-104" data-line-number="104">T10_DOPPCR_overdispersion &lt;-<span class="st">  </span><span class="kw">map_dfr</span>(T10_DOPPCR_bincounts[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)], </a>
<a class="sourceLine" id="cb10-105" data-line-number="105">                    overdispersion) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-106" data-line-number="106"><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-107" data-line-number="107"><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-108" data-line-number="108"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">&quot;T10&quot;</span>,</a>
<a class="sourceLine" id="cb10-109" data-line-number="109">         <span class="dt">tech =</span> <span class="st">&quot;DOP-PCR&quot;</span>,</a>
<a class="sourceLine" id="cb10-110" data-line-number="110">         <span class="dt">cells =</span> <span class="kw">toupper</span>(<span class="kw">names</span>(T10_DOPPCR_bincounts)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)])) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-111" data-line-number="111"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">iod =</span> <span class="st">&quot;V1&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">all_overdispersion &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">  TN1_overdispersion,</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">  TN2_overdispersion,</a>
<a class="sourceLine" id="cb11-4" data-line-number="4">  TN3_overdispersion,</a>
<a class="sourceLine" id="cb11-5" data-line-number="5">  TN4_overdispersion,</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">  T2_DOPPCR_overdispersion,</a>
<a class="sourceLine" id="cb11-7" data-line-number="7">  T4_DOPPCR_overdispersion,</a>
<a class="sourceLine" id="cb11-8" data-line-number="8">  T8_DOPPCR_overdispersion,</a>
<a class="sourceLine" id="cb11-9" data-line-number="9">  T10_DOPPCR_overdispersion,</a>
<a class="sourceLine" id="cb11-10" data-line-number="10">  DLP_htert_overdispersion,</a>
<a class="sourceLine" id="cb11-11" data-line-number="11">  DLP_xeno_overdispersion,</a>
<a class="sourceLine" id="cb11-12" data-line-number="12">  TN1_10XCNA_overdispersion,</a>
<a class="sourceLine" id="cb11-13" data-line-number="13">  TN3_10XCNA_overdispersion</a>
<a class="sourceLine" id="cb11-14" data-line-number="14">)</a>
<a class="sourceLine" id="cb11-15" data-line-number="15"></a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="co"># subsetting same cells sampled during the BAM file sampling for coverage breadth</span></a>
<a class="sourceLine" id="cb11-17" data-line-number="17">all_overdispersion &lt;-<span class="st"> </span>all_overdispersion <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-18" data-line-number="18"><span class="st">  </span><span class="kw">filter</span>(cells <span class="op">%in%</span><span class="st"> </span>all_cov<span class="op">$</span>cellname)</a>
<a class="sourceLine" id="cb11-19" data-line-number="19"></a>
<a class="sourceLine" id="cb11-20" data-line-number="20">p_iod &lt;-<span class="st"> </span>all_overdispersion <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">tech =</span> <span class="kw">as.factor</span>(tech)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-21" data-line-number="21"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">as.factor</span>(sample)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-22" data-line-number="22"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">fct_relevel</span>(sample, gtools<span class="op">::</span><span class="kw">mixedsort</span>(<span class="kw">unique</span>(all_cov<span class="op">$</span>sample)))) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-23" data-line-number="23"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tech =</span> <span class="kw">fct_relevel</span>(tech, <span class="kw">c</span>(<span class="st">&quot;ACT&quot;</span>, <span class="st">&quot;10X CNA&quot;</span>, <span class="st">&quot;DLP&quot;</span>, <span class="st">&quot;DOP-PCR&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-24" data-line-number="24"><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb11-25" data-line-number="25"><span class="st">  </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">x =</span> sample,</a>
<a class="sourceLine" id="cb11-26" data-line-number="26">                  <span class="dt">y =</span> iod,</a>
<a class="sourceLine" id="cb11-27" data-line-number="27">                  <span class="dt">color =</span> tech)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-28" data-line-number="28"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">paletteer_d</span>(<span class="st">&quot;nord::lumina&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-29" data-line-number="29"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="kw">vars</span>(tech), <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-30" data-line-number="30"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> scales<span class="op">::</span><span class="kw">pretty_breaks</span>(<span class="dt">n=</span><span class="dv">10</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-31" data-line-number="31"><span class="st">  </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb11-32" data-line-number="32"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>,</a>
<a class="sourceLine" id="cb11-33" data-line-number="33">                                   <span class="dt">hjust =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb11-34" data-line-number="34">                                   <span class="dt">vjust =</span> <span class="fl">0.5</span>),</a>
<a class="sourceLine" id="cb11-35" data-line-number="35">        <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,</a>
<a class="sourceLine" id="cb11-36" data-line-number="36">        <span class="dt">strip.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;white&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-37" data-line-number="37"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;overdispersion&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-38" data-line-number="38"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)</a>
<a class="sourceLine" id="cb11-39" data-line-number="39">p_iod</a></code></pre></div>
<p><img src="_main_files/figure-html/plots_overdispersion-1.png" width="672" /></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">p_iod_tech &lt;-</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="st">  </span>all_overdispersion <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">fct_relevel</span>(tech, <span class="kw">c</span>(<span class="st">&quot;ACT&quot;</span>, <span class="st">&quot;10X CNA&quot;</span>, <span class="st">&quot;DLP&quot;</span>,<span class="st">&quot;DOP-PCR&quot;</span>)),</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">             <span class="dt">y =</span> iod)) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="st">  </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">color =</span> tech)) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> scales<span class="op">::</span><span class="kw">pretty_breaks</span>(<span class="dt">n=</span><span class="dv">5</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="st">  </span><span class="kw">scale_color_paletteer_d</span>(<span class="st">&quot;nord::lumina&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="st">  </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;overdispersion&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-11" data-line-number="11"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>)</a>
<a class="sourceLine" id="cb12-12" data-line-number="12">p_iod_tech</a></code></pre></div>
<p><img src="_main_files/figure-html/plots_overdispersion-2.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="instructions.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="tumors.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
